// @pulumi/aws-native/codedeploy/deploymentConfig.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::CodeDeploy::DeploymentConfig
 */
class DeploymentConfig extends pulumi.CustomResource {
    /**
     * Create a DeploymentConfig resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["computePlatform"] = args ? args.computePlatform : undefined;
            resourceInputs["deploymentConfigName"] = args ? args.deploymentConfigName : undefined;
            resourceInputs["minimumHealthyHosts"] = args ? args.minimumHealthyHosts : undefined;
            resourceInputs["trafficRoutingConfig"] = args ? args.trafficRoutingConfig : undefined;
        }
        else {
            resourceInputs["computePlatform"] = undefined /*out*/;
            resourceInputs["deploymentConfigName"] = undefined /*out*/;
            resourceInputs["minimumHealthyHosts"] = undefined /*out*/;
            resourceInputs["trafficRoutingConfig"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DeploymentConfig.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DeploymentConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DeploymentConfig(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DeploymentConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DeploymentConfig.__pulumiType;
    }
}
exports.DeploymentConfig = DeploymentConfig;
/** @internal */
DeploymentConfig.__pulumiType = 'aws-native:codedeploy:DeploymentConfig';
