// @pulumi/aws-native/types/enums/apprunner/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcIngressConnectionStatus = exports.ServiceSourceCodeVersionType = exports.ServiceImageRepositoryImageRepositoryType = exports.ServiceHealthCheckConfigurationProtocol = exports.ServiceEgressConfigurationEgressType = exports.ServiceCodeConfigurationValuesRuntime = exports.ServiceCodeConfigurationConfigurationSource = exports.ObservabilityConfigurationTraceConfigurationVendor = void 0;
exports.ObservabilityConfigurationTraceConfigurationVendor = {
    Awsxray: "AWSXRAY",
};
exports.ServiceCodeConfigurationConfigurationSource = {
    Repository: "REPOSITORY",
    Api: "API",
};
exports.ServiceCodeConfigurationValuesRuntime = {
    Python3: "PYTHON_3",
    Nodejs12: "NODEJS_12",
    Nodejs14: "NODEJS_14",
    Corretto8: "CORRETTO_8",
    Corretto11: "CORRETTO_11",
    Nodejs16: "NODEJS_16",
    Go1: "GO_1",
    Dotnet6: "DOTNET_6",
    Php81: "PHP_81",
    Ruby31: "RUBY_31",
};
exports.ServiceEgressConfigurationEgressType = {
    Default: "DEFAULT",
    Vpc: "VPC",
};
exports.ServiceHealthCheckConfigurationProtocol = {
    Tcp: "TCP",
    Http: "HTTP",
};
exports.ServiceImageRepositoryImageRepositoryType = {
    Ecr: "ECR",
    EcrPublic: "ECR_PUBLIC",
};
exports.ServiceSourceCodeVersionType = {
    Branch: "BRANCH",
};
exports.VpcIngressConnectionStatus = {
    Available: "AVAILABLE",
    PendingCreation: "PENDING_CREATION",
    PendingUpdate: "PENDING_UPDATE",
    PendingDeletion: "PENDING_DELETION",
    FailedCreation: "FAILED_CREATION",
    FailedUpdate: "FAILED_UPDATE",
    FailedDeletion: "FAILED_DELETION",
    Deleted: "DELETED",
};
