// @pulumi/aws-native/groundstation/missionProfile.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MissionProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * AWS Ground Station Mission Profile resource type for CloudFormation.
 */
class MissionProfile extends pulumi.CustomResource {
    /**
     * Create a MissionProfile resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dataflowEdges === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataflowEdges'");
            }
            if ((!args || args.minimumViableContactDurationSeconds === undefined) && !opts.urn) {
                throw new Error("Missing required property 'minimumViableContactDurationSeconds'");
            }
            if ((!args || args.trackingConfigArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'trackingConfigArn'");
            }
            resourceInputs["contactPostPassDurationSeconds"] = args ? args.contactPostPassDurationSeconds : undefined;
            resourceInputs["contactPrePassDurationSeconds"] = args ? args.contactPrePassDurationSeconds : undefined;
            resourceInputs["dataflowEdges"] = args ? args.dataflowEdges : undefined;
            resourceInputs["minimumViableContactDurationSeconds"] = args ? args.minimumViableContactDurationSeconds : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["trackingConfigArn"] = args ? args.trackingConfigArn : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["region"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["contactPostPassDurationSeconds"] = undefined /*out*/;
            resourceInputs["contactPrePassDurationSeconds"] = undefined /*out*/;
            resourceInputs["dataflowEdges"] = undefined /*out*/;
            resourceInputs["minimumViableContactDurationSeconds"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["region"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["trackingConfigArn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MissionProfile.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing MissionProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MissionProfile(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MissionProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MissionProfile.__pulumiType;
    }
}
exports.MissionProfile = MissionProfile;
/** @internal */
MissionProfile.__pulumiType = 'aws-native:groundstation:MissionProfile';
