// @pulumi/aws-native/route53/recordSet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecordSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Route53::RecordSet
 *
 * @deprecated RecordSet is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class RecordSet extends pulumi.CustomResource {
    /**
     * Create a RecordSet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated RecordSet is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("RecordSet is deprecated: RecordSet is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["aliasTarget"] = args ? args.aliasTarget : undefined;
            resourceInputs["cidrRoutingConfig"] = args ? args.cidrRoutingConfig : undefined;
            resourceInputs["comment"] = args ? args.comment : undefined;
            resourceInputs["failover"] = args ? args.failover : undefined;
            resourceInputs["geoLocation"] = args ? args.geoLocation : undefined;
            resourceInputs["healthCheckId"] = args ? args.healthCheckId : undefined;
            resourceInputs["hostedZoneId"] = args ? args.hostedZoneId : undefined;
            resourceInputs["hostedZoneName"] = args ? args.hostedZoneName : undefined;
            resourceInputs["multiValueAnswer"] = args ? args.multiValueAnswer : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["region"] = args ? args.region : undefined;
            resourceInputs["resourceRecords"] = args ? args.resourceRecords : undefined;
            resourceInputs["setIdentifier"] = args ? args.setIdentifier : undefined;
            resourceInputs["tTL"] = args ? args.tTL : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["weight"] = args ? args.weight : undefined;
        }
        else {
            resourceInputs["aliasTarget"] = undefined /*out*/;
            resourceInputs["cidrRoutingConfig"] = undefined /*out*/;
            resourceInputs["comment"] = undefined /*out*/;
            resourceInputs["failover"] = undefined /*out*/;
            resourceInputs["geoLocation"] = undefined /*out*/;
            resourceInputs["healthCheckId"] = undefined /*out*/;
            resourceInputs["hostedZoneId"] = undefined /*out*/;
            resourceInputs["hostedZoneName"] = undefined /*out*/;
            resourceInputs["multiValueAnswer"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["region"] = undefined /*out*/;
            resourceInputs["resourceRecords"] = undefined /*out*/;
            resourceInputs["setIdentifier"] = undefined /*out*/;
            resourceInputs["tTL"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["weight"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RecordSet.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing RecordSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("RecordSet is deprecated: RecordSet is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new RecordSet(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RecordSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RecordSet.__pulumiType;
    }
}
exports.RecordSet = RecordSet;
/** @internal */
RecordSet.__pulumiType = 'aws-native:route53:RecordSet';
