// @pulumi/aws-native/types/enums/mediapackage/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackagingConfigurationStreamSelectionStreamOrder = exports.PackagingConfigurationHlsManifestAdMarkers = exports.PackagingConfigurationHlsEncryptionEncryptionMethod = exports.PackagingConfigurationDashPackageSegmentTemplateFormat = exports.PackagingConfigurationDashPackagePeriodTriggersItem = exports.PackagingConfigurationDashManifestScteMarkersSource = exports.PackagingConfigurationDashManifestProfile = exports.PackagingConfigurationDashManifestManifestLayout = exports.OriginEndpointStreamSelectionStreamOrder = exports.OriginEndpointOrigination = exports.OriginEndpointHlsPackagePlaylistType = exports.OriginEndpointHlsPackageAdTriggersItem = exports.OriginEndpointHlsPackageAdMarkers = exports.OriginEndpointHlsManifestPlaylistType = exports.OriginEndpointHlsManifestAdTriggersItem = exports.OriginEndpointHlsManifestAdMarkers = exports.OriginEndpointHlsEncryptionEncryptionMethod = exports.OriginEndpointEncryptionContractConfigurationPresetSpeke20Video = exports.OriginEndpointEncryptionContractConfigurationPresetSpeke20Audio = exports.OriginEndpointDashPackageUtcTiming = exports.OriginEndpointDashPackageSegmentTemplateFormat = exports.OriginEndpointDashPackageProfile = exports.OriginEndpointDashPackagePeriodTriggersItem = exports.OriginEndpointDashPackageManifestLayout = exports.OriginEndpointDashPackageAdTriggersItem = exports.OriginEndpointCmafEncryptionEncryptionMethod = exports.OriginEndpointAdsOnDeliveryRestrictions = void 0;
exports.OriginEndpointAdsOnDeliveryRestrictions = {
    None: "NONE",
    Restricted: "RESTRICTED",
    Unrestricted: "UNRESTRICTED",
    Both: "BOTH",
};
exports.OriginEndpointCmafEncryptionEncryptionMethod = {
    SampleAes: "SAMPLE_AES",
    AesCtr: "AES_CTR",
};
exports.OriginEndpointDashPackageAdTriggersItem = {
    SpliceInsert: "SPLICE_INSERT",
    Break: "BREAK",
    ProviderAdvertisement: "PROVIDER_ADVERTISEMENT",
    DistributorAdvertisement: "DISTRIBUTOR_ADVERTISEMENT",
    ProviderPlacementOpportunity: "PROVIDER_PLACEMENT_OPPORTUNITY",
    DistributorPlacementOpportunity: "DISTRIBUTOR_PLACEMENT_OPPORTUNITY",
    ProviderOverlayPlacementOpportunity: "PROVIDER_OVERLAY_PLACEMENT_OPPORTUNITY",
    DistributorOverlayPlacementOpportunity: "DISTRIBUTOR_OVERLAY_PLACEMENT_OPPORTUNITY",
};
exports.OriginEndpointDashPackageManifestLayout = {
    Full: "FULL",
    Compact: "COMPACT",
};
exports.OriginEndpointDashPackagePeriodTriggersItem = {
    Ads: "ADS",
};
exports.OriginEndpointDashPackageProfile = {
    None: "NONE",
    Hbbtv15: "HBBTV_1_5",
    Hybridcast: "HYBRIDCAST",
    DvbDash2014: "DVB_DASH_2014",
};
exports.OriginEndpointDashPackageSegmentTemplateFormat = {
    NumberWithTimeline: "NUMBER_WITH_TIMELINE",
    TimeWithTimeline: "TIME_WITH_TIMELINE",
    NumberWithDuration: "NUMBER_WITH_DURATION",
};
exports.OriginEndpointDashPackageUtcTiming = {
    HttpXsdate: "HTTP-XSDATE",
    HttpIso: "HTTP-ISO",
    HttpHead: "HTTP-HEAD",
    None: "NONE",
};
exports.OriginEndpointEncryptionContractConfigurationPresetSpeke20Audio = {
    PresetAudio1: "PRESET-AUDIO-1",
    PresetAudio2: "PRESET-AUDIO-2",
    PresetAudio3: "PRESET-AUDIO-3",
    Shared: "SHARED",
    Unencrypted: "UNENCRYPTED",
};
exports.OriginEndpointEncryptionContractConfigurationPresetSpeke20Video = {
    PresetVideo1: "PRESET-VIDEO-1",
    PresetVideo2: "PRESET-VIDEO-2",
    PresetVideo3: "PRESET-VIDEO-3",
    PresetVideo4: "PRESET-VIDEO-4",
    PresetVideo5: "PRESET-VIDEO-5",
    PresetVideo6: "PRESET-VIDEO-6",
    PresetVideo7: "PRESET-VIDEO-7",
    PresetVideo8: "PRESET-VIDEO-8",
    Shared: "SHARED",
    Unencrypted: "UNENCRYPTED",
};
exports.OriginEndpointHlsEncryptionEncryptionMethod = {
    Aes128: "AES_128",
    SampleAes: "SAMPLE_AES",
};
exports.OriginEndpointHlsManifestAdMarkers = {
    None: "NONE",
    Scte35Enhanced: "SCTE35_ENHANCED",
    Passthrough: "PASSTHROUGH",
    Daterange: "DATERANGE",
};
exports.OriginEndpointHlsManifestAdTriggersItem = {
    SpliceInsert: "SPLICE_INSERT",
    Break: "BREAK",
    ProviderAdvertisement: "PROVIDER_ADVERTISEMENT",
    DistributorAdvertisement: "DISTRIBUTOR_ADVERTISEMENT",
    ProviderPlacementOpportunity: "PROVIDER_PLACEMENT_OPPORTUNITY",
    DistributorPlacementOpportunity: "DISTRIBUTOR_PLACEMENT_OPPORTUNITY",
    ProviderOverlayPlacementOpportunity: "PROVIDER_OVERLAY_PLACEMENT_OPPORTUNITY",
    DistributorOverlayPlacementOpportunity: "DISTRIBUTOR_OVERLAY_PLACEMENT_OPPORTUNITY",
};
exports.OriginEndpointHlsManifestPlaylistType = {
    None: "NONE",
    Event: "EVENT",
    Vod: "VOD",
};
exports.OriginEndpointHlsPackageAdMarkers = {
    None: "NONE",
    Scte35Enhanced: "SCTE35_ENHANCED",
    Passthrough: "PASSTHROUGH",
    Daterange: "DATERANGE",
};
exports.OriginEndpointHlsPackageAdTriggersItem = {
    SpliceInsert: "SPLICE_INSERT",
    Break: "BREAK",
    ProviderAdvertisement: "PROVIDER_ADVERTISEMENT",
    DistributorAdvertisement: "DISTRIBUTOR_ADVERTISEMENT",
    ProviderPlacementOpportunity: "PROVIDER_PLACEMENT_OPPORTUNITY",
    DistributorPlacementOpportunity: "DISTRIBUTOR_PLACEMENT_OPPORTUNITY",
    ProviderOverlayPlacementOpportunity: "PROVIDER_OVERLAY_PLACEMENT_OPPORTUNITY",
    DistributorOverlayPlacementOpportunity: "DISTRIBUTOR_OVERLAY_PLACEMENT_OPPORTUNITY",
};
exports.OriginEndpointHlsPackagePlaylistType = {
    None: "NONE",
    Event: "EVENT",
    Vod: "VOD",
};
exports.OriginEndpointOrigination = {
    Allow: "ALLOW",
    Deny: "DENY",
};
exports.OriginEndpointStreamSelectionStreamOrder = {
    Original: "ORIGINAL",
    VideoBitrateAscending: "VIDEO_BITRATE_ASCENDING",
    VideoBitrateDescending: "VIDEO_BITRATE_DESCENDING",
};
exports.PackagingConfigurationDashManifestManifestLayout = {
    Full: "FULL",
    Compact: "COMPACT",
};
exports.PackagingConfigurationDashManifestProfile = {
    None: "NONE",
    Hbbtv15: "HBBTV_1_5",
};
exports.PackagingConfigurationDashManifestScteMarkersSource = {
    Segments: "SEGMENTS",
    Manifest: "MANIFEST",
};
exports.PackagingConfigurationDashPackagePeriodTriggersItem = {
    Ads: "ADS",
};
exports.PackagingConfigurationDashPackageSegmentTemplateFormat = {
    NumberWithTimeline: "NUMBER_WITH_TIMELINE",
    TimeWithTimeline: "TIME_WITH_TIMELINE",
    NumberWithDuration: "NUMBER_WITH_DURATION",
};
exports.PackagingConfigurationHlsEncryptionEncryptionMethod = {
    Aes128: "AES_128",
    SampleAes: "SAMPLE_AES",
};
exports.PackagingConfigurationHlsManifestAdMarkers = {
    None: "NONE",
    Scte35Enhanced: "SCTE35_ENHANCED",
    Passthrough: "PASSTHROUGH",
};
exports.PackagingConfigurationStreamSelectionStreamOrder = {
    Original: "ORIGINAL",
    VideoBitrateAscending: "VIDEO_BITRATE_ASCENDING",
    VideoBitrateDescending: "VIDEO_BITRATE_DESCENDING",
};
