// @pulumi/aws-native/synthetics/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Group = exports.getGroupOutput = exports.getGroup = exports.getCanaryOutput = exports.getCanary = exports.Canary = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Canary = null;
utilities.lazyLoad(exports, ["Canary"], () => require("./canary"));
exports.getCanary = null;
exports.getCanaryOutput = null;
utilities.lazyLoad(exports, ["getCanary", "getCanaryOutput"], () => require("./getCanary"));
exports.getGroup = null;
exports.getGroupOutput = null;
utilities.lazyLoad(exports, ["getGroup", "getGroupOutput"], () => require("./getGroup"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:synthetics:Canary":
                return new exports.Canary(name, undefined, { urn });
            case "aws-native:synthetics:Group":
                return new exports.Group(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "synthetics", _module);
