// @pulumi/aws-native/types/enums/cloudformation/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TypeActivationVersionBump = exports.TypeActivationType = exports.StackSetRegionConcurrencyType = exports.StackSetPermissionModel = exports.StackSetDeploymentTargetsAccountFilterType = exports.StackSetCapability = exports.StackSetCallAs = exports.ResourceVersionVisibility = exports.ResourceVersionProvisioningType = exports.PublisherStatus = exports.PublisherIdentityProvider = exports.PublicTypeVersionType = exports.ModuleVersionVisibility = exports.HookVersionVisibility = exports.HookTypeConfigConfigurationAlias = void 0;
exports.HookTypeConfigConfigurationAlias = {
    Default: "default",
};
exports.HookVersionVisibility = {
    Public: "PUBLIC",
    Private: "PRIVATE",
};
exports.ModuleVersionVisibility = {
    Private: "PRIVATE",
};
exports.PublicTypeVersionType = {
    Resource: "RESOURCE",
    Module: "MODULE",
    Hook: "HOOK",
};
exports.PublisherIdentityProvider = {
    AWSMarketplace: "AWS_Marketplace",
    GitHub: "GitHub",
    Bitbucket: "Bitbucket",
};
exports.PublisherStatus = {
    Verified: "VERIFIED",
    Unverified: "UNVERIFIED",
};
exports.ResourceVersionProvisioningType = {
    NonProvisionable: "NON_PROVISIONABLE",
    Immutable: "IMMUTABLE",
    FullyMutable: "FULLY_MUTABLE",
};
exports.ResourceVersionVisibility = {
    Public: "PUBLIC",
    Private: "PRIVATE",
};
exports.StackSetCallAs = {
    Self: "SELF",
    DelegatedAdmin: "DELEGATED_ADMIN",
};
exports.StackSetCapability = {
    CapabilityIam: "CAPABILITY_IAM",
    CapabilityNamedIam: "CAPABILITY_NAMED_IAM",
    CapabilityAutoExpand: "CAPABILITY_AUTO_EXPAND",
};
exports.StackSetDeploymentTargetsAccountFilterType = {
    None: "NONE",
    Union: "UNION",
    Intersection: "INTERSECTION",
    Difference: "DIFFERENCE",
};
exports.StackSetPermissionModel = {
    ServiceManaged: "SERVICE_MANAGED",
    SelfManaged: "SELF_MANAGED",
};
exports.StackSetRegionConcurrencyType = {
    Sequential: "SEQUENTIAL",
    Parallel: "PARALLEL",
};
exports.TypeActivationType = {
    Resource: "RESOURCE",
    Module: "MODULE",
    Hook: "HOOK",
};
exports.TypeActivationVersionBump = {
    Major: "MAJOR",
    Minor: "MINOR",
};
