// @pulumi/aws-native/emr/cluster.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::EMR::Cluster
 *
 * @deprecated Cluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Cluster extends pulumi.CustomResource {
    /**
     * Create a Cluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Cluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Cluster is deprecated: Cluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.instances === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instances'");
            }
            if ((!args || args.jobFlowRole === undefined) && !opts.urn) {
                throw new Error("Missing required property 'jobFlowRole'");
            }
            if ((!args || args.serviceRole === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceRole'");
            }
            resourceInputs["additionalInfo"] = args ? args.additionalInfo : undefined;
            resourceInputs["applications"] = args ? args.applications : undefined;
            resourceInputs["autoScalingRole"] = args ? args.autoScalingRole : undefined;
            resourceInputs["autoTerminationPolicy"] = args ? args.autoTerminationPolicy : undefined;
            resourceInputs["bootstrapActions"] = args ? args.bootstrapActions : undefined;
            resourceInputs["configurations"] = args ? args.configurations : undefined;
            resourceInputs["customAmiId"] = args ? args.customAmiId : undefined;
            resourceInputs["ebsRootVolumeSize"] = args ? args.ebsRootVolumeSize : undefined;
            resourceInputs["instances"] = args ? args.instances : undefined;
            resourceInputs["jobFlowRole"] = args ? args.jobFlowRole : undefined;
            resourceInputs["kerberosAttributes"] = args ? args.kerberosAttributes : undefined;
            resourceInputs["logEncryptionKmsKeyId"] = args ? args.logEncryptionKmsKeyId : undefined;
            resourceInputs["logUri"] = args ? args.logUri : undefined;
            resourceInputs["managedScalingPolicy"] = args ? args.managedScalingPolicy : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["releaseLabel"] = args ? args.releaseLabel : undefined;
            resourceInputs["scaleDownBehavior"] = args ? args.scaleDownBehavior : undefined;
            resourceInputs["securityConfiguration"] = args ? args.securityConfiguration : undefined;
            resourceInputs["serviceRole"] = args ? args.serviceRole : undefined;
            resourceInputs["stepConcurrencyLevel"] = args ? args.stepConcurrencyLevel : undefined;
            resourceInputs["steps"] = args ? args.steps : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["visibleToAllUsers"] = args ? args.visibleToAllUsers : undefined;
            resourceInputs["masterPublicDNS"] = undefined /*out*/;
        }
        else {
            resourceInputs["additionalInfo"] = undefined /*out*/;
            resourceInputs["applications"] = undefined /*out*/;
            resourceInputs["autoScalingRole"] = undefined /*out*/;
            resourceInputs["autoTerminationPolicy"] = undefined /*out*/;
            resourceInputs["bootstrapActions"] = undefined /*out*/;
            resourceInputs["configurations"] = undefined /*out*/;
            resourceInputs["customAmiId"] = undefined /*out*/;
            resourceInputs["ebsRootVolumeSize"] = undefined /*out*/;
            resourceInputs["instances"] = undefined /*out*/;
            resourceInputs["jobFlowRole"] = undefined /*out*/;
            resourceInputs["kerberosAttributes"] = undefined /*out*/;
            resourceInputs["logEncryptionKmsKeyId"] = undefined /*out*/;
            resourceInputs["logUri"] = undefined /*out*/;
            resourceInputs["managedScalingPolicy"] = undefined /*out*/;
            resourceInputs["masterPublicDNS"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["releaseLabel"] = undefined /*out*/;
            resourceInputs["scaleDownBehavior"] = undefined /*out*/;
            resourceInputs["securityConfiguration"] = undefined /*out*/;
            resourceInputs["serviceRole"] = undefined /*out*/;
            resourceInputs["stepConcurrencyLevel"] = undefined /*out*/;
            resourceInputs["steps"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["visibleToAllUsers"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Cluster.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Cluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Cluster is deprecated: Cluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Cluster(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Cluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Cluster.__pulumiType;
    }
}
exports.Cluster = Cluster;
/** @internal */
Cluster.__pulumiType = 'aws-native:emr:Cluster';
