// @pulumi/aws-native/emr/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StudioSessionMapping = exports.Studio = exports.Step = exports.SecurityConfiguration = exports.InstanceGroupConfig = exports.InstanceFleetConfig = exports.getStudioSessionMappingOutput = exports.getStudioSessionMapping = exports.getStudioOutput = exports.getStudio = exports.getStepOutput = exports.getStep = exports.getInstanceGroupConfigOutput = exports.getInstanceGroupConfig = exports.getInstanceFleetConfigOutput = exports.getInstanceFleetConfig = exports.getClusterOutput = exports.getCluster = exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getInstanceFleetConfig = null;
exports.getInstanceFleetConfigOutput = null;
utilities.lazyLoad(exports, ["getInstanceFleetConfig", "getInstanceFleetConfigOutput"], () => require("./getInstanceFleetConfig"));
exports.getInstanceGroupConfig = null;
exports.getInstanceGroupConfigOutput = null;
utilities.lazyLoad(exports, ["getInstanceGroupConfig", "getInstanceGroupConfigOutput"], () => require("./getInstanceGroupConfig"));
exports.getStep = null;
exports.getStepOutput = null;
utilities.lazyLoad(exports, ["getStep", "getStepOutput"], () => require("./getStep"));
exports.getStudio = null;
exports.getStudioOutput = null;
utilities.lazyLoad(exports, ["getStudio", "getStudioOutput"], () => require("./getStudio"));
exports.getStudioSessionMapping = null;
exports.getStudioSessionMappingOutput = null;
utilities.lazyLoad(exports, ["getStudioSessionMapping", "getStudioSessionMappingOutput"], () => require("./getStudioSessionMapping"));
exports.InstanceFleetConfig = null;
utilities.lazyLoad(exports, ["InstanceFleetConfig"], () => require("./instanceFleetConfig"));
exports.InstanceGroupConfig = null;
utilities.lazyLoad(exports, ["InstanceGroupConfig"], () => require("./instanceGroupConfig"));
exports.SecurityConfiguration = null;
utilities.lazyLoad(exports, ["SecurityConfiguration"], () => require("./securityConfiguration"));
exports.Step = null;
utilities.lazyLoad(exports, ["Step"], () => require("./step"));
exports.Studio = null;
utilities.lazyLoad(exports, ["Studio"], () => require("./studio"));
exports.StudioSessionMapping = null;
utilities.lazyLoad(exports, ["StudioSessionMapping"], () => require("./studioSessionMapping"));
// Export enums:
__exportStar(require("../types/enums/emr"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:emr:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "aws-native:emr:InstanceFleetConfig":
                return new exports.InstanceFleetConfig(name, undefined, { urn });
            case "aws-native:emr:InstanceGroupConfig":
                return new exports.InstanceGroupConfig(name, undefined, { urn });
            case "aws-native:emr:SecurityConfiguration":
                return new exports.SecurityConfiguration(name, undefined, { urn });
            case "aws-native:emr:Step":
                return new exports.Step(name, undefined, { urn });
            case "aws-native:emr:Studio":
                return new exports.Studio(name, undefined, { urn });
            case "aws-native:emr:StudioSessionMapping":
                return new exports.StudioSessionMapping(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "emr", _module);
