// @pulumi/aws-native/nimblestudio/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StudioComponent = exports.Studio = exports.StreamingImage = exports.LaunchProfile = exports.getStudioComponentOutput = exports.getStudioComponent = exports.getStudioOutput = exports.getStudio = exports.getStreamingImageOutput = exports.getStreamingImage = exports.getLaunchProfileOutput = exports.getLaunchProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getLaunchProfile = null;
exports.getLaunchProfileOutput = null;
utilities.lazyLoad(exports, ["getLaunchProfile", "getLaunchProfileOutput"], () => require("./getLaunchProfile"));
exports.getStreamingImage = null;
exports.getStreamingImageOutput = null;
utilities.lazyLoad(exports, ["getStreamingImage", "getStreamingImageOutput"], () => require("./getStreamingImage"));
exports.getStudio = null;
exports.getStudioOutput = null;
utilities.lazyLoad(exports, ["getStudio", "getStudioOutput"], () => require("./getStudio"));
exports.getStudioComponent = null;
exports.getStudioComponentOutput = null;
utilities.lazyLoad(exports, ["getStudioComponent", "getStudioComponentOutput"], () => require("./getStudioComponent"));
exports.LaunchProfile = null;
utilities.lazyLoad(exports, ["LaunchProfile"], () => require("./launchProfile"));
exports.StreamingImage = null;
utilities.lazyLoad(exports, ["StreamingImage"], () => require("./streamingImage"));
exports.Studio = null;
utilities.lazyLoad(exports, ["Studio"], () => require("./studio"));
exports.StudioComponent = null;
utilities.lazyLoad(exports, ["StudioComponent"], () => require("./studioComponent"));
// Export enums:
__exportStar(require("../types/enums/nimblestudio"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:nimblestudio:LaunchProfile":
                return new exports.LaunchProfile(name, undefined, { urn });
            case "aws-native:nimblestudio:StreamingImage":
                return new exports.StreamingImage(name, undefined, { urn });
            case "aws-native:nimblestudio:Studio":
                return new exports.Studio(name, undefined, { urn });
            case "aws-native:nimblestudio:StudioComponent":
                return new exports.StudioComponent(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "nimblestudio", _module);
