// @pulumi/aws-native/nimblestudio/streamingImage.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamingImage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a streaming session machine image that can be used to launch a streaming session
 */
class StreamingImage extends pulumi.CustomResource {
    /**
     * Create a StreamingImage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.ec2ImageId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ec2ImageId'");
            }
            if ((!args || args.studioId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'studioId'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["ec2ImageId"] = args ? args.ec2ImageId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["studioId"] = args ? args.studioId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["encryptionConfiguration"] = undefined /*out*/;
            resourceInputs["eulaIds"] = undefined /*out*/;
            resourceInputs["owner"] = undefined /*out*/;
            resourceInputs["platform"] = undefined /*out*/;
            resourceInputs["streamingImageId"] = undefined /*out*/;
        }
        else {
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["ec2ImageId"] = undefined /*out*/;
            resourceInputs["encryptionConfiguration"] = undefined /*out*/;
            resourceInputs["eulaIds"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["owner"] = undefined /*out*/;
            resourceInputs["platform"] = undefined /*out*/;
            resourceInputs["streamingImageId"] = undefined /*out*/;
            resourceInputs["studioId"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(StreamingImage.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing StreamingImage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StreamingImage(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StreamingImage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StreamingImage.__pulumiType;
    }
}
exports.StreamingImage = StreamingImage;
/** @internal */
StreamingImage.__pulumiType = 'aws-native:nimblestudio:StreamingImage';
