// @pulumi/aws-native/kinesisanalyticsv2/application.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Creates an Amazon Kinesis Data Analytics application. For information about creating a Kinesis Data Analytics application, see [Creating an Application](https://docs.aws.amazon.com/kinesisanalytics/latest/java/getting-started.html).
 */
class Application extends pulumi.CustomResource {
    /**
     * Create a Application resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.runtimeEnvironment === undefined) && !opts.urn) {
                throw new Error("Missing required property 'runtimeEnvironment'");
            }
            if ((!args || args.serviceExecutionRole === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceExecutionRole'");
            }
            resourceInputs["applicationConfiguration"] = args ? args.applicationConfiguration : undefined;
            resourceInputs["applicationDescription"] = args ? args.applicationDescription : undefined;
            resourceInputs["applicationMaintenanceConfiguration"] = args ? args.applicationMaintenanceConfiguration : undefined;
            resourceInputs["applicationMode"] = args ? args.applicationMode : undefined;
            resourceInputs["applicationName"] = args ? args.applicationName : undefined;
            resourceInputs["runConfiguration"] = args ? args.runConfiguration : undefined;
            resourceInputs["runtimeEnvironment"] = args ? args.runtimeEnvironment : undefined;
            resourceInputs["serviceExecutionRole"] = args ? args.serviceExecutionRole : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
        }
        else {
            resourceInputs["applicationConfiguration"] = undefined /*out*/;
            resourceInputs["applicationDescription"] = undefined /*out*/;
            resourceInputs["applicationMaintenanceConfiguration"] = undefined /*out*/;
            resourceInputs["applicationMode"] = undefined /*out*/;
            resourceInputs["applicationName"] = undefined /*out*/;
            resourceInputs["runConfiguration"] = undefined /*out*/;
            resourceInputs["runtimeEnvironment"] = undefined /*out*/;
            resourceInputs["serviceExecutionRole"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Application.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Application resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Application(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Application.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Application.__pulumiType;
    }
}
exports.Application = Application;
/** @internal */
Application.__pulumiType = 'aws-native:kinesisanalyticsv2:Application';
