// @pulumi/aws-native/iam/oidcprovider.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OIDCProvider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::IAM::OIDCProvider
 */
class OIDCProvider extends pulumi.CustomResource {
    /**
     * Create a OIDCProvider resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.thumbprintList === undefined) && !opts.urn) {
                throw new Error("Missing required property 'thumbprintList'");
            }
            resourceInputs["clientIdList"] = args ? args.clientIdList : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["thumbprintList"] = args ? args.thumbprintList : undefined;
            resourceInputs["url"] = args ? args.url : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["clientIdList"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["thumbprintList"] = undefined /*out*/;
            resourceInputs["url"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(OIDCProvider.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing OIDCProvider resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new OIDCProvider(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of OIDCProvider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OIDCProvider.__pulumiType;
    }
}
exports.OIDCProvider = OIDCProvider;
/** @internal */
OIDCProvider.__pulumiType = 'aws-native:iam:OIDCProvider';
