// @pulumi/aws-native/types/enums/apigateway/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsagePlanKeyKeyType = exports.MethodIntegrationType = exports.MethodIntegrationResponseContentHandling = exports.MethodIntegrationPassthroughBehavior = exports.MethodIntegrationContentHandling = exports.MethodIntegrationConnectionType = exports.MethodAuthorizationType = exports.DocumentationPartLocationType = void 0;
exports.DocumentationPartLocationType = {
    Api: "API",
    Authorizer: "AUTHORIZER",
    Model: "MODEL",
    Resource: "RESOURCE",
    Method: "METHOD",
    PathParameter: "PATH_PARAMETER",
    QueryParameter: "QUERY_PARAMETER",
    RequestHeader: "REQUEST_HEADER",
    RequestBody: "REQUEST_BODY",
    Response: "RESPONSE",
    ResponseHeader: "RESPONSE_HEADER",
    ResponseBody: "RESPONSE_BODY",
};
exports.MethodAuthorizationType = {
    None: "NONE",
    AwsIam: "AWS_IAM",
    Custom: "CUSTOM",
    CognitoUserPools: "COGNITO_USER_POOLS",
};
exports.MethodIntegrationConnectionType = {
    Internet: "INTERNET",
    VpcLink: "VPC_LINK",
};
exports.MethodIntegrationContentHandling = {
    ConvertToBinary: "CONVERT_TO_BINARY",
    ConvertToText: "CONVERT_TO_TEXT",
};
exports.MethodIntegrationPassthroughBehavior = {
    WhenNoMatch: "WHEN_NO_MATCH",
    WhenNoTemplates: "WHEN_NO_TEMPLATES",
    Never: "NEVER",
};
exports.MethodIntegrationResponseContentHandling = {
    ConvertToBinary: "CONVERT_TO_BINARY",
    ConvertToText: "CONVERT_TO_TEXT",
};
exports.MethodIntegrationType = {
    Aws: "AWS",
    AwsProxy: "AWS_PROXY",
    Http: "HTTP",
    HttpProxy: "HTTP_PROXY",
    Mock: "MOCK",
};
exports.UsagePlanKeyKeyType = {
    ApiKey: "API_KEY",
};
