// @pulumi/aws-native/glue/classifier.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Classifier = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Glue::Classifier
 *
 * @deprecated Classifier is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Classifier extends pulumi.CustomResource {
    /**
     * Create a Classifier resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Classifier is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Classifier is deprecated: Classifier is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["csvClassifier"] = args ? args.csvClassifier : undefined;
            resourceInputs["grokClassifier"] = args ? args.grokClassifier : undefined;
            resourceInputs["jsonClassifier"] = args ? args.jsonClassifier : undefined;
            resourceInputs["xMLClassifier"] = args ? args.xMLClassifier : undefined;
        }
        else {
            resourceInputs["csvClassifier"] = undefined /*out*/;
            resourceInputs["grokClassifier"] = undefined /*out*/;
            resourceInputs["jsonClassifier"] = undefined /*out*/;
            resourceInputs["xMLClassifier"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Classifier.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Classifier resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Classifier is deprecated: Classifier is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Classifier(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Classifier.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Classifier.__pulumiType;
    }
}
exports.Classifier = Classifier;
/** @internal */
Classifier.__pulumiType = 'aws-native:glue:Classifier';
