// @pulumi/aws-native/ivs/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamKey = exports.RecordingConfiguration = exports.PlaybackKeyPair = exports.getStreamKeyOutput = exports.getStreamKey = exports.getRecordingConfigurationOutput = exports.getRecordingConfiguration = exports.getPlaybackKeyPairOutput = exports.getPlaybackKeyPair = exports.getChannelOutput = exports.getChannel = exports.Channel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Channel = null;
utilities.lazyLoad(exports, ["Channel"], () => require("./channel"));
exports.getChannel = null;
exports.getChannelOutput = null;
utilities.lazyLoad(exports, ["getChannel", "getChannelOutput"], () => require("./getChannel"));
exports.getPlaybackKeyPair = null;
exports.getPlaybackKeyPairOutput = null;
utilities.lazyLoad(exports, ["getPlaybackKeyPair", "getPlaybackKeyPairOutput"], () => require("./getPlaybackKeyPair"));
exports.getRecordingConfiguration = null;
exports.getRecordingConfigurationOutput = null;
utilities.lazyLoad(exports, ["getRecordingConfiguration", "getRecordingConfigurationOutput"], () => require("./getRecordingConfiguration"));
exports.getStreamKey = null;
exports.getStreamKeyOutput = null;
utilities.lazyLoad(exports, ["getStreamKey", "getStreamKeyOutput"], () => require("./getStreamKey"));
exports.PlaybackKeyPair = null;
utilities.lazyLoad(exports, ["PlaybackKeyPair"], () => require("./playbackKeyPair"));
exports.RecordingConfiguration = null;
utilities.lazyLoad(exports, ["RecordingConfiguration"], () => require("./recordingConfiguration"));
exports.StreamKey = null;
utilities.lazyLoad(exports, ["StreamKey"], () => require("./streamKey"));
// Export enums:
__exportStar(require("../types/enums/ivs"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:ivs:Channel":
                return new exports.Channel(name, undefined, { urn });
            case "aws-native:ivs:PlaybackKeyPair":
                return new exports.PlaybackKeyPair(name, undefined, { urn });
            case "aws-native:ivs:RecordingConfiguration":
                return new exports.RecordingConfiguration(name, undefined, { urn });
            case "aws-native:ivs:StreamKey":
                return new exports.StreamKey(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "ivs", _module);
