// @pulumi/aws-native/types/enums/applicationinsights/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationSubComponentTypeConfigurationSubComponentType = exports.ApplicationLogEncoding = exports.ApplicationGroupingType = exports.ApplicationEventLevel = exports.ApplicationComponentMonitoringSettingComponentConfigurationMode = exports.ApplicationAlarmSeverity = void 0;
exports.ApplicationAlarmSeverity = {
    High: "HIGH",
    Medium: "MEDIUM",
    Low: "LOW",
};
exports.ApplicationComponentMonitoringSettingComponentConfigurationMode = {
    Default: "DEFAULT",
    DefaultWithOverwrite: "DEFAULT_WITH_OVERWRITE",
    Custom: "CUSTOM",
};
exports.ApplicationEventLevel = {
    Information: "INFORMATION",
    Warning: "WARNING",
    Error: "ERROR",
    Critical: "CRITICAL",
    Verbose: "VERBOSE",
};
exports.ApplicationGroupingType = {
    AccountBased: "ACCOUNT_BASED",
};
exports.ApplicationLogEncoding = {
    Utf8: "utf-8",
    Utf16: "utf-16",
    Ascii: "ascii",
};
exports.ApplicationSubComponentTypeConfigurationSubComponentType = {
    AWSEC2Instance: "AWS::EC2::Instance",
    AWSEC2Volume: "AWS::EC2::Volume",
};
