// @pulumi/aws-native/ec2/networkInsightsAccessScopeAnalysis.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkInsightsAccessScopeAnalysis = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::EC2::NetworkInsightsAccessScopeAnalysis
 */
class NetworkInsightsAccessScopeAnalysis extends pulumi.CustomResource {
    /**
     * Create a NetworkInsightsAccessScopeAnalysis resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.networkInsightsAccessScopeId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkInsightsAccessScopeId'");
            }
            resourceInputs["networkInsightsAccessScopeId"] = args ? args.networkInsightsAccessScopeId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["analyzedEniCount"] = undefined /*out*/;
            resourceInputs["endDate"] = undefined /*out*/;
            resourceInputs["findingsFound"] = undefined /*out*/;
            resourceInputs["networkInsightsAccessScopeAnalysisArn"] = undefined /*out*/;
            resourceInputs["networkInsightsAccessScopeAnalysisId"] = undefined /*out*/;
            resourceInputs["startDate"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["statusMessage"] = undefined /*out*/;
        }
        else {
            resourceInputs["analyzedEniCount"] = undefined /*out*/;
            resourceInputs["endDate"] = undefined /*out*/;
            resourceInputs["findingsFound"] = undefined /*out*/;
            resourceInputs["networkInsightsAccessScopeAnalysisArn"] = undefined /*out*/;
            resourceInputs["networkInsightsAccessScopeAnalysisId"] = undefined /*out*/;
            resourceInputs["networkInsightsAccessScopeId"] = undefined /*out*/;
            resourceInputs["startDate"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["statusMessage"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(NetworkInsightsAccessScopeAnalysis.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing NetworkInsightsAccessScopeAnalysis resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkInsightsAccessScopeAnalysis(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkInsightsAccessScopeAnalysis.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkInsightsAccessScopeAnalysis.__pulumiType;
    }
}
exports.NetworkInsightsAccessScopeAnalysis = NetworkInsightsAccessScopeAnalysis;
/** @internal */
NetworkInsightsAccessScopeAnalysis.__pulumiType = 'aws-native:ec2:NetworkInsightsAccessScopeAnalysis';
