// @pulumi/aws-native/ec2/networkInsightsAccessScope.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkInsightsAccessScope = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::EC2::NetworkInsightsAccessScope
 */
class NetworkInsightsAccessScope extends pulumi.CustomResource {
    /**
     * Create a NetworkInsightsAccessScope resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["excludePaths"] = args ? args.excludePaths : undefined;
            resourceInputs["matchPaths"] = args ? args.matchPaths : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["createdDate"] = undefined /*out*/;
            resourceInputs["networkInsightsAccessScopeArn"] = undefined /*out*/;
            resourceInputs["networkInsightsAccessScopeId"] = undefined /*out*/;
            resourceInputs["updatedDate"] = undefined /*out*/;
        }
        else {
            resourceInputs["createdDate"] = undefined /*out*/;
            resourceInputs["excludePaths"] = undefined /*out*/;
            resourceInputs["matchPaths"] = undefined /*out*/;
            resourceInputs["networkInsightsAccessScopeArn"] = undefined /*out*/;
            resourceInputs["networkInsightsAccessScopeId"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["updatedDate"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(NetworkInsightsAccessScope.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing NetworkInsightsAccessScope resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkInsightsAccessScope(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkInsightsAccessScope.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkInsightsAccessScope.__pulumiType;
    }
}
exports.NetworkInsightsAccessScope = NetworkInsightsAccessScope;
/** @internal */
NetworkInsightsAccessScope.__pulumiType = 'aws-native:ec2:NetworkInsightsAccessScope';
