// @pulumi/aws-native/ec2/securityGroupIngress.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityGroupIngress = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::EC2::SecurityGroupIngress
 *
 * @deprecated SecurityGroupIngress is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class SecurityGroupIngress extends pulumi.CustomResource {
    /**
     * Create a SecurityGroupIngress resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated SecurityGroupIngress is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("SecurityGroupIngress is deprecated: SecurityGroupIngress is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.ipProtocol === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ipProtocol'");
            }
            resourceInputs["cidrIp"] = args ? args.cidrIp : undefined;
            resourceInputs["cidrIpv6"] = args ? args.cidrIpv6 : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["fromPort"] = args ? args.fromPort : undefined;
            resourceInputs["groupId"] = args ? args.groupId : undefined;
            resourceInputs["groupName"] = args ? args.groupName : undefined;
            resourceInputs["ipProtocol"] = args ? args.ipProtocol : undefined;
            resourceInputs["sourcePrefixListId"] = args ? args.sourcePrefixListId : undefined;
            resourceInputs["sourceSecurityGroupId"] = args ? args.sourceSecurityGroupId : undefined;
            resourceInputs["sourceSecurityGroupName"] = args ? args.sourceSecurityGroupName : undefined;
            resourceInputs["sourceSecurityGroupOwnerId"] = args ? args.sourceSecurityGroupOwnerId : undefined;
            resourceInputs["toPort"] = args ? args.toPort : undefined;
        }
        else {
            resourceInputs["cidrIp"] = undefined /*out*/;
            resourceInputs["cidrIpv6"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["fromPort"] = undefined /*out*/;
            resourceInputs["groupId"] = undefined /*out*/;
            resourceInputs["groupName"] = undefined /*out*/;
            resourceInputs["ipProtocol"] = undefined /*out*/;
            resourceInputs["sourcePrefixListId"] = undefined /*out*/;
            resourceInputs["sourceSecurityGroupId"] = undefined /*out*/;
            resourceInputs["sourceSecurityGroupName"] = undefined /*out*/;
            resourceInputs["sourceSecurityGroupOwnerId"] = undefined /*out*/;
            resourceInputs["toPort"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SecurityGroupIngress.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SecurityGroupIngress resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SecurityGroupIngress is deprecated: SecurityGroupIngress is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new SecurityGroupIngress(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SecurityGroupIngress.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecurityGroupIngress.__pulumiType;
    }
}
exports.SecurityGroupIngress = SecurityGroupIngress;
/** @internal */
SecurityGroupIngress.__pulumiType = 'aws-native:ec2:SecurityGroupIngress';
