// @pulumi/aws-native/ec2/vpcendpointService.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VPCEndpointService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::EC2::VPCEndpointService
 *
 * @deprecated VPCEndpointService is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class VPCEndpointService extends pulumi.CustomResource {
    /**
     * Create a VPCEndpointService resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated VPCEndpointService is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("VPCEndpointService is deprecated: VPCEndpointService is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["acceptanceRequired"] = args ? args.acceptanceRequired : undefined;
            resourceInputs["contributorInsightsEnabled"] = args ? args.contributorInsightsEnabled : undefined;
            resourceInputs["gatewayLoadBalancerArns"] = args ? args.gatewayLoadBalancerArns : undefined;
            resourceInputs["networkLoadBalancerArns"] = args ? args.networkLoadBalancerArns : undefined;
            resourceInputs["payerResponsibility"] = args ? args.payerResponsibility : undefined;
        }
        else {
            resourceInputs["acceptanceRequired"] = undefined /*out*/;
            resourceInputs["contributorInsightsEnabled"] = undefined /*out*/;
            resourceInputs["gatewayLoadBalancerArns"] = undefined /*out*/;
            resourceInputs["networkLoadBalancerArns"] = undefined /*out*/;
            resourceInputs["payerResponsibility"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VPCEndpointService.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing VPCEndpointService resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("VPCEndpointService is deprecated: VPCEndpointService is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new VPCEndpointService(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VPCEndpointService.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VPCEndpointService.__pulumiType;
    }
}
exports.VPCEndpointService = VPCEndpointService;
/** @internal */
VPCEndpointService.__pulumiType = 'aws-native:ec2:VPCEndpointService';
