// @pulumi/aws-native/macie/customDataIdentifier.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomDataIdentifier = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Macie CustomDataIdentifier resource schema
 */
class CustomDataIdentifier extends pulumi.CustomResource {
    /**
     * Create a CustomDataIdentifier resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.regex === undefined) && !opts.urn) {
                throw new Error("Missing required property 'regex'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["ignoreWords"] = args ? args.ignoreWords : undefined;
            resourceInputs["keywords"] = args ? args.keywords : undefined;
            resourceInputs["maximumMatchDistance"] = args ? args.maximumMatchDistance : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["regex"] = args ? args.regex : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["ignoreWords"] = undefined /*out*/;
            resourceInputs["keywords"] = undefined /*out*/;
            resourceInputs["maximumMatchDistance"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["regex"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CustomDataIdentifier.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing CustomDataIdentifier resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CustomDataIdentifier(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CustomDataIdentifier.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomDataIdentifier.__pulumiType;
    }
}
exports.CustomDataIdentifier = CustomDataIdentifier;
/** @internal */
CustomDataIdentifier.__pulumiType = 'aws-native:macie:CustomDataIdentifier';
