// @pulumi/aws-native/types/enums/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.iotfleetwise = exports.iotevents = exports.iotanalytics = exports.iot = exports.inspectorv2 = exports.imagebuilder = exports.healthlake = exports.groundstation = exports.greengrassv2 = exports.glue = exports.globalaccelerator = exports.gamelift = exports.fsx = exports.frauddetector = exports.forecast = exports.fms = exports.finspace = exports.evidently = exports.events = exports.emr = exports.elasticache = exports.eks = exports.ecs = exports.ecr = exports.ec2 = exports.devopsguru = exports.devicefarm = exports.datasync = exports.databrew = exports.customerprofiles = exports.cur = exports.connect = exports.codestarnotifications = exports.codegurureviewer = exports.codeguruprofiler = exports.cloudtrail = exports.cloudfront = exports.cloudformation = exports.ce = exports.cassandra = exports.budgets = exports.billingconductor = exports.batch = exports.auditmanager = exports.athena = exports.apprunner = exports.applicationinsights = exports.appflow = exports.apigateway = exports.amplify = void 0;
exports.servicecatalogappregistry = exports.servicecatalog = exports.sagemaker = exports.s3outposts = exports.s3 = exports.rum = exports.route53resolver = exports.route53recoverycontrol = exports.route53 = exports.rolesanywhere = exports.robomaker = exports.resourcegroups = exports.resiliencehub = exports.refactorspaces = exports.redshiftserverless = exports.redshift = exports.rds = exports.quicksight = exports.pinpoint = exports.personalize = exports.panorama = exports.nimblestudio = exports.networkfirewall = exports.mwaa = exports.msk = exports.memorydb = exports.mediatailor = exports.mediapackage = exports.mediaconnect = exports.macie = exports.m2 = exports.lookoutmetrics = exports.lookoutequipment = exports.logs = exports.location = exports.lightsail = exports.lex = exports.lambda = exports.lakeformation = exports.kms = exports.kinesisvideo = exports.kinesisfirehose = exports.kinesisanalyticsv2 = exports.kinesis = exports.kendra = exports.kafkaconnect = exports.ivs = exports.iotwireless = exports.iottwinmaker = exports.iotsitewise = void 0;
exports.Region = exports.workspaces = exports.wisdom = exports.wafv2 = exports.transfer = exports.timestream = exports.supportapp = exports.stepfunctions = exports.sso = exports.ssmincidents = exports.ssmcontacts = exports.ssm = exports.signer = void 0;
// Export sub-modules:
const amplify = require("./amplify");
exports.amplify = amplify;
const apigateway = require("./apigateway");
exports.apigateway = apigateway;
const appflow = require("./appflow");
exports.appflow = appflow;
const applicationinsights = require("./applicationinsights");
exports.applicationinsights = applicationinsights;
const apprunner = require("./apprunner");
exports.apprunner = apprunner;
const athena = require("./athena");
exports.athena = athena;
const auditmanager = require("./auditmanager");
exports.auditmanager = auditmanager;
const batch = require("./batch");
exports.batch = batch;
const billingconductor = require("./billingconductor");
exports.billingconductor = billingconductor;
const budgets = require("./budgets");
exports.budgets = budgets;
const cassandra = require("./cassandra");
exports.cassandra = cassandra;
const ce = require("./ce");
exports.ce = ce;
const cloudformation = require("./cloudformation");
exports.cloudformation = cloudformation;
const cloudfront = require("./cloudfront");
exports.cloudfront = cloudfront;
const cloudtrail = require("./cloudtrail");
exports.cloudtrail = cloudtrail;
const codeguruprofiler = require("./codeguruprofiler");
exports.codeguruprofiler = codeguruprofiler;
const codegurureviewer = require("./codegurureviewer");
exports.codegurureviewer = codegurureviewer;
const codestarnotifications = require("./codestarnotifications");
exports.codestarnotifications = codestarnotifications;
const connect = require("./connect");
exports.connect = connect;
const cur = require("./cur");
exports.cur = cur;
const customerprofiles = require("./customerprofiles");
exports.customerprofiles = customerprofiles;
const databrew = require("./databrew");
exports.databrew = databrew;
const datasync = require("./datasync");
exports.datasync = datasync;
const devicefarm = require("./devicefarm");
exports.devicefarm = devicefarm;
const devopsguru = require("./devopsguru");
exports.devopsguru = devopsguru;
const ec2 = require("./ec2");
exports.ec2 = ec2;
const ecr = require("./ecr");
exports.ecr = ecr;
const ecs = require("./ecs");
exports.ecs = ecs;
const eks = require("./eks");
exports.eks = eks;
const elasticache = require("./elasticache");
exports.elasticache = elasticache;
const emr = require("./emr");
exports.emr = emr;
const events = require("./events");
exports.events = events;
const evidently = require("./evidently");
exports.evidently = evidently;
const finspace = require("./finspace");
exports.finspace = finspace;
const fms = require("./fms");
exports.fms = fms;
const forecast = require("./forecast");
exports.forecast = forecast;
const frauddetector = require("./frauddetector");
exports.frauddetector = frauddetector;
const fsx = require("./fsx");
exports.fsx = fsx;
const gamelift = require("./gamelift");
exports.gamelift = gamelift;
const globalaccelerator = require("./globalaccelerator");
exports.globalaccelerator = globalaccelerator;
const glue = require("./glue");
exports.glue = glue;
const greengrassv2 = require("./greengrassv2");
exports.greengrassv2 = greengrassv2;
const groundstation = require("./groundstation");
exports.groundstation = groundstation;
const healthlake = require("./healthlake");
exports.healthlake = healthlake;
const imagebuilder = require("./imagebuilder");
exports.imagebuilder = imagebuilder;
const inspectorv2 = require("./inspectorv2");
exports.inspectorv2 = inspectorv2;
const iot = require("./iot");
exports.iot = iot;
const iotanalytics = require("./iotanalytics");
exports.iotanalytics = iotanalytics;
const iotevents = require("./iotevents");
exports.iotevents = iotevents;
const iotfleetwise = require("./iotfleetwise");
exports.iotfleetwise = iotfleetwise;
const iotsitewise = require("./iotsitewise");
exports.iotsitewise = iotsitewise;
const iottwinmaker = require("./iottwinmaker");
exports.iottwinmaker = iottwinmaker;
const iotwireless = require("./iotwireless");
exports.iotwireless = iotwireless;
const ivs = require("./ivs");
exports.ivs = ivs;
const kafkaconnect = require("./kafkaconnect");
exports.kafkaconnect = kafkaconnect;
const kendra = require("./kendra");
exports.kendra = kendra;
const kinesis = require("./kinesis");
exports.kinesis = kinesis;
const kinesisanalyticsv2 = require("./kinesisanalyticsv2");
exports.kinesisanalyticsv2 = kinesisanalyticsv2;
const kinesisfirehose = require("./kinesisfirehose");
exports.kinesisfirehose = kinesisfirehose;
const kinesisvideo = require("./kinesisvideo");
exports.kinesisvideo = kinesisvideo;
const kms = require("./kms");
exports.kms = kms;
const lakeformation = require("./lakeformation");
exports.lakeformation = lakeformation;
const lambda = require("./lambda");
exports.lambda = lambda;
const lex = require("./lex");
exports.lex = lex;
const lightsail = require("./lightsail");
exports.lightsail = lightsail;
const location = require("./location");
exports.location = location;
const logs = require("./logs");
exports.logs = logs;
const lookoutequipment = require("./lookoutequipment");
exports.lookoutequipment = lookoutequipment;
const lookoutmetrics = require("./lookoutmetrics");
exports.lookoutmetrics = lookoutmetrics;
const m2 = require("./m2");
exports.m2 = m2;
const macie = require("./macie");
exports.macie = macie;
const mediaconnect = require("./mediaconnect");
exports.mediaconnect = mediaconnect;
const mediapackage = require("./mediapackage");
exports.mediapackage = mediapackage;
const mediatailor = require("./mediatailor");
exports.mediatailor = mediatailor;
const memorydb = require("./memorydb");
exports.memorydb = memorydb;
const msk = require("./msk");
exports.msk = msk;
const mwaa = require("./mwaa");
exports.mwaa = mwaa;
const networkfirewall = require("./networkfirewall");
exports.networkfirewall = networkfirewall;
const nimblestudio = require("./nimblestudio");
exports.nimblestudio = nimblestudio;
const panorama = require("./panorama");
exports.panorama = panorama;
const personalize = require("./personalize");
exports.personalize = personalize;
const pinpoint = require("./pinpoint");
exports.pinpoint = pinpoint;
const quicksight = require("./quicksight");
exports.quicksight = quicksight;
const rds = require("./rds");
exports.rds = rds;
const redshift = require("./redshift");
exports.redshift = redshift;
const redshiftserverless = require("./redshiftserverless");
exports.redshiftserverless = redshiftserverless;
const refactorspaces = require("./refactorspaces");
exports.refactorspaces = refactorspaces;
const resiliencehub = require("./resiliencehub");
exports.resiliencehub = resiliencehub;
const resourcegroups = require("./resourcegroups");
exports.resourcegroups = resourcegroups;
const robomaker = require("./robomaker");
exports.robomaker = robomaker;
const rolesanywhere = require("./rolesanywhere");
exports.rolesanywhere = rolesanywhere;
const route53 = require("./route53");
exports.route53 = route53;
const route53recoverycontrol = require("./route53recoverycontrol");
exports.route53recoverycontrol = route53recoverycontrol;
const route53resolver = require("./route53resolver");
exports.route53resolver = route53resolver;
const rum = require("./rum");
exports.rum = rum;
const s3 = require("./s3");
exports.s3 = s3;
const s3outposts = require("./s3outposts");
exports.s3outposts = s3outposts;
const sagemaker = require("./sagemaker");
exports.sagemaker = sagemaker;
const servicecatalog = require("./servicecatalog");
exports.servicecatalog = servicecatalog;
const servicecatalogappregistry = require("./servicecatalogappregistry");
exports.servicecatalogappregistry = servicecatalogappregistry;
const signer = require("./signer");
exports.signer = signer;
const ssm = require("./ssm");
exports.ssm = ssm;
const ssmcontacts = require("./ssmcontacts");
exports.ssmcontacts = ssmcontacts;
const ssmincidents = require("./ssmincidents");
exports.ssmincidents = ssmincidents;
const sso = require("./sso");
exports.sso = sso;
const stepfunctions = require("./stepfunctions");
exports.stepfunctions = stepfunctions;
const supportapp = require("./supportapp");
exports.supportapp = supportapp;
const timestream = require("./timestream");
exports.timestream = timestream;
const transfer = require("./transfer");
exports.transfer = transfer;
const wafv2 = require("./wafv2");
exports.wafv2 = wafv2;
const wisdom = require("./wisdom");
exports.wisdom = wisdom;
const workspaces = require("./workspaces");
exports.workspaces = workspaces;
exports.Region = {
    /**
     * Africa (Cape Town)
     */
    AFSouth1: "af-south-1",
    /**
     * Asia Pacific (Hong Kong)
     */
    APEast1: "ap-east-1",
    /**
     * Asia Pacific (Tokyo)
     */
    APNortheast1: "ap-northeast-1",
    /**
     * Asia Pacific (Seoul)
     */
    APNortheast2: "ap-northeast-2",
    /**
     * Asia Pacific (Osaka)
     */
    APNortheast3: "ap-northeast-3",
    /**
     * Asia Pacific (Mumbai)
     */
    APSouth1: "ap-south-1",
    /**
     * Asia Pacific (Singapore)
     */
    APSoutheast1: "ap-southeast-1",
    /**
     * Asia Pacific (Sydney)
     */
    APSoutheast2: "ap-southeast-2",
    /**
     * Canada (Central)
     */
    CACentral: "ca-central-1",
    /**
     * China (Beijing)
     */
    CNNorth1: "cn-north-1",
    /**
     * China (Ningxia)
     */
    CNNorthwest1: "cn-northwest-1",
    /**
     * Europe (Frankfurt)
     */
    EUCentral1: "eu-central-1",
    /**
     * Europe (Stockholm)
     */
    EUNorth1: "eu-north-1",
    /**
     * Europe (Ireland)
     */
    EUWest1: "eu-west-1",
    /**
     * Europe (London)
     */
    EUWest2: "eu-west-2",
    /**
     * Europe (Paris)
     */
    EUWest3: "eu-west-3",
    /**
     * Europe (Milan)
     */
    EUSouth1: "eu-south-1",
    /**
     * Middle East (Bahrain)
     */
    MESouth1: "me-south-1",
    /**
     * South America (São Paulo)
     */
    SAEast1: "sa-east-1",
    /**
     * AWS GovCloud (US-East)
     */
    USGovEast1: "us-gov-east-1",
    /**
     * AWS GovCloud (US-West)
     */
    USGovWest1: "us-gov-west-1",
    /**
     * US East (N. Virginia)
     */
    USEast1: "us-east-1",
    /**
     * US East (Ohio)
     */
    USEast2: "us-east-2",
    /**
     * US West (N. California)
     */
    USWest1: "us-west-1",
    /**
     * US West (Oregon)
     */
    USWest2: "us-west-2",
};
