// @pulumi/aws-native/sns/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopicPolicy = exports.Topic = exports.Subscription = exports.getTopicPolicyOutput = exports.getTopicPolicy = exports.getTopicOutput = exports.getTopic = exports.getSubscriptionOutput = exports.getSubscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getSubscription = null;
exports.getSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getSubscription", "getSubscriptionOutput"], () => require("./getSubscription"));
exports.getTopic = null;
exports.getTopicOutput = null;
utilities.lazyLoad(exports, ["getTopic", "getTopicOutput"], () => require("./getTopic"));
exports.getTopicPolicy = null;
exports.getTopicPolicyOutput = null;
utilities.lazyLoad(exports, ["getTopicPolicy", "getTopicPolicyOutput"], () => require("./getTopicPolicy"));
exports.Subscription = null;
utilities.lazyLoad(exports, ["Subscription"], () => require("./subscription"));
exports.Topic = null;
utilities.lazyLoad(exports, ["Topic"], () => require("./topic"));
exports.TopicPolicy = null;
utilities.lazyLoad(exports, ["TopicPolicy"], () => require("./topicPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:sns:Subscription":
                return new exports.Subscription(name, undefined, { urn });
            case "aws-native:sns:Topic":
                return new exports.Topic(name, undefined, { urn });
            case "aws-native:sns:TopicPolicy":
                return new exports.TopicPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "sns", _module);
