// @pulumi/aws-native/appconfig/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostedConfigurationVersion = exports.getHostedConfigurationVersionOutput = exports.getHostedConfigurationVersion = exports.getEnvironmentOutput = exports.getEnvironment = exports.getDeploymentStrategyOutput = exports.getDeploymentStrategy = exports.getDeploymentOutput = exports.getDeployment = exports.getConfigurationProfileOutput = exports.getConfigurationProfile = exports.getApplicationOutput = exports.getApplication = exports.Environment = exports.DeploymentStrategy = exports.Deployment = exports.ConfigurationProfile = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.ConfigurationProfile = null;
utilities.lazyLoad(exports, ["ConfigurationProfile"], () => require("./configurationProfile"));
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.DeploymentStrategy = null;
utilities.lazyLoad(exports, ["DeploymentStrategy"], () => require("./deploymentStrategy"));
exports.Environment = null;
utilities.lazyLoad(exports, ["Environment"], () => require("./environment"));
exports.getApplication = null;
exports.getApplicationOutput = null;
utilities.lazyLoad(exports, ["getApplication", "getApplicationOutput"], () => require("./getApplication"));
exports.getConfigurationProfile = null;
exports.getConfigurationProfileOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfile", "getConfigurationProfileOutput"], () => require("./getConfigurationProfile"));
exports.getDeployment = null;
exports.getDeploymentOutput = null;
utilities.lazyLoad(exports, ["getDeployment", "getDeploymentOutput"], () => require("./getDeployment"));
exports.getDeploymentStrategy = null;
exports.getDeploymentStrategyOutput = null;
utilities.lazyLoad(exports, ["getDeploymentStrategy", "getDeploymentStrategyOutput"], () => require("./getDeploymentStrategy"));
exports.getEnvironment = null;
exports.getEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getEnvironment", "getEnvironmentOutput"], () => require("./getEnvironment"));
exports.getHostedConfigurationVersion = null;
exports.getHostedConfigurationVersionOutput = null;
utilities.lazyLoad(exports, ["getHostedConfigurationVersion", "getHostedConfigurationVersionOutput"], () => require("./getHostedConfigurationVersion"));
exports.HostedConfigurationVersion = null;
utilities.lazyLoad(exports, ["HostedConfigurationVersion"], () => require("./hostedConfigurationVersion"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:appconfig:Application":
                return new exports.Application(name, undefined, { urn });
            case "aws-native:appconfig:ConfigurationProfile":
                return new exports.ConfigurationProfile(name, undefined, { urn });
            case "aws-native:appconfig:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            case "aws-native:appconfig:DeploymentStrategy":
                return new exports.DeploymentStrategy(name, undefined, { urn });
            case "aws-native:appconfig:Environment":
                return new exports.Environment(name, undefined, { urn });
            case "aws-native:appconfig:HostedConfigurationVersion":
                return new exports.HostedConfigurationVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "appconfig", _module);
