// @pulumi/aws-native/connect/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserHierarchyGroup = exports.User = exports.TaskTemplate = exports.QuickConnect = exports.PhoneNumber = exports.InstanceStorageConfig = exports.Instance = exports.HoursOfOperation = exports.getUserHierarchyGroupOutput = exports.getUserHierarchyGroup = exports.getUserOutput = exports.getUser = exports.getTaskTemplateOutput = exports.getTaskTemplate = exports.getQuickConnectOutput = exports.getQuickConnect = exports.getPhoneNumberOutput = exports.getPhoneNumber = exports.getInstanceStorageConfigOutput = exports.getInstanceStorageConfig = exports.getInstanceOutput = exports.getInstance = exports.getHoursOfOperationOutput = exports.getHoursOfOperation = exports.getContactFlowModuleOutput = exports.getContactFlowModule = exports.getContactFlowOutput = exports.getContactFlow = exports.ContactFlowModule = exports.ContactFlow = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ContactFlow = null;
utilities.lazyLoad(exports, ["ContactFlow"], () => require("./contactFlow"));
exports.ContactFlowModule = null;
utilities.lazyLoad(exports, ["ContactFlowModule"], () => require("./contactFlowModule"));
exports.getContactFlow = null;
exports.getContactFlowOutput = null;
utilities.lazyLoad(exports, ["getContactFlow", "getContactFlowOutput"], () => require("./getContactFlow"));
exports.getContactFlowModule = null;
exports.getContactFlowModuleOutput = null;
utilities.lazyLoad(exports, ["getContactFlowModule", "getContactFlowModuleOutput"], () => require("./getContactFlowModule"));
exports.getHoursOfOperation = null;
exports.getHoursOfOperationOutput = null;
utilities.lazyLoad(exports, ["getHoursOfOperation", "getHoursOfOperationOutput"], () => require("./getHoursOfOperation"));
exports.getInstance = null;
exports.getInstanceOutput = null;
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
exports.getInstanceStorageConfig = null;
exports.getInstanceStorageConfigOutput = null;
utilities.lazyLoad(exports, ["getInstanceStorageConfig", "getInstanceStorageConfigOutput"], () => require("./getInstanceStorageConfig"));
exports.getPhoneNumber = null;
exports.getPhoneNumberOutput = null;
utilities.lazyLoad(exports, ["getPhoneNumber", "getPhoneNumberOutput"], () => require("./getPhoneNumber"));
exports.getQuickConnect = null;
exports.getQuickConnectOutput = null;
utilities.lazyLoad(exports, ["getQuickConnect", "getQuickConnectOutput"], () => require("./getQuickConnect"));
exports.getTaskTemplate = null;
exports.getTaskTemplateOutput = null;
utilities.lazyLoad(exports, ["getTaskTemplate", "getTaskTemplateOutput"], () => require("./getTaskTemplate"));
exports.getUser = null;
exports.getUserOutput = null;
utilities.lazyLoad(exports, ["getUser", "getUserOutput"], () => require("./getUser"));
exports.getUserHierarchyGroup = null;
exports.getUserHierarchyGroupOutput = null;
utilities.lazyLoad(exports, ["getUserHierarchyGroup", "getUserHierarchyGroupOutput"], () => require("./getUserHierarchyGroup"));
exports.HoursOfOperation = null;
utilities.lazyLoad(exports, ["HoursOfOperation"], () => require("./hoursOfOperation"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.InstanceStorageConfig = null;
utilities.lazyLoad(exports, ["InstanceStorageConfig"], () => require("./instanceStorageConfig"));
exports.PhoneNumber = null;
utilities.lazyLoad(exports, ["PhoneNumber"], () => require("./phoneNumber"));
exports.QuickConnect = null;
utilities.lazyLoad(exports, ["QuickConnect"], () => require("./quickConnect"));
exports.TaskTemplate = null;
utilities.lazyLoad(exports, ["TaskTemplate"], () => require("./taskTemplate"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
exports.UserHierarchyGroup = null;
utilities.lazyLoad(exports, ["UserHierarchyGroup"], () => require("./userHierarchyGroup"));
// Export enums:
__exportStar(require("../types/enums/connect"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:connect:ContactFlow":
                return new exports.ContactFlow(name, undefined, { urn });
            case "aws-native:connect:ContactFlowModule":
                return new exports.ContactFlowModule(name, undefined, { urn });
            case "aws-native:connect:HoursOfOperation":
                return new exports.HoursOfOperation(name, undefined, { urn });
            case "aws-native:connect:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "aws-native:connect:InstanceStorageConfig":
                return new exports.InstanceStorageConfig(name, undefined, { urn });
            case "aws-native:connect:PhoneNumber":
                return new exports.PhoneNumber(name, undefined, { urn });
            case "aws-native:connect:QuickConnect":
                return new exports.QuickConnect(name, undefined, { urn });
            case "aws-native:connect:TaskTemplate":
                return new exports.TaskTemplate(name, undefined, { urn });
            case "aws-native:connect:User":
                return new exports.User(name, undefined, { urn });
            case "aws-native:connect:UserHierarchyGroup":
                return new exports.UserHierarchyGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "connect", _module);
