// @pulumi/aws-native/connect/instanceStorageConfig.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstanceStorageConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Connect::InstanceStorageConfig
 */
class InstanceStorageConfig extends pulumi.CustomResource {
    /**
     * Create a InstanceStorageConfig resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.instanceArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceArn'");
            }
            if ((!args || args.resourceType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceType'");
            }
            if ((!args || args.storageType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageType'");
            }
            resourceInputs["instanceArn"] = args ? args.instanceArn : undefined;
            resourceInputs["kinesisFirehoseConfig"] = args ? args.kinesisFirehoseConfig : undefined;
            resourceInputs["kinesisStreamConfig"] = args ? args.kinesisStreamConfig : undefined;
            resourceInputs["kinesisVideoStreamConfig"] = args ? args.kinesisVideoStreamConfig : undefined;
            resourceInputs["resourceType"] = args ? args.resourceType : undefined;
            resourceInputs["s3Config"] = args ? args.s3Config : undefined;
            resourceInputs["storageType"] = args ? args.storageType : undefined;
            resourceInputs["associationId"] = undefined /*out*/;
        }
        else {
            resourceInputs["associationId"] = undefined /*out*/;
            resourceInputs["instanceArn"] = undefined /*out*/;
            resourceInputs["kinesisFirehoseConfig"] = undefined /*out*/;
            resourceInputs["kinesisStreamConfig"] = undefined /*out*/;
            resourceInputs["kinesisVideoStreamConfig"] = undefined /*out*/;
            resourceInputs["resourceType"] = undefined /*out*/;
            resourceInputs["s3Config"] = undefined /*out*/;
            resourceInputs["storageType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(InstanceStorageConfig.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing InstanceStorageConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new InstanceStorageConfig(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of InstanceStorageConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === InstanceStorageConfig.__pulumiType;
    }
}
exports.InstanceStorageConfig = InstanceStorageConfig;
/** @internal */
InstanceStorageConfig.__pulumiType = 'aws-native:connect:InstanceStorageConfig';
