// @pulumi/aws-native/types/enums/networkfirewall/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleGroupTypeEnum = exports.RuleGroupTargetType = exports.RuleGroupTCPFlag = exports.RuleGroupStatefulRuleAction = exports.RuleGroupRuleOrder = exports.RuleGroupHeaderProtocol = exports.RuleGroupHeaderDirection = exports.RuleGroupGeneratedRulesType = exports.LoggingConfigurationLogDestinationConfigLogType = exports.LoggingConfigurationLogDestinationConfigLogDestinationType = exports.FirewallPolicyRuleOrder = exports.FirewallPolicyOverrideAction = void 0;
exports.FirewallPolicyOverrideAction = {
    DropToAlert: "DROP_TO_ALERT",
};
exports.FirewallPolicyRuleOrder = {
    DefaultActionOrder: "DEFAULT_ACTION_ORDER",
    StrictOrder: "STRICT_ORDER",
};
exports.LoggingConfigurationLogDestinationConfigLogDestinationType = {
    S3: "S3",
    CloudWatchLogs: "CloudWatchLogs",
    KinesisDataFirehose: "KinesisDataFirehose",
};
exports.LoggingConfigurationLogDestinationConfigLogType = {
    Alert: "ALERT",
    Flow: "FLOW",
};
exports.RuleGroupGeneratedRulesType = {
    Allowlist: "ALLOWLIST",
    Denylist: "DENYLIST",
};
exports.RuleGroupHeaderDirection = {
    Forward: "FORWARD",
    Any: "ANY",
};
exports.RuleGroupHeaderProtocol = {
    Ip: "IP",
    Tcp: "TCP",
    Udp: "UDP",
    Icmp: "ICMP",
    Http: "HTTP",
    Ftp: "FTP",
    Tls: "TLS",
    Smb: "SMB",
    Dns: "DNS",
    Dcerpc: "DCERPC",
    Ssh: "SSH",
    Smtp: "SMTP",
    Imap: "IMAP",
    Msn: "MSN",
    Krb5: "KRB5",
    Ikev2: "IKEV2",
    Tftp: "TFTP",
    Ntp: "NTP",
    Dhcp: "DHCP",
};
exports.RuleGroupRuleOrder = {
    DefaultActionOrder: "DEFAULT_ACTION_ORDER",
    StrictOrder: "STRICT_ORDER",
};
exports.RuleGroupStatefulRuleAction = {
    Pass: "PASS",
    Drop: "DROP",
    Alert: "ALERT",
};
exports.RuleGroupTCPFlag = {
    Fin: "FIN",
    Syn: "SYN",
    Rst: "RST",
    Psh: "PSH",
    Ack: "ACK",
    Urg: "URG",
    Ece: "ECE",
    Cwr: "CWR",
};
exports.RuleGroupTargetType = {
    TlsSni: "TLS_SNI",
    HttpHost: "HTTP_HOST",
};
exports.RuleGroupTypeEnum = {
    Stateless: "STATELESS",
    Stateful: "STATEFUL",
};
