// @pulumi/aws-native/types/enums/gamelift/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GameServerGroupGameServerProtectionPolicy = exports.GameServerGroupDeleteOption = exports.GameServerGroupBalancingStrategy = exports.FleetType = exports.FleetNewGameSessionProtectionPolicy = exports.FleetIpPermissionProtocol = exports.FleetCertificateConfigurationCertificateType = exports.AliasRoutingStrategyType = void 0;
exports.AliasRoutingStrategyType = {
    Simple: "SIMPLE",
    Terminal: "TERMINAL",
};
exports.FleetCertificateConfigurationCertificateType = {
    Disabled: "DISABLED",
    Generated: "GENERATED",
};
exports.FleetIpPermissionProtocol = {
    Tcp: "TCP",
    Udp: "UDP",
};
exports.FleetNewGameSessionProtectionPolicy = {
    FullProtection: "FullProtection",
    NoProtection: "NoProtection",
};
exports.FleetType = {
    OnDemand: "ON_DEMAND",
    Spot: "SPOT",
};
exports.GameServerGroupBalancingStrategy = {
    SpotOnly: "SPOT_ONLY",
    SpotPreferred: "SPOT_PREFERRED",
    OnDemandOnly: "ON_DEMAND_ONLY",
};
exports.GameServerGroupDeleteOption = {
    SafeDelete: "SAFE_DELETE",
    ForceDelete: "FORCE_DELETE",
    Retain: "RETAIN",
};
exports.GameServerGroupGameServerProtectionPolicy = {
    NoProtection: "NO_PROTECTION",
    FullProtection: "FULL_PROTECTION",
};
