// @pulumi/aws-native/lex/bot.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Bot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Amazon Lex conversational bot performing automated tasks such as ordering a pizza, booking a hotel, and so on.
 */
class Bot extends pulumi.CustomResource {
    /**
     * Create a Bot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dataPrivacy === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataPrivacy'");
            }
            if ((!args || args.idleSessionTTLInSeconds === undefined) && !opts.urn) {
                throw new Error("Missing required property 'idleSessionTTLInSeconds'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["autoBuildBotLocales"] = args ? args.autoBuildBotLocales : undefined;
            resourceInputs["botFileS3Location"] = args ? args.botFileS3Location : undefined;
            resourceInputs["botLocales"] = args ? args.botLocales : undefined;
            resourceInputs["botTags"] = args ? args.botTags : undefined;
            resourceInputs["dataPrivacy"] = args ? args.dataPrivacy : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["idleSessionTTLInSeconds"] = args ? args.idleSessionTTLInSeconds : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
            resourceInputs["testBotAliasSettings"] = args ? args.testBotAliasSettings : undefined;
            resourceInputs["testBotAliasTags"] = args ? args.testBotAliasTags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["autoBuildBotLocales"] = undefined /*out*/;
            resourceInputs["botFileS3Location"] = undefined /*out*/;
            resourceInputs["botLocales"] = undefined /*out*/;
            resourceInputs["botTags"] = undefined /*out*/;
            resourceInputs["dataPrivacy"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["idleSessionTTLInSeconds"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["roleArn"] = undefined /*out*/;
            resourceInputs["testBotAliasSettings"] = undefined /*out*/;
            resourceInputs["testBotAliasTags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Bot.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Bot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Bot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Bot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Bot.__pulumiType;
    }
}
exports.Bot = Bot;
/** @internal */
Bot.__pulumiType = 'aws-native:lex:Bot';
