// @pulumi/aws-native/lex/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourcePolicy = exports.getResourcePolicyOutput = exports.getResourcePolicy = exports.getBotVersionOutput = exports.getBotVersion = exports.getBotAliasOutput = exports.getBotAlias = exports.getBotOutput = exports.getBot = exports.BotVersion = exports.BotAlias = exports.Bot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Bot = null;
utilities.lazyLoad(exports, ["Bot"], () => require("./bot"));
exports.BotAlias = null;
utilities.lazyLoad(exports, ["BotAlias"], () => require("./botAlias"));
exports.BotVersion = null;
utilities.lazyLoad(exports, ["BotVersion"], () => require("./botVersion"));
exports.getBot = null;
exports.getBotOutput = null;
utilities.lazyLoad(exports, ["getBot", "getBotOutput"], () => require("./getBot"));
exports.getBotAlias = null;
exports.getBotAliasOutput = null;
utilities.lazyLoad(exports, ["getBotAlias", "getBotAliasOutput"], () => require("./getBotAlias"));
exports.getBotVersion = null;
exports.getBotVersionOutput = null;
utilities.lazyLoad(exports, ["getBotVersion", "getBotVersionOutput"], () => require("./getBotVersion"));
exports.getResourcePolicy = null;
exports.getResourcePolicyOutput = null;
utilities.lazyLoad(exports, ["getResourcePolicy", "getResourcePolicyOutput"], () => require("./getResourcePolicy"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
// Export enums:
__exportStar(require("../types/enums/lex"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:lex:Bot":
                return new exports.Bot(name, undefined, { urn });
            case "aws-native:lex:BotAlias":
                return new exports.BotAlias(name, undefined, { urn });
            case "aws-native:lex:BotVersion":
                return new exports.BotVersion(name, undefined, { urn });
            case "aws-native:lex:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "lex", _module);
