// @pulumi/aws-native/iotsitewise/assetModel.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetModel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::IoTSiteWise::AssetModel
 */
class AssetModel extends pulumi.CustomResource {
    /**
     * Create a AssetModel resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["assetModelCompositeModels"] = args ? args.assetModelCompositeModels : undefined;
            resourceInputs["assetModelDescription"] = args ? args.assetModelDescription : undefined;
            resourceInputs["assetModelHierarchies"] = args ? args.assetModelHierarchies : undefined;
            resourceInputs["assetModelName"] = args ? args.assetModelName : undefined;
            resourceInputs["assetModelProperties"] = args ? args.assetModelProperties : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["assetModelArn"] = undefined /*out*/;
            resourceInputs["assetModelId"] = undefined /*out*/;
        }
        else {
            resourceInputs["assetModelArn"] = undefined /*out*/;
            resourceInputs["assetModelCompositeModels"] = undefined /*out*/;
            resourceInputs["assetModelDescription"] = undefined /*out*/;
            resourceInputs["assetModelHierarchies"] = undefined /*out*/;
            resourceInputs["assetModelId"] = undefined /*out*/;
            resourceInputs["assetModelName"] = undefined /*out*/;
            resourceInputs["assetModelProperties"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AssetModel.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing AssetModel resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AssetModel(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AssetModel.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AssetModel.__pulumiType;
    }
}
exports.AssetModel = AssetModel;
/** @internal */
AssetModel.__pulumiType = 'aws-native:iotsitewise:AssetModel';
