// @pulumi/aws-native/iotsitewise/asset.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Asset = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::IoTSiteWise::Asset
 */
class Asset extends pulumi.CustomResource {
    /**
     * Create a Asset resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.assetModelId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'assetModelId'");
            }
            resourceInputs["assetDescription"] = args ? args.assetDescription : undefined;
            resourceInputs["assetHierarchies"] = args ? args.assetHierarchies : undefined;
            resourceInputs["assetModelId"] = args ? args.assetModelId : undefined;
            resourceInputs["assetName"] = args ? args.assetName : undefined;
            resourceInputs["assetProperties"] = args ? args.assetProperties : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["assetArn"] = undefined /*out*/;
            resourceInputs["assetId"] = undefined /*out*/;
        }
        else {
            resourceInputs["assetArn"] = undefined /*out*/;
            resourceInputs["assetDescription"] = undefined /*out*/;
            resourceInputs["assetHierarchies"] = undefined /*out*/;
            resourceInputs["assetId"] = undefined /*out*/;
            resourceInputs["assetModelId"] = undefined /*out*/;
            resourceInputs["assetName"] = undefined /*out*/;
            resourceInputs["assetProperties"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Asset.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Asset resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Asset(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Asset.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Asset.__pulumiType;
    }
}
exports.Asset = Asset;
/** @internal */
Asset.__pulumiType = 'aws-native:iotsitewise:Asset';
