// @pulumi/aws-native/iotsitewise/dashboard.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Dashboard = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::IoTSiteWise::Dashboard
 */
class Dashboard extends pulumi.CustomResource {
    /**
     * Create a Dashboard resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dashboardDefinition === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dashboardDefinition'");
            }
            if ((!args || args.dashboardDescription === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dashboardDescription'");
            }
            resourceInputs["dashboardDefinition"] = args ? args.dashboardDefinition : undefined;
            resourceInputs["dashboardDescription"] = args ? args.dashboardDescription : undefined;
            resourceInputs["dashboardName"] = args ? args.dashboardName : undefined;
            resourceInputs["projectId"] = args ? args.projectId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["dashboardArn"] = undefined /*out*/;
            resourceInputs["dashboardId"] = undefined /*out*/;
        }
        else {
            resourceInputs["dashboardArn"] = undefined /*out*/;
            resourceInputs["dashboardDefinition"] = undefined /*out*/;
            resourceInputs["dashboardDescription"] = undefined /*out*/;
            resourceInputs["dashboardId"] = undefined /*out*/;
            resourceInputs["dashboardName"] = undefined /*out*/;
            resourceInputs["projectId"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Dashboard.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Dashboard resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Dashboard(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Dashboard.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Dashboard.__pulumiType;
    }
}
exports.Dashboard = Dashboard;
/** @internal */
Dashboard.__pulumiType = 'aws-native:iotsitewise:Dashboard';
