// @pulumi/aws-native/billingconductor/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricingRule = exports.PricingPlan = exports.getPricingRuleOutput = exports.getPricingRule = exports.getPricingPlanOutput = exports.getPricingPlan = exports.getCustomLineItemOutput = exports.getCustomLineItem = exports.getBillingGroupOutput = exports.getBillingGroup = exports.CustomLineItem = exports.BillingGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.BillingGroup = null;
utilities.lazyLoad(exports, ["BillingGroup"], () => require("./billingGroup"));
exports.CustomLineItem = null;
utilities.lazyLoad(exports, ["CustomLineItem"], () => require("./customLineItem"));
exports.getBillingGroup = null;
exports.getBillingGroupOutput = null;
utilities.lazyLoad(exports, ["getBillingGroup", "getBillingGroupOutput"], () => require("./getBillingGroup"));
exports.getCustomLineItem = null;
exports.getCustomLineItemOutput = null;
utilities.lazyLoad(exports, ["getCustomLineItem", "getCustomLineItemOutput"], () => require("./getCustomLineItem"));
exports.getPricingPlan = null;
exports.getPricingPlanOutput = null;
utilities.lazyLoad(exports, ["getPricingPlan", "getPricingPlanOutput"], () => require("./getPricingPlan"));
exports.getPricingRule = null;
exports.getPricingRuleOutput = null;
utilities.lazyLoad(exports, ["getPricingRule", "getPricingRuleOutput"], () => require("./getPricingRule"));
exports.PricingPlan = null;
utilities.lazyLoad(exports, ["PricingPlan"], () => require("./pricingPlan"));
exports.PricingRule = null;
utilities.lazyLoad(exports, ["PricingRule"], () => require("./pricingRule"));
// Export enums:
__exportStar(require("../types/enums/billingconductor"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:billingconductor:BillingGroup":
                return new exports.BillingGroup(name, undefined, { urn });
            case "aws-native:billingconductor:CustomLineItem":
                return new exports.CustomLineItem(name, undefined, { urn });
            case "aws-native:billingconductor:PricingPlan":
                return new exports.PricingPlan(name, undefined, { urn });
            case "aws-native:billingconductor:PricingRule":
                return new exports.PricingRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "billingconductor", _module);
