// @pulumi/aws-native/mediapackage/packagingConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackagingConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::MediaPackage::PackagingConfiguration
 */
class PackagingConfiguration extends pulumi.CustomResource {
    /**
     * Create a PackagingConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.packagingGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'packagingGroupId'");
            }
            resourceInputs["cmafPackage"] = args ? args.cmafPackage : undefined;
            resourceInputs["dashPackage"] = args ? args.dashPackage : undefined;
            resourceInputs["hlsPackage"] = args ? args.hlsPackage : undefined;
            resourceInputs["mssPackage"] = args ? args.mssPackage : undefined;
            resourceInputs["packagingGroupId"] = args ? args.packagingGroupId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["cmafPackage"] = undefined /*out*/;
            resourceInputs["dashPackage"] = undefined /*out*/;
            resourceInputs["hlsPackage"] = undefined /*out*/;
            resourceInputs["mssPackage"] = undefined /*out*/;
            resourceInputs["packagingGroupId"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PackagingConfiguration.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing PackagingConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PackagingConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PackagingConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PackagingConfiguration.__pulumiType;
    }
}
exports.PackagingConfiguration = PackagingConfiguration;
/** @internal */
PackagingConfiguration.__pulumiType = 'aws-native:mediapackage:PackagingConfiguration';
