// @pulumi/aws-native/managedblockchain/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Node = exports.Member = exports.getNodeOutput = exports.getNode = exports.getMemberOutput = exports.getMember = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getMember = null;
exports.getMemberOutput = null;
utilities.lazyLoad(exports, ["getMember", "getMemberOutput"], () => require("./getMember"));
exports.getNode = null;
exports.getNodeOutput = null;
utilities.lazyLoad(exports, ["getNode", "getNodeOutput"], () => require("./getNode"));
exports.Member = null;
utilities.lazyLoad(exports, ["Member"], () => require("./member"));
exports.Node = null;
utilities.lazyLoad(exports, ["Node"], () => require("./node"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:managedblockchain:Member":
                return new exports.Member(name, undefined, { urn });
            case "aws-native:managedblockchain:Node":
                return new exports.Node(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "managedblockchain", _module);
