// @pulumi/aws-native/types/enums/mediaconnect/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowSourceProtocol = exports.FlowSourceEncryptionKeyType = exports.FlowSourceEncryptionAlgorithm = exports.FlowOutputProtocol = exports.FlowOutputEncryptionKeyType = exports.FlowOutputEncryptionAlgorithm = exports.FlowFailoverConfigState = exports.FlowEntitlementEntitlementStatus = exports.FlowEntitlementEncryptionKeyType = exports.FlowEntitlementEncryptionAlgorithm = exports.FlowEncryptionKeyType = exports.FlowEncryptionAlgorithm = void 0;
exports.FlowEncryptionAlgorithm = {
    Aes128: "aes128",
    Aes192: "aes192",
    Aes256: "aes256",
};
exports.FlowEncryptionKeyType = {
    Speke: "speke",
    StaticKey: "static-key",
    SrtPassword: "srt-password",
};
exports.FlowEntitlementEncryptionAlgorithm = {
    Aes128: "aes128",
    Aes192: "aes192",
    Aes256: "aes256",
};
exports.FlowEntitlementEncryptionKeyType = {
    Speke: "speke",
    StaticKey: "static-key",
};
exports.FlowEntitlementEntitlementStatus = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.FlowFailoverConfigState = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.FlowOutputEncryptionAlgorithm = {
    Aes128: "aes128",
    Aes192: "aes192",
    Aes256: "aes256",
};
exports.FlowOutputEncryptionKeyType = {
    StaticKey: "static-key",
    SrtPassword: "srt-password",
};
exports.FlowOutputProtocol = {
    ZixiPush: "zixi-push",
    RtpFec: "rtp-fec",
    Rtp: "rtp",
    ZixiPull: "zixi-pull",
    Rist: "rist",
    SrtListener: "srt-listener",
};
exports.FlowSourceEncryptionAlgorithm = {
    Aes128: "aes128",
    Aes192: "aes192",
    Aes256: "aes256",
};
exports.FlowSourceEncryptionKeyType = {
    Speke: "speke",
    StaticKey: "static-key",
};
exports.FlowSourceProtocol = {
    ZixiPush: "zixi-push",
    RtpFec: "rtp-fec",
    Rtp: "rtp",
    Rist: "rist",
    SrtListener: "srt-listener",
};
