// @pulumi/aws-native/logs/queryDefinition.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The resource schema for AWSLogs QueryDefinition
 */
class QueryDefinition extends pulumi.CustomResource {
    /**
     * Create a QueryDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.queryString === undefined) && !opts.urn) {
                throw new Error("Missing required property 'queryString'");
            }
            resourceInputs["logGroupNames"] = args ? args.logGroupNames : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["queryString"] = args ? args.queryString : undefined;
            resourceInputs["queryDefinitionId"] = undefined /*out*/;
        }
        else {
            resourceInputs["logGroupNames"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["queryDefinitionId"] = undefined /*out*/;
            resourceInputs["queryString"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(QueryDefinition.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing QueryDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new QueryDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of QueryDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === QueryDefinition.__pulumiType;
    }
}
exports.QueryDefinition = QueryDefinition;
/** @internal */
QueryDefinition.__pulumiType = 'aws-native:logs:QueryDefinition';
