// @pulumi/aws-native/licensemanager/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.License = exports.Grant = exports.getLicenseOutput = exports.getLicense = exports.getGrantOutput = exports.getGrant = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getGrant = null;
exports.getGrantOutput = null;
utilities.lazyLoad(exports, ["getGrant", "getGrantOutput"], () => require("./getGrant"));
exports.getLicense = null;
exports.getLicenseOutput = null;
utilities.lazyLoad(exports, ["getLicense", "getLicenseOutput"], () => require("./getLicense"));
exports.Grant = null;
utilities.lazyLoad(exports, ["Grant"], () => require("./grant"));
exports.License = null;
utilities.lazyLoad(exports, ["License"], () => require("./license"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:licensemanager:Grant":
                return new exports.Grant(name, undefined, { urn });
            case "aws-native:licensemanager:License":
                return new exports.License(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "licensemanager", _module);
