// @pulumi/aws-native/supportapp/slackChannelConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SlackChannelConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An AWS Support App resource that creates, updates, lists and deletes Slack channel configurations.
 */
class SlackChannelConfiguration extends pulumi.CustomResource {
    /**
     * Create a SlackChannelConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.channelId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'channelId'");
            }
            if ((!args || args.channelRoleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'channelRoleArn'");
            }
            if ((!args || args.notifyOnCaseSeverity === undefined) && !opts.urn) {
                throw new Error("Missing required property 'notifyOnCaseSeverity'");
            }
            if ((!args || args.teamId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'teamId'");
            }
            resourceInputs["channelId"] = args ? args.channelId : undefined;
            resourceInputs["channelName"] = args ? args.channelName : undefined;
            resourceInputs["channelRoleArn"] = args ? args.channelRoleArn : undefined;
            resourceInputs["notifyOnAddCorrespondenceToCase"] = args ? args.notifyOnAddCorrespondenceToCase : undefined;
            resourceInputs["notifyOnCaseSeverity"] = args ? args.notifyOnCaseSeverity : undefined;
            resourceInputs["notifyOnCreateOrReopenCase"] = args ? args.notifyOnCreateOrReopenCase : undefined;
            resourceInputs["notifyOnResolveCase"] = args ? args.notifyOnResolveCase : undefined;
            resourceInputs["teamId"] = args ? args.teamId : undefined;
        }
        else {
            resourceInputs["channelId"] = undefined /*out*/;
            resourceInputs["channelName"] = undefined /*out*/;
            resourceInputs["channelRoleArn"] = undefined /*out*/;
            resourceInputs["notifyOnAddCorrespondenceToCase"] = undefined /*out*/;
            resourceInputs["notifyOnCaseSeverity"] = undefined /*out*/;
            resourceInputs["notifyOnCreateOrReopenCase"] = undefined /*out*/;
            resourceInputs["notifyOnResolveCase"] = undefined /*out*/;
            resourceInputs["teamId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SlackChannelConfiguration.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SlackChannelConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SlackChannelConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SlackChannelConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SlackChannelConfiguration.__pulumiType;
    }
}
exports.SlackChannelConfiguration = SlackChannelConfiguration;
/** @internal */
SlackChannelConfiguration.__pulumiType = 'aws-native:supportapp:SlackChannelConfiguration';
