// @pulumi/aws-native/servicediscovery/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Service = exports.PublicDnsNamespace = exports.PrivateDnsNamespace = exports.Instance = exports.HttpNamespace = exports.getServiceOutput = exports.getService = exports.getPublicDnsNamespaceOutput = exports.getPublicDnsNamespace = exports.getPrivateDnsNamespaceOutput = exports.getPrivateDnsNamespace = exports.getInstanceOutput = exports.getInstance = exports.getHttpNamespaceOutput = exports.getHttpNamespace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getHttpNamespace = null;
exports.getHttpNamespaceOutput = null;
utilities.lazyLoad(exports, ["getHttpNamespace", "getHttpNamespaceOutput"], () => require("./getHttpNamespace"));
exports.getInstance = null;
exports.getInstanceOutput = null;
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
exports.getPrivateDnsNamespace = null;
exports.getPrivateDnsNamespaceOutput = null;
utilities.lazyLoad(exports, ["getPrivateDnsNamespace", "getPrivateDnsNamespaceOutput"], () => require("./getPrivateDnsNamespace"));
exports.getPublicDnsNamespace = null;
exports.getPublicDnsNamespaceOutput = null;
utilities.lazyLoad(exports, ["getPublicDnsNamespace", "getPublicDnsNamespaceOutput"], () => require("./getPublicDnsNamespace"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.HttpNamespace = null;
utilities.lazyLoad(exports, ["HttpNamespace"], () => require("./httpNamespace"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.PrivateDnsNamespace = null;
utilities.lazyLoad(exports, ["PrivateDnsNamespace"], () => require("./privateDnsNamespace"));
exports.PublicDnsNamespace = null;
utilities.lazyLoad(exports, ["PublicDnsNamespace"], () => require("./publicDnsNamespace"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:servicediscovery:HttpNamespace":
                return new exports.HttpNamespace(name, undefined, { urn });
            case "aws-native:servicediscovery:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "aws-native:servicediscovery:PrivateDnsNamespace":
                return new exports.PrivateDnsNamespace(name, undefined, { urn });
            case "aws-native:servicediscovery:PublicDnsNamespace":
                return new exports.PublicDnsNamespace(name, undefined, { urn });
            case "aws-native:servicediscovery:Service":
                return new exports.Service(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "servicediscovery", _module);
