// @pulumi/aws-native/cloudformation/publicTypeVersion.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicTypeVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Test and Publish a resource that has been registered in the CloudFormation Registry.
 */
class PublicTypeVersion extends pulumi.CustomResource {
    /**
     * Create a PublicTypeVersion resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["arn"] = args ? args.arn : undefined;
            resourceInputs["logDeliveryBucket"] = args ? args.logDeliveryBucket : undefined;
            resourceInputs["publicVersionNumber"] = args ? args.publicVersionNumber : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["typeName"] = args ? args.typeName : undefined;
            resourceInputs["publicTypeArn"] = undefined /*out*/;
            resourceInputs["publisherId"] = undefined /*out*/;
            resourceInputs["typeVersionArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["logDeliveryBucket"] = undefined /*out*/;
            resourceInputs["publicTypeArn"] = undefined /*out*/;
            resourceInputs["publicVersionNumber"] = undefined /*out*/;
            resourceInputs["publisherId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["typeName"] = undefined /*out*/;
            resourceInputs["typeVersionArn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PublicTypeVersion.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing PublicTypeVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PublicTypeVersion(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PublicTypeVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PublicTypeVersion.__pulumiType;
    }
}
exports.PublicTypeVersion = PublicTypeVersion;
/** @internal */
PublicTypeVersion.__pulumiType = 'aws-native:cloudformation:PublicTypeVersion';
