// @pulumi/aws-native/cloudformation/stackSet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StackSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * StackSet as a resource provides one-click experience for provisioning a StackSet and StackInstances
 */
class StackSet extends pulumi.CustomResource {
    /**
     * Create a StackSet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.permissionModel === undefined) && !opts.urn) {
                throw new Error("Missing required property 'permissionModel'");
            }
            resourceInputs["administrationRoleARN"] = args ? args.administrationRoleARN : undefined;
            resourceInputs["autoDeployment"] = args ? args.autoDeployment : undefined;
            resourceInputs["callAs"] = args ? args.callAs : undefined;
            resourceInputs["capabilities"] = args ? args.capabilities : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["executionRoleName"] = args ? args.executionRoleName : undefined;
            resourceInputs["managedExecution"] = args ? args.managedExecution : undefined;
            resourceInputs["operationPreferences"] = args ? args.operationPreferences : undefined;
            resourceInputs["parameters"] = args ? args.parameters : undefined;
            resourceInputs["permissionModel"] = args ? args.permissionModel : undefined;
            resourceInputs["stackInstancesGroup"] = args ? args.stackInstancesGroup : undefined;
            resourceInputs["stackSetName"] = args ? args.stackSetName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["templateBody"] = args ? args.templateBody : undefined;
            resourceInputs["templateURL"] = args ? args.templateURL : undefined;
            resourceInputs["stackSetId"] = undefined /*out*/;
        }
        else {
            resourceInputs["administrationRoleARN"] = undefined /*out*/;
            resourceInputs["autoDeployment"] = undefined /*out*/;
            resourceInputs["callAs"] = undefined /*out*/;
            resourceInputs["capabilities"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["executionRoleName"] = undefined /*out*/;
            resourceInputs["managedExecution"] = undefined /*out*/;
            resourceInputs["operationPreferences"] = undefined /*out*/;
            resourceInputs["parameters"] = undefined /*out*/;
            resourceInputs["permissionModel"] = undefined /*out*/;
            resourceInputs["stackInstancesGroup"] = undefined /*out*/;
            resourceInputs["stackSetId"] = undefined /*out*/;
            resourceInputs["stackSetName"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["templateBody"] = undefined /*out*/;
            resourceInputs["templateURL"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(StackSet.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing StackSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StackSet(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StackSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StackSet.__pulumiType;
    }
}
exports.StackSet = StackSet;
/** @internal */
StackSet.__pulumiType = 'aws-native:cloudformation:StackSet';
