// @pulumi/aws-native/cloudformation/waitCondition.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WaitCondition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::CloudFormation::WaitCondition
 *
 * @deprecated WaitCondition is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class WaitCondition extends pulumi.CustomResource {
    /**
     * Create a WaitCondition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated WaitCondition is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("WaitCondition is deprecated: WaitCondition is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["count"] = args ? args.count : undefined;
            resourceInputs["handle"] = args ? args.handle : undefined;
            resourceInputs["timeout"] = args ? args.timeout : undefined;
            resourceInputs["data"] = undefined /*out*/;
        }
        else {
            resourceInputs["count"] = undefined /*out*/;
            resourceInputs["data"] = undefined /*out*/;
            resourceInputs["handle"] = undefined /*out*/;
            resourceInputs["timeout"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(WaitCondition.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing WaitCondition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("WaitCondition is deprecated: WaitCondition is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new WaitCondition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WaitCondition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WaitCondition.__pulumiType;
    }
}
exports.WaitCondition = WaitCondition;
/** @internal */
WaitCondition.__pulumiType = 'aws-native:cloudformation:WaitCondition';
