// @pulumi/aws-native/sso/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PermissionSet = exports.InstanceAccessControlAttributeConfiguration = exports.getPermissionSetOutput = exports.getPermissionSet = exports.getInstanceAccessControlAttributeConfigurationOutput = exports.getInstanceAccessControlAttributeConfiguration = exports.Assignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Assignment = null;
utilities.lazyLoad(exports, ["Assignment"], () => require("./assignment"));
exports.getInstanceAccessControlAttributeConfiguration = null;
exports.getInstanceAccessControlAttributeConfigurationOutput = null;
utilities.lazyLoad(exports, ["getInstanceAccessControlAttributeConfiguration", "getInstanceAccessControlAttributeConfigurationOutput"], () => require("./getInstanceAccessControlAttributeConfiguration"));
exports.getPermissionSet = null;
exports.getPermissionSetOutput = null;
utilities.lazyLoad(exports, ["getPermissionSet", "getPermissionSetOutput"], () => require("./getPermissionSet"));
exports.InstanceAccessControlAttributeConfiguration = null;
utilities.lazyLoad(exports, ["InstanceAccessControlAttributeConfiguration"], () => require("./instanceAccessControlAttributeConfiguration"));
exports.PermissionSet = null;
utilities.lazyLoad(exports, ["PermissionSet"], () => require("./permissionSet"));
// Export enums:
__exportStar(require("../types/enums/sso"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:sso:Assignment":
                return new exports.Assignment(name, undefined, { urn });
            case "aws-native:sso:InstanceAccessControlAttributeConfiguration":
                return new exports.InstanceAccessControlAttributeConfiguration(name, undefined, { urn });
            case "aws-native:sso:PermissionSet":
                return new exports.PermissionSet(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "sso", _module);
