// @pulumi/aws-native/evidently/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Segment = exports.Project = exports.Launch = exports.getSegmentOutput = exports.getSegment = exports.getProjectOutput = exports.getProject = exports.getLaunchOutput = exports.getLaunch = exports.getFeatureOutput = exports.getFeature = exports.getExperimentOutput = exports.getExperiment = exports.Feature = exports.Experiment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Experiment = null;
utilities.lazyLoad(exports, ["Experiment"], () => require("./experiment"));
exports.Feature = null;
utilities.lazyLoad(exports, ["Feature"], () => require("./feature"));
exports.getExperiment = null;
exports.getExperimentOutput = null;
utilities.lazyLoad(exports, ["getExperiment", "getExperimentOutput"], () => require("./getExperiment"));
exports.getFeature = null;
exports.getFeatureOutput = null;
utilities.lazyLoad(exports, ["getFeature", "getFeatureOutput"], () => require("./getFeature"));
exports.getLaunch = null;
exports.getLaunchOutput = null;
utilities.lazyLoad(exports, ["getLaunch", "getLaunchOutput"], () => require("./getLaunch"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getSegment = null;
exports.getSegmentOutput = null;
utilities.lazyLoad(exports, ["getSegment", "getSegmentOutput"], () => require("./getSegment"));
exports.Launch = null;
utilities.lazyLoad(exports, ["Launch"], () => require("./launch"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.Segment = null;
utilities.lazyLoad(exports, ["Segment"], () => require("./segment"));
// Export enums:
__exportStar(require("../types/enums/evidently"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:evidently:Experiment":
                return new exports.Experiment(name, undefined, { urn });
            case "aws-native:evidently:Feature":
                return new exports.Feature(name, undefined, { urn });
            case "aws-native:evidently:Launch":
                return new exports.Launch(name, undefined, { urn });
            case "aws-native:evidently:Project":
                return new exports.Project(name, undefined, { urn });
            case "aws-native:evidently:Segment":
                return new exports.Segment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "evidently", _module);
