// @pulumi/aws-native/dynamodb/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Table = exports.GlobalTable = exports.getTableOutput = exports.getTable = exports.getGlobalTableOutput = exports.getGlobalTable = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getGlobalTable = null;
exports.getGlobalTableOutput = null;
utilities.lazyLoad(exports, ["getGlobalTable", "getGlobalTableOutput"], () => require("./getGlobalTable"));
exports.getTable = null;
exports.getTableOutput = null;
utilities.lazyLoad(exports, ["getTable", "getTableOutput"], () => require("./getTable"));
exports.GlobalTable = null;
utilities.lazyLoad(exports, ["GlobalTable"], () => require("./globalTable"));
exports.Table = null;
utilities.lazyLoad(exports, ["Table"], () => require("./table"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:dynamodb:GlobalTable":
                return new exports.GlobalTable(name, undefined, { urn });
            case "aws-native:dynamodb:Table":
                return new exports.Table(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "dynamodb", _module);
