// @pulumi/aws-native/rekognition/streamProcessor.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamProcessor = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The AWS::Rekognition::StreamProcessor type is used to create an Amazon Rekognition StreamProcessor that you can use to analyze streaming videos.
 */
class StreamProcessor extends pulumi.CustomResource {
    /**
     * Create a StreamProcessor resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kinesisVideoStream === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kinesisVideoStream'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["boundingBoxRegionsOfInterest"] = args ? args.boundingBoxRegionsOfInterest : undefined;
            resourceInputs["connectedHomeSettings"] = args ? args.connectedHomeSettings : undefined;
            resourceInputs["dataSharingPreference"] = args ? args.dataSharingPreference : undefined;
            resourceInputs["faceSearchSettings"] = args ? args.faceSearchSettings : undefined;
            resourceInputs["kinesisDataStream"] = args ? args.kinesisDataStream : undefined;
            resourceInputs["kinesisVideoStream"] = args ? args.kinesisVideoStream : undefined;
            resourceInputs["kmsKeyId"] = args ? args.kmsKeyId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["notificationChannel"] = args ? args.notificationChannel : undefined;
            resourceInputs["polygonRegionsOfInterest"] = args ? args.polygonRegionsOfInterest : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
            resourceInputs["s3Destination"] = args ? args.s3Destination : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["statusMessage"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["boundingBoxRegionsOfInterest"] = undefined /*out*/;
            resourceInputs["connectedHomeSettings"] = undefined /*out*/;
            resourceInputs["dataSharingPreference"] = undefined /*out*/;
            resourceInputs["faceSearchSettings"] = undefined /*out*/;
            resourceInputs["kinesisDataStream"] = undefined /*out*/;
            resourceInputs["kinesisVideoStream"] = undefined /*out*/;
            resourceInputs["kmsKeyId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["notificationChannel"] = undefined /*out*/;
            resourceInputs["polygonRegionsOfInterest"] = undefined /*out*/;
            resourceInputs["roleArn"] = undefined /*out*/;
            resourceInputs["s3Destination"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["statusMessage"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(StreamProcessor.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing StreamProcessor resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StreamProcessor(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StreamProcessor.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StreamProcessor.__pulumiType;
    }
}
exports.StreamProcessor = StreamProcessor;
/** @internal */
StreamProcessor.__pulumiType = 'aws-native:rekognition:StreamProcessor';
