// @pulumi/aws-native/elasticache/cacheCluster.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CacheCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::ElastiCache::CacheCluster
 *
 * @deprecated CacheCluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class CacheCluster extends pulumi.CustomResource {
    /**
     * Create a CacheCluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated CacheCluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("CacheCluster is deprecated: CacheCluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.cacheNodeType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cacheNodeType'");
            }
            if ((!args || args.engine === undefined) && !opts.urn) {
                throw new Error("Missing required property 'engine'");
            }
            if ((!args || args.numCacheNodes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'numCacheNodes'");
            }
            resourceInputs["aZMode"] = args ? args.aZMode : undefined;
            resourceInputs["autoMinorVersionUpgrade"] = args ? args.autoMinorVersionUpgrade : undefined;
            resourceInputs["cacheNodeType"] = args ? args.cacheNodeType : undefined;
            resourceInputs["cacheParameterGroupName"] = args ? args.cacheParameterGroupName : undefined;
            resourceInputs["cacheSecurityGroupNames"] = args ? args.cacheSecurityGroupNames : undefined;
            resourceInputs["cacheSubnetGroupName"] = args ? args.cacheSubnetGroupName : undefined;
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["configurationEndpointAddress"] = args ? args.configurationEndpointAddress : undefined;
            resourceInputs["configurationEndpointPort"] = args ? args.configurationEndpointPort : undefined;
            resourceInputs["engine"] = args ? args.engine : undefined;
            resourceInputs["engineVersion"] = args ? args.engineVersion : undefined;
            resourceInputs["logDeliveryConfigurations"] = args ? args.logDeliveryConfigurations : undefined;
            resourceInputs["notificationTopicArn"] = args ? args.notificationTopicArn : undefined;
            resourceInputs["numCacheNodes"] = args ? args.numCacheNodes : undefined;
            resourceInputs["port"] = args ? args.port : undefined;
            resourceInputs["preferredAvailabilityZone"] = args ? args.preferredAvailabilityZone : undefined;
            resourceInputs["preferredAvailabilityZones"] = args ? args.preferredAvailabilityZones : undefined;
            resourceInputs["preferredMaintenanceWindow"] = args ? args.preferredMaintenanceWindow : undefined;
            resourceInputs["redisEndpointAddress"] = args ? args.redisEndpointAddress : undefined;
            resourceInputs["redisEndpointPort"] = args ? args.redisEndpointPort : undefined;
            resourceInputs["snapshotArns"] = args ? args.snapshotArns : undefined;
            resourceInputs["snapshotName"] = args ? args.snapshotName : undefined;
            resourceInputs["snapshotRetentionLimit"] = args ? args.snapshotRetentionLimit : undefined;
            resourceInputs["snapshotWindow"] = args ? args.snapshotWindow : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["transitEncryptionEnabled"] = args ? args.transitEncryptionEnabled : undefined;
            resourceInputs["vpcSecurityGroupIds"] = args ? args.vpcSecurityGroupIds : undefined;
        }
        else {
            resourceInputs["aZMode"] = undefined /*out*/;
            resourceInputs["autoMinorVersionUpgrade"] = undefined /*out*/;
            resourceInputs["cacheNodeType"] = undefined /*out*/;
            resourceInputs["cacheParameterGroupName"] = undefined /*out*/;
            resourceInputs["cacheSecurityGroupNames"] = undefined /*out*/;
            resourceInputs["cacheSubnetGroupName"] = undefined /*out*/;
            resourceInputs["clusterName"] = undefined /*out*/;
            resourceInputs["configurationEndpointAddress"] = undefined /*out*/;
            resourceInputs["configurationEndpointPort"] = undefined /*out*/;
            resourceInputs["engine"] = undefined /*out*/;
            resourceInputs["engineVersion"] = undefined /*out*/;
            resourceInputs["logDeliveryConfigurations"] = undefined /*out*/;
            resourceInputs["notificationTopicArn"] = undefined /*out*/;
            resourceInputs["numCacheNodes"] = undefined /*out*/;
            resourceInputs["port"] = undefined /*out*/;
            resourceInputs["preferredAvailabilityZone"] = undefined /*out*/;
            resourceInputs["preferredAvailabilityZones"] = undefined /*out*/;
            resourceInputs["preferredMaintenanceWindow"] = undefined /*out*/;
            resourceInputs["redisEndpointAddress"] = undefined /*out*/;
            resourceInputs["redisEndpointPort"] = undefined /*out*/;
            resourceInputs["snapshotArns"] = undefined /*out*/;
            resourceInputs["snapshotName"] = undefined /*out*/;
            resourceInputs["snapshotRetentionLimit"] = undefined /*out*/;
            resourceInputs["snapshotWindow"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["transitEncryptionEnabled"] = undefined /*out*/;
            resourceInputs["vpcSecurityGroupIds"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CacheCluster.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing CacheCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("CacheCluster is deprecated: CacheCluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new CacheCluster(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CacheCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CacheCluster.__pulumiType;
    }
}
exports.CacheCluster = CacheCluster;
/** @internal */
CacheCluster.__pulumiType = 'aws-native:elasticache:CacheCluster';
