// @pulumi/aws-native/aps/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.RuleGroupsNamespace = exports.getWorkspaceOutput = exports.getWorkspace = exports.getRuleGroupsNamespaceOutput = exports.getRuleGroupsNamespace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getRuleGroupsNamespace = null;
exports.getRuleGroupsNamespaceOutput = null;
utilities.lazyLoad(exports, ["getRuleGroupsNamespace", "getRuleGroupsNamespaceOutput"], () => require("./getRuleGroupsNamespace"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.RuleGroupsNamespace = null;
utilities.lazyLoad(exports, ["RuleGroupsNamespace"], () => require("./ruleGroupsNamespace"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:aps:RuleGroupsNamespace":
                return new exports.RuleGroupsNamespace(name, undefined, { urn });
            case "aws-native:aps:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "aps", _module);
