// @pulumi/aws-native/cloudtrail/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Trail = exports.getTrailOutput = exports.getTrail = exports.getEventDataStoreOutput = exports.getEventDataStore = exports.EventDataStore = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.EventDataStore = null;
utilities.lazyLoad(exports, ["EventDataStore"], () => require("./eventDataStore"));
exports.getEventDataStore = null;
exports.getEventDataStoreOutput = null;
utilities.lazyLoad(exports, ["getEventDataStore", "getEventDataStoreOutput"], () => require("./getEventDataStore"));
exports.getTrail = null;
exports.getTrailOutput = null;
utilities.lazyLoad(exports, ["getTrail", "getTrailOutput"], () => require("./getTrail"));
exports.Trail = null;
utilities.lazyLoad(exports, ["Trail"], () => require("./trail"));
// Export enums:
__exportStar(require("../types/enums/cloudtrail"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:cloudtrail:EventDataStore":
                return new exports.EventDataStore(name, undefined, { urn });
            case "aws-native:cloudtrail:Trail":
                return new exports.Trail(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "cloudtrail", _module);
