// @pulumi/aws-native/iotfleetwise/campaign.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Campaign = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::IoTFleetWise::Campaign Resource Type
 *
 * @deprecated Campaign is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Campaign extends pulumi.CustomResource {
    /**
     * Create a Campaign resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Campaign is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Campaign is deprecated: Campaign is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.action === undefined) && !opts.urn) {
                throw new Error("Missing required property 'action'");
            }
            if ((!args || args.collectionScheme === undefined) && !opts.urn) {
                throw new Error("Missing required property 'collectionScheme'");
            }
            if ((!args || args.signalCatalogArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'signalCatalogArn'");
            }
            if ((!args || args.targetArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetArn'");
            }
            resourceInputs["action"] = args ? args.action : undefined;
            resourceInputs["collectionScheme"] = args ? args.collectionScheme : undefined;
            resourceInputs["compression"] = args ? args.compression : undefined;
            resourceInputs["dataExtraDimensions"] = args ? args.dataExtraDimensions : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["diagnosticsMode"] = args ? args.diagnosticsMode : undefined;
            resourceInputs["expiryTime"] = args ? args.expiryTime : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["postTriggerCollectionDuration"] = args ? args.postTriggerCollectionDuration : undefined;
            resourceInputs["priority"] = args ? args.priority : undefined;
            resourceInputs["signalCatalogArn"] = args ? args.signalCatalogArn : undefined;
            resourceInputs["signalsToCollect"] = args ? args.signalsToCollect : undefined;
            resourceInputs["spoolingMode"] = args ? args.spoolingMode : undefined;
            resourceInputs["startTime"] = args ? args.startTime : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["targetArn"] = args ? args.targetArn : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["lastModificationTime"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        else {
            resourceInputs["action"] = undefined /*out*/;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["collectionScheme"] = undefined /*out*/;
            resourceInputs["compression"] = undefined /*out*/;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["dataExtraDimensions"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["diagnosticsMode"] = undefined /*out*/;
            resourceInputs["expiryTime"] = undefined /*out*/;
            resourceInputs["lastModificationTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["postTriggerCollectionDuration"] = undefined /*out*/;
            resourceInputs["priority"] = undefined /*out*/;
            resourceInputs["signalCatalogArn"] = undefined /*out*/;
            resourceInputs["signalsToCollect"] = undefined /*out*/;
            resourceInputs["spoolingMode"] = undefined /*out*/;
            resourceInputs["startTime"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["targetArn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Campaign.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Campaign resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Campaign is deprecated: Campaign is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Campaign(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Campaign.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Campaign.__pulumiType;
    }
}
exports.Campaign = Campaign;
/** @internal */
Campaign.__pulumiType = 'aws-native:iotfleetwise:Campaign';
