// @pulumi/aws-native/cloudwatch/anomalyDetector.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnomalyDetector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::CloudWatch::AnomalyDetector
 *
 * @deprecated AnomalyDetector is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class AnomalyDetector extends pulumi.CustomResource {
    /**
     * Create a AnomalyDetector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated AnomalyDetector is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("AnomalyDetector is deprecated: AnomalyDetector is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["configuration"] = args ? args.configuration : undefined;
            resourceInputs["dimensions"] = args ? args.dimensions : undefined;
            resourceInputs["metricMathAnomalyDetector"] = args ? args.metricMathAnomalyDetector : undefined;
            resourceInputs["metricName"] = args ? args.metricName : undefined;
            resourceInputs["namespace"] = args ? args.namespace : undefined;
            resourceInputs["singleMetricAnomalyDetector"] = args ? args.singleMetricAnomalyDetector : undefined;
            resourceInputs["stat"] = args ? args.stat : undefined;
        }
        else {
            resourceInputs["configuration"] = undefined /*out*/;
            resourceInputs["dimensions"] = undefined /*out*/;
            resourceInputs["metricMathAnomalyDetector"] = undefined /*out*/;
            resourceInputs["metricName"] = undefined /*out*/;
            resourceInputs["namespace"] = undefined /*out*/;
            resourceInputs["singleMetricAnomalyDetector"] = undefined /*out*/;
            resourceInputs["stat"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AnomalyDetector.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing AnomalyDetector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("AnomalyDetector is deprecated: AnomalyDetector is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new AnomalyDetector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AnomalyDetector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AnomalyDetector.__pulumiType;
    }
}
exports.AnomalyDetector = AnomalyDetector;
/** @internal */
AnomalyDetector.__pulumiType = 'aws-native:cloudwatch:AnomalyDetector';
