// @pulumi/aws-native/cloudwatch/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricStream = exports.InsightRule = exports.getMetricStreamOutput = exports.getMetricStream = exports.getInsightRuleOutput = exports.getInsightRule = exports.getDashboardOutput = exports.getDashboard = exports.getCompositeAlarmOutput = exports.getCompositeAlarm = exports.getAnomalyDetectorOutput = exports.getAnomalyDetector = exports.getAlarmOutput = exports.getAlarm = exports.Dashboard = exports.CompositeAlarm = exports.AnomalyDetector = exports.Alarm = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Alarm = null;
utilities.lazyLoad(exports, ["Alarm"], () => require("./alarm"));
exports.AnomalyDetector = null;
utilities.lazyLoad(exports, ["AnomalyDetector"], () => require("./anomalyDetector"));
exports.CompositeAlarm = null;
utilities.lazyLoad(exports, ["CompositeAlarm"], () => require("./compositeAlarm"));
exports.Dashboard = null;
utilities.lazyLoad(exports, ["Dashboard"], () => require("./dashboard"));
exports.getAlarm = null;
exports.getAlarmOutput = null;
utilities.lazyLoad(exports, ["getAlarm", "getAlarmOutput"], () => require("./getAlarm"));
exports.getAnomalyDetector = null;
exports.getAnomalyDetectorOutput = null;
utilities.lazyLoad(exports, ["getAnomalyDetector", "getAnomalyDetectorOutput"], () => require("./getAnomalyDetector"));
exports.getCompositeAlarm = null;
exports.getCompositeAlarmOutput = null;
utilities.lazyLoad(exports, ["getCompositeAlarm", "getCompositeAlarmOutput"], () => require("./getCompositeAlarm"));
exports.getDashboard = null;
exports.getDashboardOutput = null;
utilities.lazyLoad(exports, ["getDashboard", "getDashboardOutput"], () => require("./getDashboard"));
exports.getInsightRule = null;
exports.getInsightRuleOutput = null;
utilities.lazyLoad(exports, ["getInsightRule", "getInsightRuleOutput"], () => require("./getInsightRule"));
exports.getMetricStream = null;
exports.getMetricStreamOutput = null;
utilities.lazyLoad(exports, ["getMetricStream", "getMetricStreamOutput"], () => require("./getMetricStream"));
exports.InsightRule = null;
utilities.lazyLoad(exports, ["InsightRule"], () => require("./insightRule"));
exports.MetricStream = null;
utilities.lazyLoad(exports, ["MetricStream"], () => require("./metricStream"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:cloudwatch:Alarm":
                return new exports.Alarm(name, undefined, { urn });
            case "aws-native:cloudwatch:AnomalyDetector":
                return new exports.AnomalyDetector(name, undefined, { urn });
            case "aws-native:cloudwatch:CompositeAlarm":
                return new exports.CompositeAlarm(name, undefined, { urn });
            case "aws-native:cloudwatch:Dashboard":
                return new exports.Dashboard(name, undefined, { urn });
            case "aws-native:cloudwatch:InsightRule":
                return new exports.InsightRule(name, undefined, { urn });
            case "aws-native:cloudwatch:MetricStream":
                return new exports.MetricStream(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "cloudwatch", _module);
