// @pulumi/aws-native/cognito/identityPool.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Cognito::IdentityPool
 *
 * @deprecated IdentityPool is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class IdentityPool extends pulumi.CustomResource {
    /**
     * Create a IdentityPool resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated IdentityPool is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("IdentityPool is deprecated: IdentityPool is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.allowUnauthenticatedIdentities === undefined) && !opts.urn) {
                throw new Error("Missing required property 'allowUnauthenticatedIdentities'");
            }
            resourceInputs["allowClassicFlow"] = args ? args.allowClassicFlow : undefined;
            resourceInputs["allowUnauthenticatedIdentities"] = args ? args.allowUnauthenticatedIdentities : undefined;
            resourceInputs["cognitoEvents"] = args ? args.cognitoEvents : undefined;
            resourceInputs["cognitoIdentityProviders"] = args ? args.cognitoIdentityProviders : undefined;
            resourceInputs["cognitoStreams"] = args ? args.cognitoStreams : undefined;
            resourceInputs["developerProviderName"] = args ? args.developerProviderName : undefined;
            resourceInputs["identityPoolName"] = args ? args.identityPoolName : undefined;
            resourceInputs["openIdConnectProviderARNs"] = args ? args.openIdConnectProviderARNs : undefined;
            resourceInputs["pushSync"] = args ? args.pushSync : undefined;
            resourceInputs["samlProviderARNs"] = args ? args.samlProviderARNs : undefined;
            resourceInputs["supportedLoginProviders"] = args ? args.supportedLoginProviders : undefined;
            resourceInputs["name"] = undefined /*out*/;
        }
        else {
            resourceInputs["allowClassicFlow"] = undefined /*out*/;
            resourceInputs["allowUnauthenticatedIdentities"] = undefined /*out*/;
            resourceInputs["cognitoEvents"] = undefined /*out*/;
            resourceInputs["cognitoIdentityProviders"] = undefined /*out*/;
            resourceInputs["cognitoStreams"] = undefined /*out*/;
            resourceInputs["developerProviderName"] = undefined /*out*/;
            resourceInputs["identityPoolName"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["openIdConnectProviderARNs"] = undefined /*out*/;
            resourceInputs["pushSync"] = undefined /*out*/;
            resourceInputs["samlProviderARNs"] = undefined /*out*/;
            resourceInputs["supportedLoginProviders"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(IdentityPool.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing IdentityPool resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("IdentityPool is deprecated: IdentityPool is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new IdentityPool(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IdentityPool.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IdentityPool.__pulumiType;
    }
}
exports.IdentityPool = IdentityPool;
/** @internal */
IdentityPool.__pulumiType = 'aws-native:cognito:IdentityPool';
