// @pulumi/aws-native/types/enums/ssm/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentUpdateMethod = exports.DocumentType = exports.DocumentFormat = exports.DocumentAttachmentsSourceKey = exports.AssociationSyncCompliance = exports.AssociationComplianceSeverity = void 0;
exports.AssociationComplianceSeverity = {
    Critical: "CRITICAL",
    High: "HIGH",
    Medium: "MEDIUM",
    Low: "LOW",
    Unspecified: "UNSPECIFIED",
};
exports.AssociationSyncCompliance = {
    Auto: "AUTO",
    Manual: "MANUAL",
};
exports.DocumentAttachmentsSourceKey = {
    SourceUrl: "SourceUrl",
    S3FileUrl: "S3FileUrl",
    AttachmentReference: "AttachmentReference",
};
exports.DocumentFormat = {
    Yaml: "YAML",
    Json: "JSON",
    Text: "TEXT",
};
exports.DocumentType = {
    ApplicationConfiguration: "ApplicationConfiguration",
    ApplicationConfigurationSchema: "ApplicationConfigurationSchema",
    Automation: "Automation",
    AutomationChangeTemplate: "Automation.ChangeTemplate",
    ChangeCalendar: "ChangeCalendar",
    CloudFormation: "CloudFormation",
    Command: "Command",
    DeploymentStrategy: "DeploymentStrategy",
    Package: "Package",
    Policy: "Policy",
    ProblemAnalysis: "ProblemAnalysis",
    ProblemAnalysisTemplate: "ProblemAnalysisTemplate",
    Session: "Session",
};
exports.DocumentUpdateMethod = {
    Replace: "Replace",
    NewVersion: "NewVersion",
};
