// @pulumi/aws-native/secretsmanager/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecretTargetAttachment = exports.Secret = exports.RotationSchedule = exports.ResourcePolicy = exports.getSecretTargetAttachmentOutput = exports.getSecretTargetAttachment = exports.getSecretOutput = exports.getSecret = exports.getRotationScheduleOutput = exports.getRotationSchedule = exports.getResourcePolicyOutput = exports.getResourcePolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getResourcePolicy = null;
exports.getResourcePolicyOutput = null;
utilities.lazyLoad(exports, ["getResourcePolicy", "getResourcePolicyOutput"], () => require("./getResourcePolicy"));
exports.getRotationSchedule = null;
exports.getRotationScheduleOutput = null;
utilities.lazyLoad(exports, ["getRotationSchedule", "getRotationScheduleOutput"], () => require("./getRotationSchedule"));
exports.getSecret = null;
exports.getSecretOutput = null;
utilities.lazyLoad(exports, ["getSecret", "getSecretOutput"], () => require("./getSecret"));
exports.getSecretTargetAttachment = null;
exports.getSecretTargetAttachmentOutput = null;
utilities.lazyLoad(exports, ["getSecretTargetAttachment", "getSecretTargetAttachmentOutput"], () => require("./getSecretTargetAttachment"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.RotationSchedule = null;
utilities.lazyLoad(exports, ["RotationSchedule"], () => require("./rotationSchedule"));
exports.Secret = null;
utilities.lazyLoad(exports, ["Secret"], () => require("./secret"));
exports.SecretTargetAttachment = null;
utilities.lazyLoad(exports, ["SecretTargetAttachment"], () => require("./secretTargetAttachment"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:secretsmanager:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws-native:secretsmanager:RotationSchedule":
                return new exports.RotationSchedule(name, undefined, { urn });
            case "aws-native:secretsmanager:Secret":
                return new exports.Secret(name, undefined, { urn });
            case "aws-native:secretsmanager:SecretTargetAttachment":
                return new exports.SecretTargetAttachment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "secretsmanager", _module);
