// @pulumi/aws-native/types/enums/quicksight/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThemeType = exports.ThemeResourceStatus = exports.ThemeErrorType = exports.TemplateResourceStatus = exports.TemplateErrorType = exports.DataSourceType = exports.DataSourceResourceStatus = exports.DataSourceErrorInfoType = exports.DataSetRowLevelPermissionPolicy = exports.DataSetRowLevelPermissionFormatVersion = exports.DataSetImportMode = exports.DataSetGeoSpatialCountryCode = exports.DataSetColumnDataType = exports.DashboardUIState = exports.DashboardResourceStatus = exports.DashboardErrorType = exports.DashboardBehavior = exports.AnalysisResourceStatus = exports.AnalysisErrorType = void 0;
exports.AnalysisErrorType = {
    AccessDenied: "ACCESS_DENIED",
    SourceNotFound: "SOURCE_NOT_FOUND",
    DataSetNotFound: "DATA_SET_NOT_FOUND",
    InternalFailure: "INTERNAL_FAILURE",
    ParameterValueIncompatible: "PARAMETER_VALUE_INCOMPATIBLE",
    ParameterTypeInvalid: "PARAMETER_TYPE_INVALID",
    ParameterNotFound: "PARAMETER_NOT_FOUND",
    ColumnTypeMismatch: "COLUMN_TYPE_MISMATCH",
    ColumnGeographicRoleMismatch: "COLUMN_GEOGRAPHIC_ROLE_MISMATCH",
    ColumnReplacementMissing: "COLUMN_REPLACEMENT_MISSING",
};
exports.AnalysisResourceStatus = {
    CreationInProgress: "CREATION_IN_PROGRESS",
    CreationSuccessful: "CREATION_SUCCESSFUL",
    CreationFailed: "CREATION_FAILED",
    UpdateInProgress: "UPDATE_IN_PROGRESS",
    UpdateSuccessful: "UPDATE_SUCCESSFUL",
    UpdateFailed: "UPDATE_FAILED",
    Deleted: "DELETED",
};
exports.DashboardBehavior = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.DashboardErrorType = {
    AccessDenied: "ACCESS_DENIED",
    SourceNotFound: "SOURCE_NOT_FOUND",
    DataSetNotFound: "DATA_SET_NOT_FOUND",
    InternalFailure: "INTERNAL_FAILURE",
    ParameterValueIncompatible: "PARAMETER_VALUE_INCOMPATIBLE",
    ParameterTypeInvalid: "PARAMETER_TYPE_INVALID",
    ParameterNotFound: "PARAMETER_NOT_FOUND",
    ColumnTypeMismatch: "COLUMN_TYPE_MISMATCH",
    ColumnGeographicRoleMismatch: "COLUMN_GEOGRAPHIC_ROLE_MISMATCH",
    ColumnReplacementMissing: "COLUMN_REPLACEMENT_MISSING",
};
exports.DashboardResourceStatus = {
    CreationInProgress: "CREATION_IN_PROGRESS",
    CreationSuccessful: "CREATION_SUCCESSFUL",
    CreationFailed: "CREATION_FAILED",
    UpdateInProgress: "UPDATE_IN_PROGRESS",
    UpdateSuccessful: "UPDATE_SUCCESSFUL",
    UpdateFailed: "UPDATE_FAILED",
    Deleted: "DELETED",
};
exports.DashboardUIState = {
    Expanded: "EXPANDED",
    Collapsed: "COLLAPSED",
};
exports.DataSetColumnDataType = {
    String: "STRING",
    Integer: "INTEGER",
    Decimal: "DECIMAL",
    Datetime: "DATETIME",
};
exports.DataSetGeoSpatialCountryCode = {
    Us: "US",
};
exports.DataSetImportMode = {
    Spice: "SPICE",
    DirectQuery: "DIRECT_QUERY",
};
exports.DataSetRowLevelPermissionFormatVersion = {
    Version1: "VERSION_1",
    Version2: "VERSION_2",
};
exports.DataSetRowLevelPermissionPolicy = {
    GrantAccess: "GRANT_ACCESS",
    DenyAccess: "DENY_ACCESS",
};
exports.DataSourceErrorInfoType = {
    AccessDenied: "ACCESS_DENIED",
    CopySourceNotFound: "COPY_SOURCE_NOT_FOUND",
    Timeout: "TIMEOUT",
    EngineVersionNotSupported: "ENGINE_VERSION_NOT_SUPPORTED",
    UnknownHost: "UNKNOWN_HOST",
    GenericSqlFailure: "GENERIC_SQL_FAILURE",
    Conflict: "CONFLICT",
    Unknown: "UNKNOWN",
};
exports.DataSourceResourceStatus = {
    CreationInProgress: "CREATION_IN_PROGRESS",
    CreationSuccessful: "CREATION_SUCCESSFUL",
    CreationFailed: "CREATION_FAILED",
    UpdateInProgress: "UPDATE_IN_PROGRESS",
    UpdateSuccessful: "UPDATE_SUCCESSFUL",
    UpdateFailed: "UPDATE_FAILED",
    Deleted: "DELETED",
};
exports.DataSourceType = {
    AdobeAnalytics: "ADOBE_ANALYTICS",
    AmazonElasticsearch: "AMAZON_ELASTICSEARCH",
    AmazonOpensearch: "AMAZON_OPENSEARCH",
    Athena: "ATHENA",
    Aurora: "AURORA",
    AuroraPostgresql: "AURORA_POSTGRESQL",
    AwsIotAnalytics: "AWS_IOT_ANALYTICS",
    Github: "GITHUB",
    Jira: "JIRA",
    Mariadb: "MARIADB",
    Mysql: "MYSQL",
    Oracle: "ORACLE",
    Postgresql: "POSTGRESQL",
    Presto: "PRESTO",
    Redshift: "REDSHIFT",
    S3: "S3",
    Salesforce: "SALESFORCE",
    Servicenow: "SERVICENOW",
    Snowflake: "SNOWFLAKE",
    Spark: "SPARK",
    Sqlserver: "SQLSERVER",
    Teradata: "TERADATA",
    Twitter: "TWITTER",
    Timestream: "TIMESTREAM",
};
exports.TemplateErrorType = {
    SourceNotFound: "SOURCE_NOT_FOUND",
    DataSetNotFound: "DATA_SET_NOT_FOUND",
    InternalFailure: "INTERNAL_FAILURE",
    AccessDenied: "ACCESS_DENIED",
};
exports.TemplateResourceStatus = {
    CreationInProgress: "CREATION_IN_PROGRESS",
    CreationSuccessful: "CREATION_SUCCESSFUL",
    CreationFailed: "CREATION_FAILED",
    UpdateInProgress: "UPDATE_IN_PROGRESS",
    UpdateSuccessful: "UPDATE_SUCCESSFUL",
    UpdateFailed: "UPDATE_FAILED",
    Deleted: "DELETED",
};
exports.ThemeErrorType = {
    InternalFailure: "INTERNAL_FAILURE",
};
exports.ThemeResourceStatus = {
    CreationInProgress: "CREATION_IN_PROGRESS",
    CreationSuccessful: "CREATION_SUCCESSFUL",
    CreationFailed: "CREATION_FAILED",
    UpdateInProgress: "UPDATE_IN_PROGRESS",
    UpdateSuccessful: "UPDATE_SUCCESSFUL",
    UpdateFailed: "UPDATE_FAILED",
    Deleted: "DELETED",
};
exports.ThemeType = {
    Quicksight: "QUICKSIGHT",
    Custom: "CUSTOM",
    All: "ALL",
};
