// @pulumi/aws-native/types/enums/ce/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostCategoryRuleVersion = exports.AnomalySubscriptionSubscriberType = exports.AnomalySubscriptionSubscriberStatus = exports.AnomalySubscriptionFrequency = exports.AnomalyMonitorMonitorType = exports.AnomalyMonitorMonitorDimension = void 0;
exports.AnomalyMonitorMonitorDimension = {
    Service: "SERVICE",
};
exports.AnomalyMonitorMonitorType = {
    Dimensional: "DIMENSIONAL",
    Custom: "CUSTOM",
};
exports.AnomalySubscriptionFrequency = {
    Daily: "DAILY",
    Immediate: "IMMEDIATE",
    Weekly: "WEEKLY",
};
exports.AnomalySubscriptionSubscriberStatus = {
    Confirmed: "CONFIRMED",
    Declined: "DECLINED",
};
exports.AnomalySubscriptionSubscriberType = {
    Email: "EMAIL",
    Sns: "SNS",
};
exports.CostCategoryRuleVersion = {
    CostCategoryExpressionV1: "CostCategoryExpression.v1",
};
