// @pulumi/aws-native/qldb/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Stream = exports.Ledger = exports.getStreamOutput = exports.getStream = exports.getLedgerOutput = exports.getLedger = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getLedger = null;
exports.getLedgerOutput = null;
utilities.lazyLoad(exports, ["getLedger", "getLedgerOutput"], () => require("./getLedger"));
exports.getStream = null;
exports.getStreamOutput = null;
utilities.lazyLoad(exports, ["getStream", "getStreamOutput"], () => require("./getStream"));
exports.Ledger = null;
utilities.lazyLoad(exports, ["Ledger"], () => require("./ledger"));
exports.Stream = null;
utilities.lazyLoad(exports, ["Stream"], () => require("./stream"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:qldb:Ledger":
                return new exports.Ledger(name, undefined, { urn });
            case "aws-native:qldb:Stream":
                return new exports.Stream(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "qldb", _module);
