// @pulumi/aws-native/cloudfront/function.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Function = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::CloudFront::Function
 */
class Function extends pulumi.CustomResource {
    /**
     * Create a Function resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["autoPublish"] = args ? args.autoPublish : undefined;
            resourceInputs["functionCode"] = args ? args.functionCode : undefined;
            resourceInputs["functionConfig"] = args ? args.functionConfig : undefined;
            resourceInputs["functionMetadata"] = args ? args.functionMetadata : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["functionARN"] = undefined /*out*/;
            resourceInputs["stage"] = undefined /*out*/;
        }
        else {
            resourceInputs["autoPublish"] = undefined /*out*/;
            resourceInputs["functionARN"] = undefined /*out*/;
            resourceInputs["functionCode"] = undefined /*out*/;
            resourceInputs["functionConfig"] = undefined /*out*/;
            resourceInputs["functionMetadata"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["stage"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Function.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Function resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Function(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Function.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Function.__pulumiType;
    }
}
exports.Function = Function;
/** @internal */
Function.__pulumiType = 'aws-native:cloudfront:Function';
