// @pulumi/aws-native/pinpoint/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getVoiceChannel = exports.getSmsTemplateOutput = exports.getSmsTemplate = exports.getSegmentOutput = exports.getSegment = exports.getSMSChannelOutput = exports.getSMSChannel = exports.getPushTemplateOutput = exports.getPushTemplate = exports.getInAppTemplateOutput = exports.getInAppTemplate = exports.getGCMChannelOutput = exports.getGCMChannel = exports.getEventStreamOutput = exports.getEventStream = exports.getEmailTemplateOutput = exports.getEmailTemplate = exports.getEmailChannelOutput = exports.getEmailChannel = exports.getCampaignOutput = exports.getCampaign = exports.getBaiduChannelOutput = exports.getBaiduChannel = exports.getApplicationSettingsOutput = exports.getApplicationSettings = exports.getAppOutput = exports.getApp = exports.getAPNSVoipSandboxChannelOutput = exports.getAPNSVoipSandboxChannel = exports.getAPNSVoipChannelOutput = exports.getAPNSVoipChannel = exports.getAPNSSandboxChannelOutput = exports.getAPNSSandboxChannel = exports.getAPNSChannelOutput = exports.getAPNSChannel = exports.getADMChannelOutput = exports.getADMChannel = exports.GCMChannel = exports.EventStream = exports.EmailTemplate = exports.EmailChannel = exports.Campaign = exports.BaiduChannel = exports.ApplicationSettings = exports.App = exports.APNSVoipSandboxChannel = exports.APNSVoipChannel = exports.APNSSandboxChannel = exports.APNSChannel = exports.ADMChannel = void 0;
exports.VoiceChannel = exports.SMSChannel = exports.SmsTemplate = exports.Segment = exports.PushTemplate = exports.InAppTemplate = exports.getVoiceChannelOutput = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ADMChannel = null;
utilities.lazyLoad(exports, ["ADMChannel"], () => require("./admchannel"));
exports.APNSChannel = null;
utilities.lazyLoad(exports, ["APNSChannel"], () => require("./apnschannel"));
exports.APNSSandboxChannel = null;
utilities.lazyLoad(exports, ["APNSSandboxChannel"], () => require("./apnssandboxChannel"));
exports.APNSVoipChannel = null;
utilities.lazyLoad(exports, ["APNSVoipChannel"], () => require("./apnsvoipChannel"));
exports.APNSVoipSandboxChannel = null;
utilities.lazyLoad(exports, ["APNSVoipSandboxChannel"], () => require("./apnsvoipSandboxChannel"));
exports.App = null;
utilities.lazyLoad(exports, ["App"], () => require("./app"));
exports.ApplicationSettings = null;
utilities.lazyLoad(exports, ["ApplicationSettings"], () => require("./applicationSettings"));
exports.BaiduChannel = null;
utilities.lazyLoad(exports, ["BaiduChannel"], () => require("./baiduChannel"));
exports.Campaign = null;
utilities.lazyLoad(exports, ["Campaign"], () => require("./campaign"));
exports.EmailChannel = null;
utilities.lazyLoad(exports, ["EmailChannel"], () => require("./emailChannel"));
exports.EmailTemplate = null;
utilities.lazyLoad(exports, ["EmailTemplate"], () => require("./emailTemplate"));
exports.EventStream = null;
utilities.lazyLoad(exports, ["EventStream"], () => require("./eventStream"));
exports.GCMChannel = null;
utilities.lazyLoad(exports, ["GCMChannel"], () => require("./gcmchannel"));
exports.getADMChannel = null;
exports.getADMChannelOutput = null;
utilities.lazyLoad(exports, ["getADMChannel", "getADMChannelOutput"], () => require("./getADMChannel"));
exports.getAPNSChannel = null;
exports.getAPNSChannelOutput = null;
utilities.lazyLoad(exports, ["getAPNSChannel", "getAPNSChannelOutput"], () => require("./getAPNSChannel"));
exports.getAPNSSandboxChannel = null;
exports.getAPNSSandboxChannelOutput = null;
utilities.lazyLoad(exports, ["getAPNSSandboxChannel", "getAPNSSandboxChannelOutput"], () => require("./getAPNSSandboxChannel"));
exports.getAPNSVoipChannel = null;
exports.getAPNSVoipChannelOutput = null;
utilities.lazyLoad(exports, ["getAPNSVoipChannel", "getAPNSVoipChannelOutput"], () => require("./getAPNSVoipChannel"));
exports.getAPNSVoipSandboxChannel = null;
exports.getAPNSVoipSandboxChannelOutput = null;
utilities.lazyLoad(exports, ["getAPNSVoipSandboxChannel", "getAPNSVoipSandboxChannelOutput"], () => require("./getAPNSVoipSandboxChannel"));
exports.getApp = null;
exports.getAppOutput = null;
utilities.lazyLoad(exports, ["getApp", "getAppOutput"], () => require("./getApp"));
exports.getApplicationSettings = null;
exports.getApplicationSettingsOutput = null;
utilities.lazyLoad(exports, ["getApplicationSettings", "getApplicationSettingsOutput"], () => require("./getApplicationSettings"));
exports.getBaiduChannel = null;
exports.getBaiduChannelOutput = null;
utilities.lazyLoad(exports, ["getBaiduChannel", "getBaiduChannelOutput"], () => require("./getBaiduChannel"));
exports.getCampaign = null;
exports.getCampaignOutput = null;
utilities.lazyLoad(exports, ["getCampaign", "getCampaignOutput"], () => require("./getCampaign"));
exports.getEmailChannel = null;
exports.getEmailChannelOutput = null;
utilities.lazyLoad(exports, ["getEmailChannel", "getEmailChannelOutput"], () => require("./getEmailChannel"));
exports.getEmailTemplate = null;
exports.getEmailTemplateOutput = null;
utilities.lazyLoad(exports, ["getEmailTemplate", "getEmailTemplateOutput"], () => require("./getEmailTemplate"));
exports.getEventStream = null;
exports.getEventStreamOutput = null;
utilities.lazyLoad(exports, ["getEventStream", "getEventStreamOutput"], () => require("./getEventStream"));
exports.getGCMChannel = null;
exports.getGCMChannelOutput = null;
utilities.lazyLoad(exports, ["getGCMChannel", "getGCMChannelOutput"], () => require("./getGCMChannel"));
exports.getInAppTemplate = null;
exports.getInAppTemplateOutput = null;
utilities.lazyLoad(exports, ["getInAppTemplate", "getInAppTemplateOutput"], () => require("./getInAppTemplate"));
exports.getPushTemplate = null;
exports.getPushTemplateOutput = null;
utilities.lazyLoad(exports, ["getPushTemplate", "getPushTemplateOutput"], () => require("./getPushTemplate"));
exports.getSMSChannel = null;
exports.getSMSChannelOutput = null;
utilities.lazyLoad(exports, ["getSMSChannel", "getSMSChannelOutput"], () => require("./getSMSChannel"));
exports.getSegment = null;
exports.getSegmentOutput = null;
utilities.lazyLoad(exports, ["getSegment", "getSegmentOutput"], () => require("./getSegment"));
exports.getSmsTemplate = null;
exports.getSmsTemplateOutput = null;
utilities.lazyLoad(exports, ["getSmsTemplate", "getSmsTemplateOutput"], () => require("./getSmsTemplate"));
exports.getVoiceChannel = null;
exports.getVoiceChannelOutput = null;
utilities.lazyLoad(exports, ["getVoiceChannel", "getVoiceChannelOutput"], () => require("./getVoiceChannel"));
exports.InAppTemplate = null;
utilities.lazyLoad(exports, ["InAppTemplate"], () => require("./inAppTemplate"));
exports.PushTemplate = null;
utilities.lazyLoad(exports, ["PushTemplate"], () => require("./pushTemplate"));
exports.Segment = null;
utilities.lazyLoad(exports, ["Segment"], () => require("./segment"));
exports.SmsTemplate = null;
utilities.lazyLoad(exports, ["SmsTemplate"], () => require("./smsTemplate"));
exports.SMSChannel = null;
utilities.lazyLoad(exports, ["SMSChannel"], () => require("./smschannel"));
exports.VoiceChannel = null;
utilities.lazyLoad(exports, ["VoiceChannel"], () => require("./voiceChannel"));
// Export enums:
__exportStar(require("../types/enums/pinpoint"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:pinpoint:ADMChannel":
                return new exports.ADMChannel(name, undefined, { urn });
            case "aws-native:pinpoint:APNSChannel":
                return new exports.APNSChannel(name, undefined, { urn });
            case "aws-native:pinpoint:APNSSandboxChannel":
                return new exports.APNSSandboxChannel(name, undefined, { urn });
            case "aws-native:pinpoint:APNSVoipChannel":
                return new exports.APNSVoipChannel(name, undefined, { urn });
            case "aws-native:pinpoint:APNSVoipSandboxChannel":
                return new exports.APNSVoipSandboxChannel(name, undefined, { urn });
            case "aws-native:pinpoint:App":
                return new exports.App(name, undefined, { urn });
            case "aws-native:pinpoint:ApplicationSettings":
                return new exports.ApplicationSettings(name, undefined, { urn });
            case "aws-native:pinpoint:BaiduChannel":
                return new exports.BaiduChannel(name, undefined, { urn });
            case "aws-native:pinpoint:Campaign":
                return new exports.Campaign(name, undefined, { urn });
            case "aws-native:pinpoint:EmailChannel":
                return new exports.EmailChannel(name, undefined, { urn });
            case "aws-native:pinpoint:EmailTemplate":
                return new exports.EmailTemplate(name, undefined, { urn });
            case "aws-native:pinpoint:EventStream":
                return new exports.EventStream(name, undefined, { urn });
            case "aws-native:pinpoint:GCMChannel":
                return new exports.GCMChannel(name, undefined, { urn });
            case "aws-native:pinpoint:InAppTemplate":
                return new exports.InAppTemplate(name, undefined, { urn });
            case "aws-native:pinpoint:PushTemplate":
                return new exports.PushTemplate(name, undefined, { urn });
            case "aws-native:pinpoint:SMSChannel":
                return new exports.SMSChannel(name, undefined, { urn });
            case "aws-native:pinpoint:Segment":
                return new exports.Segment(name, undefined, { urn });
            case "aws-native:pinpoint:SmsTemplate":
                return new exports.SmsTemplate(name, undefined, { urn });
            case "aws-native:pinpoint:VoiceChannel":
                return new exports.VoiceChannel(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "pinpoint", _module);
