// @pulumi/aws-native/pinpoint/pushTemplate.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PushTemplate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Pinpoint::PushTemplate
 *
 * @deprecated PushTemplate is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class PushTemplate extends pulumi.CustomResource {
    /**
     * Create a PushTemplate resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated PushTemplate is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("PushTemplate is deprecated: PushTemplate is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.templateName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'templateName'");
            }
            resourceInputs["aDM"] = args ? args.aDM : undefined;
            resourceInputs["aPNS"] = args ? args.aPNS : undefined;
            resourceInputs["baidu"] = args ? args.baidu : undefined;
            resourceInputs["default"] = args ? args.default : undefined;
            resourceInputs["defaultSubstitutions"] = args ? args.defaultSubstitutions : undefined;
            resourceInputs["gCM"] = args ? args.gCM : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["templateDescription"] = args ? args.templateDescription : undefined;
            resourceInputs["templateName"] = args ? args.templateName : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        else {
            resourceInputs["aDM"] = undefined /*out*/;
            resourceInputs["aPNS"] = undefined /*out*/;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["baidu"] = undefined /*out*/;
            resourceInputs["default"] = undefined /*out*/;
            resourceInputs["defaultSubstitutions"] = undefined /*out*/;
            resourceInputs["gCM"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["templateDescription"] = undefined /*out*/;
            resourceInputs["templateName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PushTemplate.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing PushTemplate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("PushTemplate is deprecated: PushTemplate is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new PushTemplate(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PushTemplate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PushTemplate.__pulumiType;
    }
}
exports.PushTemplate = PushTemplate;
/** @internal */
PushTemplate.__pulumiType = 'aws-native:pinpoint:PushTemplate';
