// @pulumi/aws-native/types/enums/logs/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricFilterMetricTransformationUnit = void 0;
exports.MetricFilterMetricTransformationUnit = {
    Seconds: "Seconds",
    Microseconds: "Microseconds",
    Milliseconds: "Milliseconds",
    Bytes: "Bytes",
    Kilobytes: "Kilobytes",
    Megabytes: "Megabytes",
    Gigabytes: "Gigabytes",
    Terabytes: "Terabytes",
    Bits: "Bits",
    Kilobits: "Kilobits",
    Megabits: "Megabits",
    Gigabits: "Gigabits",
    Terabits: "Terabits",
    Percent: "Percent",
    Count: "Count",
    BytesSecond: "Bytes/Second",
    KilobytesSecond: "Kilobytes/Second",
    MegabytesSecond: "Megabytes/Second",
    GigabytesSecond: "Gigabytes/Second",
    TerabytesSecond: "Terabytes/Second",
    BitsSecond: "Bits/Second",
    KilobitsSecond: "Kilobits/Second",
    MegabitsSecond: "Megabits/Second",
    GigabitsSecond: "Gigabits/Second",
    TerabitsSecond: "Terabits/Second",
    CountSecond: "Count/Second",
    None: "None",
};
