// @pulumi/aws-native/ssmincidents/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponsePlan = exports.ReplicationSet = exports.getResponsePlanOutput = exports.getResponsePlan = exports.getReplicationSetOutput = exports.getReplicationSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getReplicationSet = null;
exports.getReplicationSetOutput = null;
utilities.lazyLoad(exports, ["getReplicationSet", "getReplicationSetOutput"], () => require("./getReplicationSet"));
exports.getResponsePlan = null;
exports.getResponsePlanOutput = null;
utilities.lazyLoad(exports, ["getResponsePlan", "getResponsePlanOutput"], () => require("./getResponsePlan"));
exports.ReplicationSet = null;
utilities.lazyLoad(exports, ["ReplicationSet"], () => require("./replicationSet"));
exports.ResponsePlan = null;
utilities.lazyLoad(exports, ["ResponsePlan"], () => require("./responsePlan"));
// Export enums:
__exportStar(require("../types/enums/ssmincidents"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:ssmincidents:ReplicationSet":
                return new exports.ReplicationSet(name, undefined, { urn });
            case "aws-native:ssmincidents:ResponsePlan":
                return new exports.ResponsePlan(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "ssmincidents", _module);
