// @pulumi/aws-native/redshiftserverless/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workgroup = exports.Namespace = exports.getWorkgroupOutput = exports.getWorkgroup = exports.getNamespaceOutput = exports.getNamespace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getNamespace = null;
exports.getNamespaceOutput = null;
utilities.lazyLoad(exports, ["getNamespace", "getNamespaceOutput"], () => require("./getNamespace"));
exports.getWorkgroup = null;
exports.getWorkgroupOutput = null;
utilities.lazyLoad(exports, ["getWorkgroup", "getWorkgroupOutput"], () => require("./getWorkgroup"));
exports.Namespace = null;
utilities.lazyLoad(exports, ["Namespace"], () => require("./namespace"));
exports.Workgroup = null;
utilities.lazyLoad(exports, ["Workgroup"], () => require("./workgroup"));
// Export enums:
__exportStar(require("../types/enums/redshiftserverless"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:redshiftserverless:Namespace":
                return new exports.Namespace(name, undefined, { urn });
            case "aws-native:redshiftserverless:Workgroup":
                return new exports.Workgroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "redshiftserverless", _module);
