// @pulumi/aws-native/databrew/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schedule = exports.Ruleset = exports.Recipe = exports.Project = exports.Job = exports.getScheduleOutput = exports.getSchedule = exports.getRulesetOutput = exports.getRuleset = exports.getRecipeOutput = exports.getRecipe = exports.getProjectOutput = exports.getProject = exports.getJobOutput = exports.getJob = exports.getDatasetOutput = exports.getDataset = exports.Dataset = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Dataset = null;
utilities.lazyLoad(exports, ["Dataset"], () => require("./dataset"));
exports.getDataset = null;
exports.getDatasetOutput = null;
utilities.lazyLoad(exports, ["getDataset", "getDatasetOutput"], () => require("./getDataset"));
exports.getJob = null;
exports.getJobOutput = null;
utilities.lazyLoad(exports, ["getJob", "getJobOutput"], () => require("./getJob"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getRecipe = null;
exports.getRecipeOutput = null;
utilities.lazyLoad(exports, ["getRecipe", "getRecipeOutput"], () => require("./getRecipe"));
exports.getRuleset = null;
exports.getRulesetOutput = null;
utilities.lazyLoad(exports, ["getRuleset", "getRulesetOutput"], () => require("./getRuleset"));
exports.getSchedule = null;
exports.getScheduleOutput = null;
utilities.lazyLoad(exports, ["getSchedule", "getScheduleOutput"], () => require("./getSchedule"));
exports.Job = null;
utilities.lazyLoad(exports, ["Job"], () => require("./job"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.Recipe = null;
utilities.lazyLoad(exports, ["Recipe"], () => require("./recipe"));
exports.Ruleset = null;
utilities.lazyLoad(exports, ["Ruleset"], () => require("./ruleset"));
exports.Schedule = null;
utilities.lazyLoad(exports, ["Schedule"], () => require("./schedule"));
// Export enums:
__exportStar(require("../types/enums/databrew"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:databrew:Dataset":
                return new exports.Dataset(name, undefined, { urn });
            case "aws-native:databrew:Job":
                return new exports.Job(name, undefined, { urn });
            case "aws-native:databrew:Project":
                return new exports.Project(name, undefined, { urn });
            case "aws-native:databrew:Recipe":
                return new exports.Recipe(name, undefined, { urn });
            case "aws-native:databrew:Ruleset":
                return new exports.Ruleset(name, undefined, { urn });
            case "aws-native:databrew:Schedule":
                return new exports.Schedule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "databrew", _module);
