// @pulumi/aws-native/batch/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchedulingPolicy = exports.JobQueue = exports.JobDefinition = exports.getSchedulingPolicyOutput = exports.getSchedulingPolicy = exports.getJobQueueOutput = exports.getJobQueue = exports.getJobDefinitionOutput = exports.getJobDefinition = exports.getComputeEnvironmentOutput = exports.getComputeEnvironment = exports.ComputeEnvironment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ComputeEnvironment = null;
utilities.lazyLoad(exports, ["ComputeEnvironment"], () => require("./computeEnvironment"));
exports.getComputeEnvironment = null;
exports.getComputeEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getComputeEnvironment", "getComputeEnvironmentOutput"], () => require("./getComputeEnvironment"));
exports.getJobDefinition = null;
exports.getJobDefinitionOutput = null;
utilities.lazyLoad(exports, ["getJobDefinition", "getJobDefinitionOutput"], () => require("./getJobDefinition"));
exports.getJobQueue = null;
exports.getJobQueueOutput = null;
utilities.lazyLoad(exports, ["getJobQueue", "getJobQueueOutput"], () => require("./getJobQueue"));
exports.getSchedulingPolicy = null;
exports.getSchedulingPolicyOutput = null;
utilities.lazyLoad(exports, ["getSchedulingPolicy", "getSchedulingPolicyOutput"], () => require("./getSchedulingPolicy"));
exports.JobDefinition = null;
utilities.lazyLoad(exports, ["JobDefinition"], () => require("./jobDefinition"));
exports.JobQueue = null;
utilities.lazyLoad(exports, ["JobQueue"], () => require("./jobQueue"));
exports.SchedulingPolicy = null;
utilities.lazyLoad(exports, ["SchedulingPolicy"], () => require("./schedulingPolicy"));
// Export enums:
__exportStar(require("../types/enums/batch"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:batch:ComputeEnvironment":
                return new exports.ComputeEnvironment(name, undefined, { urn });
            case "aws-native:batch:JobDefinition":
                return new exports.JobDefinition(name, undefined, { urn });
            case "aws-native:batch:JobQueue":
                return new exports.JobQueue(name, undefined, { urn });
            case "aws-native:batch:SchedulingPolicy":
                return new exports.SchedulingPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "batch", _module);
