// @pulumi/aws-native/appintegrations/dataIntegration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataIntegration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::AppIntegrations::DataIntegration
 */
class DataIntegration extends pulumi.CustomResource {
    /**
     * Create a DataIntegration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kmsKey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kmsKey'");
            }
            if ((!args || args.scheduleConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scheduleConfig'");
            }
            if ((!args || args.sourceURI === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceURI'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["kmsKey"] = args ? args.kmsKey : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["scheduleConfig"] = args ? args.scheduleConfig : undefined;
            resourceInputs["sourceURI"] = args ? args.sourceURI : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["dataIntegrationArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["dataIntegrationArn"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["kmsKey"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["scheduleConfig"] = undefined /*out*/;
            resourceInputs["sourceURI"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DataIntegration.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DataIntegration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DataIntegration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DataIntegration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataIntegration.__pulumiType;
    }
}
exports.DataIntegration = DataIntegration;
/** @internal */
DataIntegration.__pulumiType = 'aws-native:appintegrations:DataIntegration';
