// @pulumi/aws-native/appintegrations/eventIntegration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventIntegration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::AppIntegrations::EventIntegration
 */
class EventIntegration extends pulumi.CustomResource {
    /**
     * Create a EventIntegration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.eventBridgeBus === undefined) && !opts.urn) {
                throw new Error("Missing required property 'eventBridgeBus'");
            }
            if ((!args || args.eventFilter === undefined) && !opts.urn) {
                throw new Error("Missing required property 'eventFilter'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["eventBridgeBus"] = args ? args.eventBridgeBus : undefined;
            resourceInputs["eventFilter"] = args ? args.eventFilter : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["associations"] = undefined /*out*/;
            resourceInputs["eventIntegrationArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["associations"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["eventBridgeBus"] = undefined /*out*/;
            resourceInputs["eventFilter"] = undefined /*out*/;
            resourceInputs["eventIntegrationArn"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EventIntegration.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing EventIntegration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EventIntegration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EventIntegration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EventIntegration.__pulumiType;
    }
}
exports.EventIntegration = EventIntegration;
/** @internal */
EventIntegration.__pulumiType = 'aws-native:appintegrations:EventIntegration';
