// @pulumi/aws-native/appintegrations/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getEventIntegrationOutput = exports.getEventIntegration = exports.getDataIntegrationOutput = exports.getDataIntegration = exports.EventIntegration = exports.DataIntegration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DataIntegration = null;
utilities.lazyLoad(exports, ["DataIntegration"], () => require("./dataIntegration"));
exports.EventIntegration = null;
utilities.lazyLoad(exports, ["EventIntegration"], () => require("./eventIntegration"));
exports.getDataIntegration = null;
exports.getDataIntegrationOutput = null;
utilities.lazyLoad(exports, ["getDataIntegration", "getDataIntegrationOutput"], () => require("./getDataIntegration"));
exports.getEventIntegration = null;
exports.getEventIntegrationOutput = null;
utilities.lazyLoad(exports, ["getEventIntegration", "getEventIntegrationOutput"], () => require("./getEventIntegration"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:appintegrations:DataIntegration":
                return new exports.DataIntegration(name, undefined, { urn });
            case "aws-native:appintegrations:EventIntegration":
                return new exports.EventIntegration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "appintegrations", _module);
