// @pulumi/aws-native/lightsail/loadBalancerTlsCertificate.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoadBalancerTlsCertificate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Lightsail::LoadBalancerTlsCertificate
 */
class LoadBalancerTlsCertificate extends pulumi.CustomResource {
    /**
     * Create a LoadBalancerTlsCertificate resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.certificateDomainName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'certificateDomainName'");
            }
            if ((!args || args.certificateName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'certificateName'");
            }
            if ((!args || args.loadBalancerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'loadBalancerName'");
            }
            resourceInputs["certificateAlternativeNames"] = args ? args.certificateAlternativeNames : undefined;
            resourceInputs["certificateDomainName"] = args ? args.certificateDomainName : undefined;
            resourceInputs["certificateName"] = args ? args.certificateName : undefined;
            resourceInputs["httpsRedirectionEnabled"] = args ? args.httpsRedirectionEnabled : undefined;
            resourceInputs["isAttached"] = args ? args.isAttached : undefined;
            resourceInputs["loadBalancerName"] = args ? args.loadBalancerName : undefined;
            resourceInputs["loadBalancerTlsCertificateArn"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        else {
            resourceInputs["certificateAlternativeNames"] = undefined /*out*/;
            resourceInputs["certificateDomainName"] = undefined /*out*/;
            resourceInputs["certificateName"] = undefined /*out*/;
            resourceInputs["httpsRedirectionEnabled"] = undefined /*out*/;
            resourceInputs["isAttached"] = undefined /*out*/;
            resourceInputs["loadBalancerName"] = undefined /*out*/;
            resourceInputs["loadBalancerTlsCertificateArn"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LoadBalancerTlsCertificate.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing LoadBalancerTlsCertificate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LoadBalancerTlsCertificate(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LoadBalancerTlsCertificate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LoadBalancerTlsCertificate.__pulumiType;
    }
}
exports.LoadBalancerTlsCertificate = LoadBalancerTlsCertificate;
/** @internal */
LoadBalancerTlsCertificate.__pulumiType = 'aws-native:lightsail:LoadBalancerTlsCertificate';
