// @pulumi/aws-native/eventschemas/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schema = exports.RegistryPolicy = exports.Registry = exports.getSchemaOutput = exports.getSchema = exports.getRegistryPolicyOutput = exports.getRegistryPolicy = exports.getRegistryOutput = exports.getRegistry = exports.getDiscovererOutput = exports.getDiscoverer = exports.Discoverer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Discoverer = null;
utilities.lazyLoad(exports, ["Discoverer"], () => require("./discoverer"));
exports.getDiscoverer = null;
exports.getDiscovererOutput = null;
utilities.lazyLoad(exports, ["getDiscoverer", "getDiscovererOutput"], () => require("./getDiscoverer"));
exports.getRegistry = null;
exports.getRegistryOutput = null;
utilities.lazyLoad(exports, ["getRegistry", "getRegistryOutput"], () => require("./getRegistry"));
exports.getRegistryPolicy = null;
exports.getRegistryPolicyOutput = null;
utilities.lazyLoad(exports, ["getRegistryPolicy", "getRegistryPolicyOutput"], () => require("./getRegistryPolicy"));
exports.getSchema = null;
exports.getSchemaOutput = null;
utilities.lazyLoad(exports, ["getSchema", "getSchemaOutput"], () => require("./getSchema"));
exports.Registry = null;
utilities.lazyLoad(exports, ["Registry"], () => require("./registry"));
exports.RegistryPolicy = null;
utilities.lazyLoad(exports, ["RegistryPolicy"], () => require("./registryPolicy"));
exports.Schema = null;
utilities.lazyLoad(exports, ["Schema"], () => require("./schema"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:eventschemas:Discoverer":
                return new exports.Discoverer(name, undefined, { urn });
            case "aws-native:eventschemas:Registry":
                return new exports.Registry(name, undefined, { urn });
            case "aws-native:eventschemas:RegistryPolicy":
                return new exports.RegistryPolicy(name, undefined, { urn });
            case "aws-native:eventschemas:Schema":
                return new exports.Schema(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "eventschemas", _module);
