// @pulumi/aws-native/devopsguru/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceCollection = exports.NotificationChannel = exports.getResourceCollectionOutput = exports.getResourceCollection = exports.getNotificationChannelOutput = exports.getNotificationChannel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getNotificationChannel = null;
exports.getNotificationChannelOutput = null;
utilities.lazyLoad(exports, ["getNotificationChannel", "getNotificationChannelOutput"], () => require("./getNotificationChannel"));
exports.getResourceCollection = null;
exports.getResourceCollectionOutput = null;
utilities.lazyLoad(exports, ["getResourceCollection", "getResourceCollectionOutput"], () => require("./getResourceCollection"));
exports.NotificationChannel = null;
utilities.lazyLoad(exports, ["NotificationChannel"], () => require("./notificationChannel"));
exports.ResourceCollection = null;
utilities.lazyLoad(exports, ["ResourceCollection"], () => require("./resourceCollection"));
// Export enums:
__exportStar(require("../types/enums/devopsguru"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:devopsguru:NotificationChannel":
                return new exports.NotificationChannel(name, undefined, { urn });
            case "aws-native:devopsguru:ResourceCollection":
                return new exports.ResourceCollection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "devopsguru", _module);
