// @pulumi/aws-native/types/enums/redshift/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduledActionState = exports.EventSubscriptionStatus = exports.EventSubscriptionSourceType = exports.EventSubscriptionSeverity = exports.EventSubscriptionEventCategoriesItem = void 0;
exports.EventSubscriptionEventCategoriesItem = {
    Configuration: "configuration",
    Management: "management",
    Monitoring: "monitoring",
    Security: "security",
    Pending: "pending",
};
exports.EventSubscriptionSeverity = {
    Error: "ERROR",
    Info: "INFO",
};
exports.EventSubscriptionSourceType = {
    Cluster: "cluster",
    ClusterParameterGroup: "cluster-parameter-group",
    ClusterSecurityGroup: "cluster-security-group",
    ClusterSnapshot: "cluster-snapshot",
    ScheduledAction: "scheduled-action",
};
exports.EventSubscriptionStatus = {
    Active: "active",
    NoPermission: "no-permission",
    TopicNotExist: "topic-not-exist",
};
exports.ScheduledActionState = {
    Active: "ACTIVE",
    Disabled: "DISABLED",
};
