// @pulumi/aws-native/iotanalytics/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Pipeline = exports.getPipelineOutput = exports.getPipeline = exports.getDatastoreOutput = exports.getDatastore = exports.getDatasetOutput = exports.getDataset = exports.getChannelOutput = exports.getChannel = exports.Datastore = exports.Dataset = exports.Channel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Channel = null;
utilities.lazyLoad(exports, ["Channel"], () => require("./channel"));
exports.Dataset = null;
utilities.lazyLoad(exports, ["Dataset"], () => require("./dataset"));
exports.Datastore = null;
utilities.lazyLoad(exports, ["Datastore"], () => require("./datastore"));
exports.getChannel = null;
exports.getChannelOutput = null;
utilities.lazyLoad(exports, ["getChannel", "getChannelOutput"], () => require("./getChannel"));
exports.getDataset = null;
exports.getDatasetOutput = null;
utilities.lazyLoad(exports, ["getDataset", "getDatasetOutput"], () => require("./getDataset"));
exports.getDatastore = null;
exports.getDatastoreOutput = null;
utilities.lazyLoad(exports, ["getDatastore", "getDatastoreOutput"], () => require("./getDatastore"));
exports.getPipeline = null;
exports.getPipelineOutput = null;
utilities.lazyLoad(exports, ["getPipeline", "getPipelineOutput"], () => require("./getPipeline"));
exports.Pipeline = null;
utilities.lazyLoad(exports, ["Pipeline"], () => require("./pipeline"));
// Export enums:
__exportStar(require("../types/enums/iotanalytics"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:iotanalytics:Channel":
                return new exports.Channel(name, undefined, { urn });
            case "aws-native:iotanalytics:Dataset":
                return new exports.Dataset(name, undefined, { urn });
            case "aws-native:iotanalytics:Datastore":
                return new exports.Datastore(name, undefined, { urn });
            case "aws-native:iotanalytics:Pipeline":
                return new exports.Pipeline(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "iotanalytics", _module);
