// @pulumi/aws-native/directoryservice/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimpleAD = exports.MicrosoftAD = exports.getSimpleADOutput = exports.getSimpleAD = exports.getMicrosoftADOutput = exports.getMicrosoftAD = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getMicrosoftAD = null;
exports.getMicrosoftADOutput = null;
utilities.lazyLoad(exports, ["getMicrosoftAD", "getMicrosoftADOutput"], () => require("./getMicrosoftAD"));
exports.getSimpleAD = null;
exports.getSimpleADOutput = null;
utilities.lazyLoad(exports, ["getSimpleAD", "getSimpleADOutput"], () => require("./getSimpleAD"));
exports.MicrosoftAD = null;
utilities.lazyLoad(exports, ["MicrosoftAD"], () => require("./microsoftAD"));
exports.SimpleAD = null;
utilities.lazyLoad(exports, ["SimpleAD"], () => require("./simpleAD"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:directoryservice:MicrosoftAD":
                return new exports.MicrosoftAD(name, undefined, { urn });
            case "aws-native:directoryservice:SimpleAD":
                return new exports.SimpleAD(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "directoryservice", _module);
