// @pulumi/aws-native/apprunner/observabilityConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The AWS::AppRunner::ObservabilityConfiguration resource  is an AWS App Runner resource type that specifies an App Runner observability configuration
 */
class ObservabilityConfiguration extends pulumi.CustomResource {
    /**
     * Create a ObservabilityConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["observabilityConfigurationName"] = args ? args.observabilityConfigurationName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["traceConfiguration"] = args ? args.traceConfiguration : undefined;
            resourceInputs["latest"] = undefined /*out*/;
            resourceInputs["observabilityConfigurationArn"] = undefined /*out*/;
            resourceInputs["observabilityConfigurationRevision"] = undefined /*out*/;
        }
        else {
            resourceInputs["latest"] = undefined /*out*/;
            resourceInputs["observabilityConfigurationArn"] = undefined /*out*/;
            resourceInputs["observabilityConfigurationName"] = undefined /*out*/;
            resourceInputs["observabilityConfigurationRevision"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["traceConfiguration"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ObservabilityConfiguration.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ObservabilityConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ObservabilityConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ObservabilityConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ObservabilityConfiguration.__pulumiType;
    }
}
exports.ObservabilityConfiguration = ObservabilityConfiguration;
/** @internal */
ObservabilityConfiguration.__pulumiType = 'aws-native:apprunner:ObservabilityConfiguration';
