// @pulumi/aws-native/types/enums/ecs/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskSetScaleUnit = exports.TaskSetLaunchType = exports.TaskSetAwsVpcConfigurationAssignPublicIp = exports.TaskDefinitionEFSVolumeConfigurationTransitEncryption = exports.TaskDefinitionAuthorizationConfigIAM = exports.ServiceSchedulingStrategy = exports.ServicePropagateTags = exports.ServicePlacementStrategyType = exports.ServicePlacementConstraintType = exports.ServiceLaunchType = exports.ServiceDeploymentControllerType = exports.ServiceAwsVpcConfigurationAssignPublicIp = exports.ClusterCapacityProviderAssociationsCapacityProvider = exports.CapacityProviderManagedScalingStatus = exports.CapacityProviderAutoScalingGroupProviderManagedTerminationProtection = void 0;
exports.CapacityProviderAutoScalingGroupProviderManagedTerminationProtection = {
    Disabled: "DISABLED",
    Enabled: "ENABLED",
};
exports.CapacityProviderManagedScalingStatus = {
    Disabled: "DISABLED",
    Enabled: "ENABLED",
};
exports.ClusterCapacityProviderAssociationsCapacityProvider = {
    Fargate: "FARGATE",
    FargateSpot: "FARGATE_SPOT",
};
exports.ServiceAwsVpcConfigurationAssignPublicIp = {
    Disabled: "DISABLED",
    Enabled: "ENABLED",
};
exports.ServiceDeploymentControllerType = {
    CodeDeploy: "CODE_DEPLOY",
    Ecs: "ECS",
    External: "EXTERNAL",
};
exports.ServiceLaunchType = {
    Ec2: "EC2",
    Fargate: "FARGATE",
    External: "EXTERNAL",
};
exports.ServicePlacementConstraintType = {
    DistinctInstance: "distinctInstance",
    MemberOf: "memberOf",
};
exports.ServicePlacementStrategyType = {
    Binpack: "binpack",
    Random: "random",
    Spread: "spread",
};
exports.ServicePropagateTags = {
    Service: "SERVICE",
    TaskDefinition: "TASK_DEFINITION",
};
exports.ServiceSchedulingStrategy = {
    Daemon: "DAEMON",
    Replica: "REPLICA",
};
exports.TaskDefinitionAuthorizationConfigIAM = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.TaskDefinitionEFSVolumeConfigurationTransitEncryption = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.TaskSetAwsVpcConfigurationAssignPublicIp = {
    Disabled: "DISABLED",
    Enabled: "ENABLED",
};
exports.TaskSetLaunchType = {
    Ec2: "EC2",
    Fargate: "FARGATE",
};
exports.TaskSetScaleUnit = {
    Percent: "PERCENT",
};
