// @pulumi/aws-native/ecr/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Repository = exports.ReplicationConfiguration = exports.RegistryPolicy = exports.PullThroughCacheRule = exports.PublicRepository = exports.getRepositoryOutput = exports.getRepository = exports.getReplicationConfigurationOutput = exports.getReplicationConfiguration = exports.getRegistryPolicyOutput = exports.getRegistryPolicy = exports.getPublicRepositoryOutput = exports.getPublicRepository = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getPublicRepository = null;
exports.getPublicRepositoryOutput = null;
utilities.lazyLoad(exports, ["getPublicRepository", "getPublicRepositoryOutput"], () => require("./getPublicRepository"));
exports.getRegistryPolicy = null;
exports.getRegistryPolicyOutput = null;
utilities.lazyLoad(exports, ["getRegistryPolicy", "getRegistryPolicyOutput"], () => require("./getRegistryPolicy"));
exports.getReplicationConfiguration = null;
exports.getReplicationConfigurationOutput = null;
utilities.lazyLoad(exports, ["getReplicationConfiguration", "getReplicationConfigurationOutput"], () => require("./getReplicationConfiguration"));
exports.getRepository = null;
exports.getRepositoryOutput = null;
utilities.lazyLoad(exports, ["getRepository", "getRepositoryOutput"], () => require("./getRepository"));
exports.PublicRepository = null;
utilities.lazyLoad(exports, ["PublicRepository"], () => require("./publicRepository"));
exports.PullThroughCacheRule = null;
utilities.lazyLoad(exports, ["PullThroughCacheRule"], () => require("./pullThroughCacheRule"));
exports.RegistryPolicy = null;
utilities.lazyLoad(exports, ["RegistryPolicy"], () => require("./registryPolicy"));
exports.ReplicationConfiguration = null;
utilities.lazyLoad(exports, ["ReplicationConfiguration"], () => require("./replicationConfiguration"));
exports.Repository = null;
utilities.lazyLoad(exports, ["Repository"], () => require("./repository"));
// Export enums:
__exportStar(require("../types/enums/ecr"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:ecr:PublicRepository":
                return new exports.PublicRepository(name, undefined, { urn });
            case "aws-native:ecr:PullThroughCacheRule":
                return new exports.PullThroughCacheRule(name, undefined, { urn });
            case "aws-native:ecr:RegistryPolicy":
                return new exports.RegistryPolicy(name, undefined, { urn });
            case "aws-native:ecr:ReplicationConfiguration":
                return new exports.ReplicationConfiguration(name, undefined, { urn });
            case "aws-native:ecr:Repository":
                return new exports.Repository(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "ecr", _module);
