// @pulumi/aws-native/types/enums/forecast/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatasetType = exports.DatasetGroupDomain = exports.DatasetDomain = exports.DatasetAttributesItemPropertiesAttributeType = void 0;
exports.DatasetAttributesItemPropertiesAttributeType = {
    String: "string",
    Integer: "integer",
    Float: "float",
    Timestamp: "timestamp",
    Geolocation: "geolocation",
};
exports.DatasetDomain = {
    Retail: "RETAIL",
    Custom: "CUSTOM",
    InventoryPlanning: "INVENTORY_PLANNING",
    Ec2Capacity: "EC2_CAPACITY",
    WorkForce: "WORK_FORCE",
    WebTraffic: "WEB_TRAFFIC",
    Metrics: "METRICS",
};
exports.DatasetGroupDomain = {
    Retail: "RETAIL",
    Custom: "CUSTOM",
    InventoryPlanning: "INVENTORY_PLANNING",
    Ec2Capacity: "EC2_CAPACITY",
    WorkForce: "WORK_FORCE",
    WebTraffic: "WEB_TRAFFIC",
    Metrics: "METRICS",
};
exports.DatasetType = {
    TargetTimeSeries: "TARGET_TIME_SERIES",
    RelatedTimeSeries: "RELATED_TIME_SERIES",
    ItemMetadata: "ITEM_METADATA",
};
