// @pulumi/aws-native/elasticloadbalancing/loadBalancer.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoadBalancer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::ElasticLoadBalancing::LoadBalancer
 *
 * @deprecated LoadBalancer is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class LoadBalancer extends pulumi.CustomResource {
    /**
     * Create a LoadBalancer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated LoadBalancer is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("LoadBalancer is deprecated: LoadBalancer is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.listeners === undefined) && !opts.urn) {
                throw new Error("Missing required property 'listeners'");
            }
            resourceInputs["accessLoggingPolicy"] = args ? args.accessLoggingPolicy : undefined;
            resourceInputs["appCookieStickinessPolicy"] = args ? args.appCookieStickinessPolicy : undefined;
            resourceInputs["availabilityZones"] = args ? args.availabilityZones : undefined;
            resourceInputs["connectionDrainingPolicy"] = args ? args.connectionDrainingPolicy : undefined;
            resourceInputs["connectionSettings"] = args ? args.connectionSettings : undefined;
            resourceInputs["crossZone"] = args ? args.crossZone : undefined;
            resourceInputs["healthCheck"] = args ? args.healthCheck : undefined;
            resourceInputs["instances"] = args ? args.instances : undefined;
            resourceInputs["lBCookieStickinessPolicy"] = args ? args.lBCookieStickinessPolicy : undefined;
            resourceInputs["listeners"] = args ? args.listeners : undefined;
            resourceInputs["loadBalancerName"] = args ? args.loadBalancerName : undefined;
            resourceInputs["policies"] = args ? args.policies : undefined;
            resourceInputs["scheme"] = args ? args.scheme : undefined;
            resourceInputs["securityGroups"] = args ? args.securityGroups : undefined;
            resourceInputs["sourceSecurityGroupGroupName"] = args ? args.sourceSecurityGroupGroupName : undefined;
            resourceInputs["sourceSecurityGroupOwnerAlias"] = args ? args.sourceSecurityGroupOwnerAlias : undefined;
            resourceInputs["subnets"] = args ? args.subnets : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["canonicalHostedZoneName"] = undefined /*out*/;
            resourceInputs["canonicalHostedZoneNameID"] = undefined /*out*/;
            resourceInputs["dNSName"] = undefined /*out*/;
        }
        else {
            resourceInputs["accessLoggingPolicy"] = undefined /*out*/;
            resourceInputs["appCookieStickinessPolicy"] = undefined /*out*/;
            resourceInputs["availabilityZones"] = undefined /*out*/;
            resourceInputs["canonicalHostedZoneName"] = undefined /*out*/;
            resourceInputs["canonicalHostedZoneNameID"] = undefined /*out*/;
            resourceInputs["connectionDrainingPolicy"] = undefined /*out*/;
            resourceInputs["connectionSettings"] = undefined /*out*/;
            resourceInputs["crossZone"] = undefined /*out*/;
            resourceInputs["dNSName"] = undefined /*out*/;
            resourceInputs["healthCheck"] = undefined /*out*/;
            resourceInputs["instances"] = undefined /*out*/;
            resourceInputs["lBCookieStickinessPolicy"] = undefined /*out*/;
            resourceInputs["listeners"] = undefined /*out*/;
            resourceInputs["loadBalancerName"] = undefined /*out*/;
            resourceInputs["policies"] = undefined /*out*/;
            resourceInputs["scheme"] = undefined /*out*/;
            resourceInputs["securityGroups"] = undefined /*out*/;
            resourceInputs["sourceSecurityGroupGroupName"] = undefined /*out*/;
            resourceInputs["sourceSecurityGroupOwnerAlias"] = undefined /*out*/;
            resourceInputs["subnets"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LoadBalancer.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing LoadBalancer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("LoadBalancer is deprecated: LoadBalancer is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new LoadBalancer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LoadBalancer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LoadBalancer.__pulumiType;
    }
}
exports.LoadBalancer = LoadBalancer;
/** @internal */
LoadBalancer.__pulumiType = 'aws-native:elasticloadbalancing:LoadBalancer';
