// @pulumi/aws-native/efs/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MountTarget = exports.getMountTargetOutput = exports.getMountTarget = exports.getFileSystemOutput = exports.getFileSystem = exports.getAccessPointOutput = exports.getAccessPoint = exports.FileSystem = exports.AccessPoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccessPoint = null;
utilities.lazyLoad(exports, ["AccessPoint"], () => require("./accessPoint"));
exports.FileSystem = null;
utilities.lazyLoad(exports, ["FileSystem"], () => require("./fileSystem"));
exports.getAccessPoint = null;
exports.getAccessPointOutput = null;
utilities.lazyLoad(exports, ["getAccessPoint", "getAccessPointOutput"], () => require("./getAccessPoint"));
exports.getFileSystem = null;
exports.getFileSystemOutput = null;
utilities.lazyLoad(exports, ["getFileSystem", "getFileSystemOutput"], () => require("./getFileSystem"));
exports.getMountTarget = null;
exports.getMountTargetOutput = null;
utilities.lazyLoad(exports, ["getMountTarget", "getMountTargetOutput"], () => require("./getMountTarget"));
exports.MountTarget = null;
utilities.lazyLoad(exports, ["MountTarget"], () => require("./mountTarget"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:efs:AccessPoint":
                return new exports.AccessPoint(name, undefined, { urn });
            case "aws-native:efs:FileSystem":
                return new exports.FileSystem(name, undefined, { urn });
            case "aws-native:efs:MountTarget":
                return new exports.MountTarget(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "efs", _module);
