// @pulumi/aws-native/frauddetector/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Variable = exports.Outcome = exports.Label = exports.getVariableOutput = exports.getVariable = exports.getOutcomeOutput = exports.getOutcome = exports.getLabelOutput = exports.getLabel = exports.getEventTypeOutput = exports.getEventType = exports.getEntityTypeOutput = exports.getEntityType = exports.getDetectorOutput = exports.getDetector = exports.EventType = exports.EntityType = exports.Detector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Detector = null;
utilities.lazyLoad(exports, ["Detector"], () => require("./detector"));
exports.EntityType = null;
utilities.lazyLoad(exports, ["EntityType"], () => require("./entityType"));
exports.EventType = null;
utilities.lazyLoad(exports, ["EventType"], () => require("./eventType"));
exports.getDetector = null;
exports.getDetectorOutput = null;
utilities.lazyLoad(exports, ["getDetector", "getDetectorOutput"], () => require("./getDetector"));
exports.getEntityType = null;
exports.getEntityTypeOutput = null;
utilities.lazyLoad(exports, ["getEntityType", "getEntityTypeOutput"], () => require("./getEntityType"));
exports.getEventType = null;
exports.getEventTypeOutput = null;
utilities.lazyLoad(exports, ["getEventType", "getEventTypeOutput"], () => require("./getEventType"));
exports.getLabel = null;
exports.getLabelOutput = null;
utilities.lazyLoad(exports, ["getLabel", "getLabelOutput"], () => require("./getLabel"));
exports.getOutcome = null;
exports.getOutcomeOutput = null;
utilities.lazyLoad(exports, ["getOutcome", "getOutcomeOutput"], () => require("./getOutcome"));
exports.getVariable = null;
exports.getVariableOutput = null;
utilities.lazyLoad(exports, ["getVariable", "getVariableOutput"], () => require("./getVariable"));
exports.Label = null;
utilities.lazyLoad(exports, ["Label"], () => require("./label"));
exports.Outcome = null;
utilities.lazyLoad(exports, ["Outcome"], () => require("./outcome"));
exports.Variable = null;
utilities.lazyLoad(exports, ["Variable"], () => require("./variable"));
// Export enums:
__exportStar(require("../types/enums/frauddetector"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:frauddetector:Detector":
                return new exports.Detector(name, undefined, { urn });
            case "aws-native:frauddetector:EntityType":
                return new exports.EntityType(name, undefined, { urn });
            case "aws-native:frauddetector:EventType":
                return new exports.EventType(name, undefined, { urn });
            case "aws-native:frauddetector:Label":
                return new exports.Label(name, undefined, { urn });
            case "aws-native:frauddetector:Outcome":
                return new exports.Outcome(name, undefined, { urn });
            case "aws-native:frauddetector:Variable":
                return new exports.Variable(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "frauddetector", _module);
