// @pulumi/aws-native/forecast/datasetGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatasetGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a dataset group that holds a collection of related datasets
 */
class DatasetGroup extends pulumi.CustomResource {
    /**
     * Create a DatasetGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.domain === undefined) && !opts.urn) {
                throw new Error("Missing required property 'domain'");
            }
            resourceInputs["datasetArns"] = args ? args.datasetArns : undefined;
            resourceInputs["datasetGroupName"] = args ? args.datasetGroupName : undefined;
            resourceInputs["domain"] = args ? args.domain : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["datasetGroupArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["datasetArns"] = undefined /*out*/;
            resourceInputs["datasetGroupArn"] = undefined /*out*/;
            resourceInputs["datasetGroupName"] = undefined /*out*/;
            resourceInputs["domain"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DatasetGroup.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DatasetGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DatasetGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DatasetGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DatasetGroup.__pulumiType;
    }
}
exports.DatasetGroup = DatasetGroup;
/** @internal */
DatasetGroup.__pulumiType = 'aws-native:forecast:DatasetGroup';
