// @pulumi/aws-native/forecast/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getDatasetGroupOutput = exports.getDatasetGroup = exports.getDatasetOutput = exports.getDataset = exports.DatasetGroup = exports.Dataset = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Dataset = null;
utilities.lazyLoad(exports, ["Dataset"], () => require("./dataset"));
exports.DatasetGroup = null;
utilities.lazyLoad(exports, ["DatasetGroup"], () => require("./datasetGroup"));
exports.getDataset = null;
exports.getDatasetOutput = null;
utilities.lazyLoad(exports, ["getDataset", "getDatasetOutput"], () => require("./getDataset"));
exports.getDatasetGroup = null;
exports.getDatasetGroupOutput = null;
utilities.lazyLoad(exports, ["getDatasetGroup", "getDatasetGroupOutput"], () => require("./getDatasetGroup"));
// Export enums:
__exportStar(require("../types/enums/forecast"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:forecast:Dataset":
                return new exports.Dataset(name, undefined, { urn });
            case "aws-native:forecast:DatasetGroup":
                return new exports.DatasetGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "forecast", _module);
