// @pulumi/aws-native/greengrass/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionDefinitionVersion = exports.SubscriptionDefinition = exports.ResourceDefinitionVersion = exports.ResourceDefinition = exports.LoggerDefinitionVersion = exports.LoggerDefinition = exports.GroupVersion = exports.Group = exports.getSubscriptionDefinitionVersionOutput = exports.getSubscriptionDefinitionVersion = exports.getSubscriptionDefinitionOutput = exports.getSubscriptionDefinition = exports.getResourceDefinitionVersionOutput = exports.getResourceDefinitionVersion = exports.getResourceDefinitionOutput = exports.getResourceDefinition = exports.getLoggerDefinitionVersionOutput = exports.getLoggerDefinitionVersion = exports.getLoggerDefinitionOutput = exports.getLoggerDefinition = exports.getGroupVersionOutput = exports.getGroupVersion = exports.getGroupOutput = exports.getGroup = exports.getFunctionDefinitionVersionOutput = exports.getFunctionDefinitionVersion = exports.getFunctionDefinitionOutput = exports.getFunctionDefinition = exports.getDeviceDefinitionVersionOutput = exports.getDeviceDefinitionVersion = exports.getDeviceDefinitionOutput = exports.getDeviceDefinition = exports.getCoreDefinitionVersionOutput = exports.getCoreDefinitionVersion = exports.getCoreDefinitionOutput = exports.getCoreDefinition = exports.getConnectorDefinitionVersionOutput = exports.getConnectorDefinitionVersion = exports.getConnectorDefinitionOutput = exports.getConnectorDefinition = exports.FunctionDefinitionVersion = exports.FunctionDefinition = exports.DeviceDefinitionVersion = exports.DeviceDefinition = exports.CoreDefinitionVersion = exports.CoreDefinition = exports.ConnectorDefinitionVersion = exports.ConnectorDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ConnectorDefinition = null;
utilities.lazyLoad(exports, ["ConnectorDefinition"], () => require("./connectorDefinition"));
exports.ConnectorDefinitionVersion = null;
utilities.lazyLoad(exports, ["ConnectorDefinitionVersion"], () => require("./connectorDefinitionVersion"));
exports.CoreDefinition = null;
utilities.lazyLoad(exports, ["CoreDefinition"], () => require("./coreDefinition"));
exports.CoreDefinitionVersion = null;
utilities.lazyLoad(exports, ["CoreDefinitionVersion"], () => require("./coreDefinitionVersion"));
exports.DeviceDefinition = null;
utilities.lazyLoad(exports, ["DeviceDefinition"], () => require("./deviceDefinition"));
exports.DeviceDefinitionVersion = null;
utilities.lazyLoad(exports, ["DeviceDefinitionVersion"], () => require("./deviceDefinitionVersion"));
exports.FunctionDefinition = null;
utilities.lazyLoad(exports, ["FunctionDefinition"], () => require("./functionDefinition"));
exports.FunctionDefinitionVersion = null;
utilities.lazyLoad(exports, ["FunctionDefinitionVersion"], () => require("./functionDefinitionVersion"));
exports.getConnectorDefinition = null;
exports.getConnectorDefinitionOutput = null;
utilities.lazyLoad(exports, ["getConnectorDefinition", "getConnectorDefinitionOutput"], () => require("./getConnectorDefinition"));
exports.getConnectorDefinitionVersion = null;
exports.getConnectorDefinitionVersionOutput = null;
utilities.lazyLoad(exports, ["getConnectorDefinitionVersion", "getConnectorDefinitionVersionOutput"], () => require("./getConnectorDefinitionVersion"));
exports.getCoreDefinition = null;
exports.getCoreDefinitionOutput = null;
utilities.lazyLoad(exports, ["getCoreDefinition", "getCoreDefinitionOutput"], () => require("./getCoreDefinition"));
exports.getCoreDefinitionVersion = null;
exports.getCoreDefinitionVersionOutput = null;
utilities.lazyLoad(exports, ["getCoreDefinitionVersion", "getCoreDefinitionVersionOutput"], () => require("./getCoreDefinitionVersion"));
exports.getDeviceDefinition = null;
exports.getDeviceDefinitionOutput = null;
utilities.lazyLoad(exports, ["getDeviceDefinition", "getDeviceDefinitionOutput"], () => require("./getDeviceDefinition"));
exports.getDeviceDefinitionVersion = null;
exports.getDeviceDefinitionVersionOutput = null;
utilities.lazyLoad(exports, ["getDeviceDefinitionVersion", "getDeviceDefinitionVersionOutput"], () => require("./getDeviceDefinitionVersion"));
exports.getFunctionDefinition = null;
exports.getFunctionDefinitionOutput = null;
utilities.lazyLoad(exports, ["getFunctionDefinition", "getFunctionDefinitionOutput"], () => require("./getFunctionDefinition"));
exports.getFunctionDefinitionVersion = null;
exports.getFunctionDefinitionVersionOutput = null;
utilities.lazyLoad(exports, ["getFunctionDefinitionVersion", "getFunctionDefinitionVersionOutput"], () => require("./getFunctionDefinitionVersion"));
exports.getGroup = null;
exports.getGroupOutput = null;
utilities.lazyLoad(exports, ["getGroup", "getGroupOutput"], () => require("./getGroup"));
exports.getGroupVersion = null;
exports.getGroupVersionOutput = null;
utilities.lazyLoad(exports, ["getGroupVersion", "getGroupVersionOutput"], () => require("./getGroupVersion"));
exports.getLoggerDefinition = null;
exports.getLoggerDefinitionOutput = null;
utilities.lazyLoad(exports, ["getLoggerDefinition", "getLoggerDefinitionOutput"], () => require("./getLoggerDefinition"));
exports.getLoggerDefinitionVersion = null;
exports.getLoggerDefinitionVersionOutput = null;
utilities.lazyLoad(exports, ["getLoggerDefinitionVersion", "getLoggerDefinitionVersionOutput"], () => require("./getLoggerDefinitionVersion"));
exports.getResourceDefinition = null;
exports.getResourceDefinitionOutput = null;
utilities.lazyLoad(exports, ["getResourceDefinition", "getResourceDefinitionOutput"], () => require("./getResourceDefinition"));
exports.getResourceDefinitionVersion = null;
exports.getResourceDefinitionVersionOutput = null;
utilities.lazyLoad(exports, ["getResourceDefinitionVersion", "getResourceDefinitionVersionOutput"], () => require("./getResourceDefinitionVersion"));
exports.getSubscriptionDefinition = null;
exports.getSubscriptionDefinitionOutput = null;
utilities.lazyLoad(exports, ["getSubscriptionDefinition", "getSubscriptionDefinitionOutput"], () => require("./getSubscriptionDefinition"));
exports.getSubscriptionDefinitionVersion = null;
exports.getSubscriptionDefinitionVersionOutput = null;
utilities.lazyLoad(exports, ["getSubscriptionDefinitionVersion", "getSubscriptionDefinitionVersionOutput"], () => require("./getSubscriptionDefinitionVersion"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.GroupVersion = null;
utilities.lazyLoad(exports, ["GroupVersion"], () => require("./groupVersion"));
exports.LoggerDefinition = null;
utilities.lazyLoad(exports, ["LoggerDefinition"], () => require("./loggerDefinition"));
exports.LoggerDefinitionVersion = null;
utilities.lazyLoad(exports, ["LoggerDefinitionVersion"], () => require("./loggerDefinitionVersion"));
exports.ResourceDefinition = null;
utilities.lazyLoad(exports, ["ResourceDefinition"], () => require("./resourceDefinition"));
exports.ResourceDefinitionVersion = null;
utilities.lazyLoad(exports, ["ResourceDefinitionVersion"], () => require("./resourceDefinitionVersion"));
exports.SubscriptionDefinition = null;
utilities.lazyLoad(exports, ["SubscriptionDefinition"], () => require("./subscriptionDefinition"));
exports.SubscriptionDefinitionVersion = null;
utilities.lazyLoad(exports, ["SubscriptionDefinitionVersion"], () => require("./subscriptionDefinitionVersion"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:greengrass:ConnectorDefinition":
                return new exports.ConnectorDefinition(name, undefined, { urn });
            case "aws-native:greengrass:ConnectorDefinitionVersion":
                return new exports.ConnectorDefinitionVersion(name, undefined, { urn });
            case "aws-native:greengrass:CoreDefinition":
                return new exports.CoreDefinition(name, undefined, { urn });
            case "aws-native:greengrass:CoreDefinitionVersion":
                return new exports.CoreDefinitionVersion(name, undefined, { urn });
            case "aws-native:greengrass:DeviceDefinition":
                return new exports.DeviceDefinition(name, undefined, { urn });
            case "aws-native:greengrass:DeviceDefinitionVersion":
                return new exports.DeviceDefinitionVersion(name, undefined, { urn });
            case "aws-native:greengrass:FunctionDefinition":
                return new exports.FunctionDefinition(name, undefined, { urn });
            case "aws-native:greengrass:FunctionDefinitionVersion":
                return new exports.FunctionDefinitionVersion(name, undefined, { urn });
            case "aws-native:greengrass:Group":
                return new exports.Group(name, undefined, { urn });
            case "aws-native:greengrass:GroupVersion":
                return new exports.GroupVersion(name, undefined, { urn });
            case "aws-native:greengrass:LoggerDefinition":
                return new exports.LoggerDefinition(name, undefined, { urn });
            case "aws-native:greengrass:LoggerDefinitionVersion":
                return new exports.LoggerDefinitionVersion(name, undefined, { urn });
            case "aws-native:greengrass:ResourceDefinition":
                return new exports.ResourceDefinition(name, undefined, { urn });
            case "aws-native:greengrass:ResourceDefinitionVersion":
                return new exports.ResourceDefinitionVersion(name, undefined, { urn });
            case "aws-native:greengrass:SubscriptionDefinition":
                return new exports.SubscriptionDefinition(name, undefined, { urn });
            case "aws-native:greengrass:SubscriptionDefinitionVersion":
                return new exports.SubscriptionDefinitionVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "greengrass", _module);
