// @pulumi/aws-native/configuration/aggregationAuthorization.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregationAuthorization = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Config::AggregationAuthorization
 */
class AggregationAuthorization extends pulumi.CustomResource {
    /**
     * Create a AggregationAuthorization resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.authorizedAccountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'authorizedAccountId'");
            }
            if ((!args || args.authorizedAwsRegion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'authorizedAwsRegion'");
            }
            resourceInputs["authorizedAccountId"] = args ? args.authorizedAccountId : undefined;
            resourceInputs["authorizedAwsRegion"] = args ? args.authorizedAwsRegion : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["aggregationAuthorizationArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["aggregationAuthorizationArn"] = undefined /*out*/;
            resourceInputs["authorizedAccountId"] = undefined /*out*/;
            resourceInputs["authorizedAwsRegion"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AggregationAuthorization.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing AggregationAuthorization resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AggregationAuthorization(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AggregationAuthorization.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AggregationAuthorization.__pulumiType;
    }
}
exports.AggregationAuthorization = AggregationAuthorization;
/** @internal */
AggregationAuthorization.__pulumiType = 'aws-native:configuration:AggregationAuthorization';
