// @pulumi/aws-native/configuration/storedQuery.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredQuery = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Config::StoredQuery
 */
class StoredQuery extends pulumi.CustomResource {
    /**
     * Create a StoredQuery resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.queryExpression === undefined) && !opts.urn) {
                throw new Error("Missing required property 'queryExpression'");
            }
            if ((!args || args.queryName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'queryName'");
            }
            resourceInputs["queryDescription"] = args ? args.queryDescription : undefined;
            resourceInputs["queryExpression"] = args ? args.queryExpression : undefined;
            resourceInputs["queryName"] = args ? args.queryName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["queryArn"] = undefined /*out*/;
            resourceInputs["queryId"] = undefined /*out*/;
        }
        else {
            resourceInputs["queryArn"] = undefined /*out*/;
            resourceInputs["queryDescription"] = undefined /*out*/;
            resourceInputs["queryExpression"] = undefined /*out*/;
            resourceInputs["queryId"] = undefined /*out*/;
            resourceInputs["queryName"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(StoredQuery.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing StoredQuery resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StoredQuery(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StoredQuery.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StoredQuery.__pulumiType;
    }
}
exports.StoredQuery = StoredQuery;
/** @internal */
StoredQuery.__pulumiType = 'aws-native:configuration:StoredQuery';
