// @pulumi/aws-native/route53recoveryreadiness/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceSet = exports.RecoveryGroup = exports.ReadinessCheck = exports.getResourceSetOutput = exports.getResourceSet = exports.getRecoveryGroupOutput = exports.getRecoveryGroup = exports.getReadinessCheckOutput = exports.getReadinessCheck = exports.getCellOutput = exports.getCell = exports.Cell = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Cell = null;
utilities.lazyLoad(exports, ["Cell"], () => require("./cell"));
exports.getCell = null;
exports.getCellOutput = null;
utilities.lazyLoad(exports, ["getCell", "getCellOutput"], () => require("./getCell"));
exports.getReadinessCheck = null;
exports.getReadinessCheckOutput = null;
utilities.lazyLoad(exports, ["getReadinessCheck", "getReadinessCheckOutput"], () => require("./getReadinessCheck"));
exports.getRecoveryGroup = null;
exports.getRecoveryGroupOutput = null;
utilities.lazyLoad(exports, ["getRecoveryGroup", "getRecoveryGroupOutput"], () => require("./getRecoveryGroup"));
exports.getResourceSet = null;
exports.getResourceSetOutput = null;
utilities.lazyLoad(exports, ["getResourceSet", "getResourceSetOutput"], () => require("./getResourceSet"));
exports.ReadinessCheck = null;
utilities.lazyLoad(exports, ["ReadinessCheck"], () => require("./readinessCheck"));
exports.RecoveryGroup = null;
utilities.lazyLoad(exports, ["RecoveryGroup"], () => require("./recoveryGroup"));
exports.ResourceSet = null;
utilities.lazyLoad(exports, ["ResourceSet"], () => require("./resourceSet"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:route53recoveryreadiness:Cell":
                return new exports.Cell(name, undefined, { urn });
            case "aws-native:route53recoveryreadiness:ReadinessCheck":
                return new exports.ReadinessCheck(name, undefined, { urn });
            case "aws-native:route53recoveryreadiness:RecoveryGroup":
                return new exports.RecoveryGroup(name, undefined, { urn });
            case "aws-native:route53recoveryreadiness:ResourceSet":
                return new exports.ResourceSet(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "route53recoveryreadiness", _module);
