// @pulumi/aws-native/types/enums/imagebuilder/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.InfrastructureConfigurationInstanceMetadataOptionsHttpTokens = exports.ImageRecipeEbsInstanceBlockDeviceSpecificationVolumeType = exports.ImagePipelineStatus = exports.ImagePipelineSchedulePipelineExecutionStartCondition = exports.DistributionConfigurationTargetContainerRepositoryService = exports.ContainerRecipeTargetContainerRepositoryService = exports.ContainerRecipePlatformOverride = exports.ContainerRecipeEbsInstanceBlockDeviceSpecificationVolumeType = exports.ContainerRecipeContainerType = exports.ComponentType = exports.ComponentPlatform = void 0;
exports.ComponentPlatform = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.ComponentType = {
    Build: "BUILD",
    Test: "TEST",
};
exports.ContainerRecipeContainerType = {
    Docker: "DOCKER",
};
exports.ContainerRecipeEbsInstanceBlockDeviceSpecificationVolumeType = {
    Standard: "standard",
    Io1: "io1",
    Io2: "io2",
    Gp2: "gp2",
    Gp3: "gp3",
    Sc1: "sc1",
    St1: "st1",
};
exports.ContainerRecipePlatformOverride = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.ContainerRecipeTargetContainerRepositoryService = {
    Ecr: "ECR",
};
exports.DistributionConfigurationTargetContainerRepositoryService = {
    Ecr: "ECR",
};
exports.ImagePipelineSchedulePipelineExecutionStartCondition = {
    ExpressionMatchOnly: "EXPRESSION_MATCH_ONLY",
    ExpressionMatchAndDependencyUpdatesAvailable: "EXPRESSION_MATCH_AND_DEPENDENCY_UPDATES_AVAILABLE",
};
exports.ImagePipelineStatus = {
    Disabled: "DISABLED",
    Enabled: "ENABLED",
};
exports.ImageRecipeEbsInstanceBlockDeviceSpecificationVolumeType = {
    Standard: "standard",
    Io1: "io1",
    Io2: "io2",
    Gp2: "gp2",
    Gp3: "gp3",
    Sc1: "sc1",
    St1: "st1",
};
exports.InfrastructureConfigurationInstanceMetadataOptionsHttpTokens = {
    Required: "required",
    Optional: "optional",
};
