// @pulumi/aws-native/sagemaker/endpointConfig.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::SageMaker::EndpointConfig
 *
 * @deprecated EndpointConfig is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class EndpointConfig extends pulumi.CustomResource {
    /**
     * Create a EndpointConfig resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated EndpointConfig is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("EndpointConfig is deprecated: EndpointConfig is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.productionVariants === undefined) && !opts.urn) {
                throw new Error("Missing required property 'productionVariants'");
            }
            resourceInputs["asyncInferenceConfig"] = args ? args.asyncInferenceConfig : undefined;
            resourceInputs["dataCaptureConfig"] = args ? args.dataCaptureConfig : undefined;
            resourceInputs["endpointConfigName"] = args ? args.endpointConfigName : undefined;
            resourceInputs["explainerConfig"] = args ? args.explainerConfig : undefined;
            resourceInputs["kmsKeyId"] = args ? args.kmsKeyId : undefined;
            resourceInputs["productionVariants"] = args ? args.productionVariants : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
        }
        else {
            resourceInputs["asyncInferenceConfig"] = undefined /*out*/;
            resourceInputs["dataCaptureConfig"] = undefined /*out*/;
            resourceInputs["endpointConfigName"] = undefined /*out*/;
            resourceInputs["explainerConfig"] = undefined /*out*/;
            resourceInputs["kmsKeyId"] = undefined /*out*/;
            resourceInputs["productionVariants"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EndpointConfig.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing EndpointConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("EndpointConfig is deprecated: EndpointConfig is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new EndpointConfig(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EndpointConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EndpointConfig.__pulumiType;
    }
}
exports.EndpointConfig = EndpointConfig;
/** @internal */
EndpointConfig.__pulumiType = 'aws-native:sagemaker:EndpointConfig';
