// @pulumi/aws-native/sagemaker/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPipelineOutput = exports.getPipeline = exports.getNotebookInstanceLifecycleConfigOutput = exports.getNotebookInstanceLifecycleConfig = exports.getNotebookInstanceOutput = exports.getNotebookInstance = exports.getMonitoringScheduleOutput = exports.getMonitoringSchedule = exports.getModelQualityJobDefinitionOutput = exports.getModelQualityJobDefinition = exports.getModelPackageGroupOutput = exports.getModelPackageGroup = exports.getModelPackageOutput = exports.getModelPackage = exports.getModelExplainabilityJobDefinitionOutput = exports.getModelExplainabilityJobDefinition = exports.getModelBiasJobDefinitionOutput = exports.getModelBiasJobDefinition = exports.getModelOutput = exports.getModel = exports.getImageVersionOutput = exports.getImageVersion = exports.getImageOutput = exports.getImage = exports.getEndpointConfigOutput = exports.getEndpointConfig = exports.getEndpointOutput = exports.getEndpoint = exports.getDomainOutput = exports.getDomain = exports.getDeviceFleetOutput = exports.getDeviceFleet = exports.getDataQualityJobDefinitionOutput = exports.getDataQualityJobDefinition = exports.getCodeRepositoryOutput = exports.getCodeRepository = exports.getAppImageConfigOutput = exports.getAppImageConfig = exports.getAppOutput = exports.getApp = exports.FeatureGroup = exports.EndpointConfig = exports.Endpoint = exports.Domain = exports.DeviceFleet = exports.Device = exports.DataQualityJobDefinition = exports.CodeRepository = exports.AppImageConfig = exports.App = void 0;
exports.Workteam = exports.UserProfile = exports.Project = exports.Pipeline = exports.NotebookInstanceLifecycleConfig = exports.NotebookInstance = exports.MonitoringSchedule = exports.ModelQualityJobDefinition = exports.ModelPackageGroup = exports.ModelPackage = exports.ModelExplainabilityJobDefinition = exports.ModelBiasJobDefinition = exports.Model = exports.ImageVersion = exports.Image = exports.getWorkteamOutput = exports.getWorkteam = exports.getUserProfileOutput = exports.getUserProfile = exports.getProjectOutput = exports.getProject = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.App = null;
utilities.lazyLoad(exports, ["App"], () => require("./app"));
exports.AppImageConfig = null;
utilities.lazyLoad(exports, ["AppImageConfig"], () => require("./appImageConfig"));
exports.CodeRepository = null;
utilities.lazyLoad(exports, ["CodeRepository"], () => require("./codeRepository"));
exports.DataQualityJobDefinition = null;
utilities.lazyLoad(exports, ["DataQualityJobDefinition"], () => require("./dataQualityJobDefinition"));
exports.Device = null;
utilities.lazyLoad(exports, ["Device"], () => require("./device"));
exports.DeviceFleet = null;
utilities.lazyLoad(exports, ["DeviceFleet"], () => require("./deviceFleet"));
exports.Domain = null;
utilities.lazyLoad(exports, ["Domain"], () => require("./domain"));
exports.Endpoint = null;
utilities.lazyLoad(exports, ["Endpoint"], () => require("./endpoint"));
exports.EndpointConfig = null;
utilities.lazyLoad(exports, ["EndpointConfig"], () => require("./endpointConfig"));
exports.FeatureGroup = null;
utilities.lazyLoad(exports, ["FeatureGroup"], () => require("./featureGroup"));
exports.getApp = null;
exports.getAppOutput = null;
utilities.lazyLoad(exports, ["getApp", "getAppOutput"], () => require("./getApp"));
exports.getAppImageConfig = null;
exports.getAppImageConfigOutput = null;
utilities.lazyLoad(exports, ["getAppImageConfig", "getAppImageConfigOutput"], () => require("./getAppImageConfig"));
exports.getCodeRepository = null;
exports.getCodeRepositoryOutput = null;
utilities.lazyLoad(exports, ["getCodeRepository", "getCodeRepositoryOutput"], () => require("./getCodeRepository"));
exports.getDataQualityJobDefinition = null;
exports.getDataQualityJobDefinitionOutput = null;
utilities.lazyLoad(exports, ["getDataQualityJobDefinition", "getDataQualityJobDefinitionOutput"], () => require("./getDataQualityJobDefinition"));
exports.getDeviceFleet = null;
exports.getDeviceFleetOutput = null;
utilities.lazyLoad(exports, ["getDeviceFleet", "getDeviceFleetOutput"], () => require("./getDeviceFleet"));
exports.getDomain = null;
exports.getDomainOutput = null;
utilities.lazyLoad(exports, ["getDomain", "getDomainOutput"], () => require("./getDomain"));
exports.getEndpoint = null;
exports.getEndpointOutput = null;
utilities.lazyLoad(exports, ["getEndpoint", "getEndpointOutput"], () => require("./getEndpoint"));
exports.getEndpointConfig = null;
exports.getEndpointConfigOutput = null;
utilities.lazyLoad(exports, ["getEndpointConfig", "getEndpointConfigOutput"], () => require("./getEndpointConfig"));
exports.getImage = null;
exports.getImageOutput = null;
utilities.lazyLoad(exports, ["getImage", "getImageOutput"], () => require("./getImage"));
exports.getImageVersion = null;
exports.getImageVersionOutput = null;
utilities.lazyLoad(exports, ["getImageVersion", "getImageVersionOutput"], () => require("./getImageVersion"));
exports.getModel = null;
exports.getModelOutput = null;
utilities.lazyLoad(exports, ["getModel", "getModelOutput"], () => require("./getModel"));
exports.getModelBiasJobDefinition = null;
exports.getModelBiasJobDefinitionOutput = null;
utilities.lazyLoad(exports, ["getModelBiasJobDefinition", "getModelBiasJobDefinitionOutput"], () => require("./getModelBiasJobDefinition"));
exports.getModelExplainabilityJobDefinition = null;
exports.getModelExplainabilityJobDefinitionOutput = null;
utilities.lazyLoad(exports, ["getModelExplainabilityJobDefinition", "getModelExplainabilityJobDefinitionOutput"], () => require("./getModelExplainabilityJobDefinition"));
exports.getModelPackage = null;
exports.getModelPackageOutput = null;
utilities.lazyLoad(exports, ["getModelPackage", "getModelPackageOutput"], () => require("./getModelPackage"));
exports.getModelPackageGroup = null;
exports.getModelPackageGroupOutput = null;
utilities.lazyLoad(exports, ["getModelPackageGroup", "getModelPackageGroupOutput"], () => require("./getModelPackageGroup"));
exports.getModelQualityJobDefinition = null;
exports.getModelQualityJobDefinitionOutput = null;
utilities.lazyLoad(exports, ["getModelQualityJobDefinition", "getModelQualityJobDefinitionOutput"], () => require("./getModelQualityJobDefinition"));
exports.getMonitoringSchedule = null;
exports.getMonitoringScheduleOutput = null;
utilities.lazyLoad(exports, ["getMonitoringSchedule", "getMonitoringScheduleOutput"], () => require("./getMonitoringSchedule"));
exports.getNotebookInstance = null;
exports.getNotebookInstanceOutput = null;
utilities.lazyLoad(exports, ["getNotebookInstance", "getNotebookInstanceOutput"], () => require("./getNotebookInstance"));
exports.getNotebookInstanceLifecycleConfig = null;
exports.getNotebookInstanceLifecycleConfigOutput = null;
utilities.lazyLoad(exports, ["getNotebookInstanceLifecycleConfig", "getNotebookInstanceLifecycleConfigOutput"], () => require("./getNotebookInstanceLifecycleConfig"));
exports.getPipeline = null;
exports.getPipelineOutput = null;
utilities.lazyLoad(exports, ["getPipeline", "getPipelineOutput"], () => require("./getPipeline"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getUserProfile = null;
exports.getUserProfileOutput = null;
utilities.lazyLoad(exports, ["getUserProfile", "getUserProfileOutput"], () => require("./getUserProfile"));
exports.getWorkteam = null;
exports.getWorkteamOutput = null;
utilities.lazyLoad(exports, ["getWorkteam", "getWorkteamOutput"], () => require("./getWorkteam"));
exports.Image = null;
utilities.lazyLoad(exports, ["Image"], () => require("./image"));
exports.ImageVersion = null;
utilities.lazyLoad(exports, ["ImageVersion"], () => require("./imageVersion"));
exports.Model = null;
utilities.lazyLoad(exports, ["Model"], () => require("./model"));
exports.ModelBiasJobDefinition = null;
utilities.lazyLoad(exports, ["ModelBiasJobDefinition"], () => require("./modelBiasJobDefinition"));
exports.ModelExplainabilityJobDefinition = null;
utilities.lazyLoad(exports, ["ModelExplainabilityJobDefinition"], () => require("./modelExplainabilityJobDefinition"));
exports.ModelPackage = null;
utilities.lazyLoad(exports, ["ModelPackage"], () => require("./modelPackage"));
exports.ModelPackageGroup = null;
utilities.lazyLoad(exports, ["ModelPackageGroup"], () => require("./modelPackageGroup"));
exports.ModelQualityJobDefinition = null;
utilities.lazyLoad(exports, ["ModelQualityJobDefinition"], () => require("./modelQualityJobDefinition"));
exports.MonitoringSchedule = null;
utilities.lazyLoad(exports, ["MonitoringSchedule"], () => require("./monitoringSchedule"));
exports.NotebookInstance = null;
utilities.lazyLoad(exports, ["NotebookInstance"], () => require("./notebookInstance"));
exports.NotebookInstanceLifecycleConfig = null;
utilities.lazyLoad(exports, ["NotebookInstanceLifecycleConfig"], () => require("./notebookInstanceLifecycleConfig"));
exports.Pipeline = null;
utilities.lazyLoad(exports, ["Pipeline"], () => require("./pipeline"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.UserProfile = null;
utilities.lazyLoad(exports, ["UserProfile"], () => require("./userProfile"));
exports.Workteam = null;
utilities.lazyLoad(exports, ["Workteam"], () => require("./workteam"));
// Export enums:
__exportStar(require("../types/enums/sagemaker"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:sagemaker:App":
                return new exports.App(name, undefined, { urn });
            case "aws-native:sagemaker:AppImageConfig":
                return new exports.AppImageConfig(name, undefined, { urn });
            case "aws-native:sagemaker:CodeRepository":
                return new exports.CodeRepository(name, undefined, { urn });
            case "aws-native:sagemaker:DataQualityJobDefinition":
                return new exports.DataQualityJobDefinition(name, undefined, { urn });
            case "aws-native:sagemaker:Device":
                return new exports.Device(name, undefined, { urn });
            case "aws-native:sagemaker:DeviceFleet":
                return new exports.DeviceFleet(name, undefined, { urn });
            case "aws-native:sagemaker:Domain":
                return new exports.Domain(name, undefined, { urn });
            case "aws-native:sagemaker:Endpoint":
                return new exports.Endpoint(name, undefined, { urn });
            case "aws-native:sagemaker:EndpointConfig":
                return new exports.EndpointConfig(name, undefined, { urn });
            case "aws-native:sagemaker:FeatureGroup":
                return new exports.FeatureGroup(name, undefined, { urn });
            case "aws-native:sagemaker:Image":
                return new exports.Image(name, undefined, { urn });
            case "aws-native:sagemaker:ImageVersion":
                return new exports.ImageVersion(name, undefined, { urn });
            case "aws-native:sagemaker:Model":
                return new exports.Model(name, undefined, { urn });
            case "aws-native:sagemaker:ModelBiasJobDefinition":
                return new exports.ModelBiasJobDefinition(name, undefined, { urn });
            case "aws-native:sagemaker:ModelExplainabilityJobDefinition":
                return new exports.ModelExplainabilityJobDefinition(name, undefined, { urn });
            case "aws-native:sagemaker:ModelPackage":
                return new exports.ModelPackage(name, undefined, { urn });
            case "aws-native:sagemaker:ModelPackageGroup":
                return new exports.ModelPackageGroup(name, undefined, { urn });
            case "aws-native:sagemaker:ModelQualityJobDefinition":
                return new exports.ModelQualityJobDefinition(name, undefined, { urn });
            case "aws-native:sagemaker:MonitoringSchedule":
                return new exports.MonitoringSchedule(name, undefined, { urn });
            case "aws-native:sagemaker:NotebookInstance":
                return new exports.NotebookInstance(name, undefined, { urn });
            case "aws-native:sagemaker:NotebookInstanceLifecycleConfig":
                return new exports.NotebookInstanceLifecycleConfig(name, undefined, { urn });
            case "aws-native:sagemaker:Pipeline":
                return new exports.Pipeline(name, undefined, { urn });
            case "aws-native:sagemaker:Project":
                return new exports.Project(name, undefined, { urn });
            case "aws-native:sagemaker:UserProfile":
                return new exports.UserProfile(name, undefined, { urn });
            case "aws-native:sagemaker:Workteam":
                return new exports.Workteam(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "sagemaker", _module);
