// @pulumi/aws-native/sagemaker/modelQualityJobDefinition.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelQualityJobDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::SageMaker::ModelQualityJobDefinition
 */
class ModelQualityJobDefinition extends pulumi.CustomResource {
    /**
     * Create a ModelQualityJobDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.jobResources === undefined) && !opts.urn) {
                throw new Error("Missing required property 'jobResources'");
            }
            if ((!args || args.modelQualityAppSpecification === undefined) && !opts.urn) {
                throw new Error("Missing required property 'modelQualityAppSpecification'");
            }
            if ((!args || args.modelQualityJobInput === undefined) && !opts.urn) {
                throw new Error("Missing required property 'modelQualityJobInput'");
            }
            if ((!args || args.modelQualityJobOutputConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'modelQualityJobOutputConfig'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["endpointName"] = args ? args.endpointName : undefined;
            resourceInputs["jobDefinitionName"] = args ? args.jobDefinitionName : undefined;
            resourceInputs["jobResources"] = args ? args.jobResources : undefined;
            resourceInputs["modelQualityAppSpecification"] = args ? args.modelQualityAppSpecification : undefined;
            resourceInputs["modelQualityBaselineConfig"] = args ? args.modelQualityBaselineConfig : undefined;
            resourceInputs["modelQualityJobInput"] = args ? args.modelQualityJobInput : undefined;
            resourceInputs["modelQualityJobOutputConfig"] = args ? args.modelQualityJobOutputConfig : undefined;
            resourceInputs["networkConfig"] = args ? args.networkConfig : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
            resourceInputs["stoppingCondition"] = args ? args.stoppingCondition : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["jobDefinitionArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["endpointName"] = undefined /*out*/;
            resourceInputs["jobDefinitionArn"] = undefined /*out*/;
            resourceInputs["jobDefinitionName"] = undefined /*out*/;
            resourceInputs["jobResources"] = undefined /*out*/;
            resourceInputs["modelQualityAppSpecification"] = undefined /*out*/;
            resourceInputs["modelQualityBaselineConfig"] = undefined /*out*/;
            resourceInputs["modelQualityJobInput"] = undefined /*out*/;
            resourceInputs["modelQualityJobOutputConfig"] = undefined /*out*/;
            resourceInputs["networkConfig"] = undefined /*out*/;
            resourceInputs["roleArn"] = undefined /*out*/;
            resourceInputs["stoppingCondition"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ModelQualityJobDefinition.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ModelQualityJobDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ModelQualityJobDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ModelQualityJobDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ModelQualityJobDefinition.__pulumiType;
    }
}
exports.ModelQualityJobDefinition = ModelQualityJobDefinition;
/** @internal */
ModelQualityJobDefinition.__pulumiType = 'aws-native:sagemaker:ModelQualityJobDefinition';
