// @pulumi/aws-native/sagemaker/monitoringSchedule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitoringSchedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::SageMaker::MonitoringSchedule
 */
class MonitoringSchedule extends pulumi.CustomResource {
    /**
     * Create a MonitoringSchedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.monitoringScheduleConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'monitoringScheduleConfig'");
            }
            resourceInputs["endpointName"] = args ? args.endpointName : undefined;
            resourceInputs["failureReason"] = args ? args.failureReason : undefined;
            resourceInputs["lastMonitoringExecutionSummary"] = args ? args.lastMonitoringExecutionSummary : undefined;
            resourceInputs["monitoringScheduleConfig"] = args ? args.monitoringScheduleConfig : undefined;
            resourceInputs["monitoringScheduleName"] = args ? args.monitoringScheduleName : undefined;
            resourceInputs["monitoringScheduleStatus"] = args ? args.monitoringScheduleStatus : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["monitoringScheduleArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["endpointName"] = undefined /*out*/;
            resourceInputs["failureReason"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["lastMonitoringExecutionSummary"] = undefined /*out*/;
            resourceInputs["monitoringScheduleArn"] = undefined /*out*/;
            resourceInputs["monitoringScheduleConfig"] = undefined /*out*/;
            resourceInputs["monitoringScheduleName"] = undefined /*out*/;
            resourceInputs["monitoringScheduleStatus"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MonitoringSchedule.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing MonitoringSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MonitoringSchedule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MonitoringSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MonitoringSchedule.__pulumiType;
    }
}
exports.MonitoringSchedule = MonitoringSchedule;
/** @internal */
MonitoringSchedule.__pulumiType = 'aws-native:sagemaker:MonitoringSchedule';
