// @pulumi/aws-native/codebuild/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceCredential = exports.ReportGroup = exports.Project = exports.getSourceCredentialOutput = exports.getSourceCredential = exports.getReportGroupOutput = exports.getReportGroup = exports.getProjectOutput = exports.getProject = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getReportGroup = null;
exports.getReportGroupOutput = null;
utilities.lazyLoad(exports, ["getReportGroup", "getReportGroupOutput"], () => require("./getReportGroup"));
exports.getSourceCredential = null;
exports.getSourceCredentialOutput = null;
utilities.lazyLoad(exports, ["getSourceCredential", "getSourceCredentialOutput"], () => require("./getSourceCredential"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.ReportGroup = null;
utilities.lazyLoad(exports, ["ReportGroup"], () => require("./reportGroup"));
exports.SourceCredential = null;
utilities.lazyLoad(exports, ["SourceCredential"], () => require("./sourceCredential"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:codebuild:Project":
                return new exports.Project(name, undefined, { urn });
            case "aws-native:codebuild:ReportGroup":
                return new exports.ReportGroup(name, undefined, { urn });
            case "aws-native:codebuild:SourceCredential":
                return new exports.SourceCredential(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "codebuild", _module);
