// @pulumi/aws-native/resiliencehub/app.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.App = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type Definition for AWS::ResilienceHub::App.
 */
class App extends pulumi.CustomResource {
    /**
     * Create a App resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.appTemplateBody === undefined) && !opts.urn) {
                throw new Error("Missing required property 'appTemplateBody'");
            }
            if ((!args || args.resourceMappings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceMappings'");
            }
            resourceInputs["appAssessmentSchedule"] = args ? args.appAssessmentSchedule : undefined;
            resourceInputs["appTemplateBody"] = args ? args.appTemplateBody : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resiliencyPolicyArn"] = args ? args.resiliencyPolicyArn : undefined;
            resourceInputs["resourceMappings"] = args ? args.resourceMappings : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["appArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["appArn"] = undefined /*out*/;
            resourceInputs["appAssessmentSchedule"] = undefined /*out*/;
            resourceInputs["appTemplateBody"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["resiliencyPolicyArn"] = undefined /*out*/;
            resourceInputs["resourceMappings"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(App.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing App resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new App(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of App.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === App.__pulumiType;
    }
}
exports.App = App;
/** @internal */
App.__pulumiType = 'aws-native:resiliencehub:App';
