// @pulumi/aws-native/resiliencehub/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResiliencyPolicy = exports.getResiliencyPolicyOutput = exports.getResiliencyPolicy = exports.getAppOutput = exports.getApp = exports.App = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.App = null;
utilities.lazyLoad(exports, ["App"], () => require("./app"));
exports.getApp = null;
exports.getAppOutput = null;
utilities.lazyLoad(exports, ["getApp", "getAppOutput"], () => require("./getApp"));
exports.getResiliencyPolicy = null;
exports.getResiliencyPolicyOutput = null;
utilities.lazyLoad(exports, ["getResiliencyPolicy", "getResiliencyPolicyOutput"], () => require("./getResiliencyPolicy"));
exports.ResiliencyPolicy = null;
utilities.lazyLoad(exports, ["ResiliencyPolicy"], () => require("./resiliencyPolicy"));
// Export enums:
__exportStar(require("../types/enums/resiliencehub"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:resiliencehub:App":
                return new exports.App(name, undefined, { urn });
            case "aws-native:resiliencehub:ResiliencyPolicy":
                return new exports.ResiliencyPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "resiliencehub", _module);
