// @pulumi/aws-native/opsworks/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Volume = exports.UserProfile = exports.Stack = exports.Layer = exports.Instance = exports.getVolumeOutput = exports.getVolume = exports.getUserProfileOutput = exports.getUserProfile = exports.getStackOutput = exports.getStack = exports.getLayerOutput = exports.getLayer = exports.getInstanceOutput = exports.getInstance = exports.getElasticLoadBalancerAttachmentOutput = exports.getElasticLoadBalancerAttachment = exports.getAppOutput = exports.getApp = exports.ElasticLoadBalancerAttachment = exports.App = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.App = null;
utilities.lazyLoad(exports, ["App"], () => require("./app"));
exports.ElasticLoadBalancerAttachment = null;
utilities.lazyLoad(exports, ["ElasticLoadBalancerAttachment"], () => require("./elasticLoadBalancerAttachment"));
exports.getApp = null;
exports.getAppOutput = null;
utilities.lazyLoad(exports, ["getApp", "getAppOutput"], () => require("./getApp"));
exports.getElasticLoadBalancerAttachment = null;
exports.getElasticLoadBalancerAttachmentOutput = null;
utilities.lazyLoad(exports, ["getElasticLoadBalancerAttachment", "getElasticLoadBalancerAttachmentOutput"], () => require("./getElasticLoadBalancerAttachment"));
exports.getInstance = null;
exports.getInstanceOutput = null;
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
exports.getLayer = null;
exports.getLayerOutput = null;
utilities.lazyLoad(exports, ["getLayer", "getLayerOutput"], () => require("./getLayer"));
exports.getStack = null;
exports.getStackOutput = null;
utilities.lazyLoad(exports, ["getStack", "getStackOutput"], () => require("./getStack"));
exports.getUserProfile = null;
exports.getUserProfileOutput = null;
utilities.lazyLoad(exports, ["getUserProfile", "getUserProfileOutput"], () => require("./getUserProfile"));
exports.getVolume = null;
exports.getVolumeOutput = null;
utilities.lazyLoad(exports, ["getVolume", "getVolumeOutput"], () => require("./getVolume"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.Layer = null;
utilities.lazyLoad(exports, ["Layer"], () => require("./layer"));
exports.Stack = null;
utilities.lazyLoad(exports, ["Stack"], () => require("./stack"));
exports.UserProfile = null;
utilities.lazyLoad(exports, ["UserProfile"], () => require("./userProfile"));
exports.Volume = null;
utilities.lazyLoad(exports, ["Volume"], () => require("./volume"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:opsworks:App":
                return new exports.App(name, undefined, { urn });
            case "aws-native:opsworks:ElasticLoadBalancerAttachment":
                return new exports.ElasticLoadBalancerAttachment(name, undefined, { urn });
            case "aws-native:opsworks:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "aws-native:opsworks:Layer":
                return new exports.Layer(name, undefined, { urn });
            case "aws-native:opsworks:Stack":
                return new exports.Stack(name, undefined, { urn });
            case "aws-native:opsworks:UserProfile":
                return new exports.UserProfile(name, undefined, { urn });
            case "aws-native:opsworks:Volume":
                return new exports.Volume(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "opsworks", _module);
