// @pulumi/aws-native/devicefarm/networkProfile.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * AWS::DeviceFarm::NetworkProfile creates a new DF Network Profile
 */
class NetworkProfile extends pulumi.CustomResource {
    /**
     * Create a NetworkProfile resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.projectArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'projectArn'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["downlinkBandwidthBits"] = args ? args.downlinkBandwidthBits : undefined;
            resourceInputs["downlinkDelayMs"] = args ? args.downlinkDelayMs : undefined;
            resourceInputs["downlinkJitterMs"] = args ? args.downlinkJitterMs : undefined;
            resourceInputs["downlinkLossPercent"] = args ? args.downlinkLossPercent : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["projectArn"] = args ? args.projectArn : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["uplinkBandwidthBits"] = args ? args.uplinkBandwidthBits : undefined;
            resourceInputs["uplinkDelayMs"] = args ? args.uplinkDelayMs : undefined;
            resourceInputs["uplinkJitterMs"] = args ? args.uplinkJitterMs : undefined;
            resourceInputs["uplinkLossPercent"] = args ? args.uplinkLossPercent : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["downlinkBandwidthBits"] = undefined /*out*/;
            resourceInputs["downlinkDelayMs"] = undefined /*out*/;
            resourceInputs["downlinkJitterMs"] = undefined /*out*/;
            resourceInputs["downlinkLossPercent"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["projectArn"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["uplinkBandwidthBits"] = undefined /*out*/;
            resourceInputs["uplinkDelayMs"] = undefined /*out*/;
            resourceInputs["uplinkJitterMs"] = undefined /*out*/;
            resourceInputs["uplinkLossPercent"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(NetworkProfile.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing NetworkProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkProfile(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkProfile.__pulumiType;
    }
}
exports.NetworkProfile = NetworkProfile;
/** @internal */
NetworkProfile.__pulumiType = 'aws-native:devicefarm:NetworkProfile';
