// @pulumi/aws-native/types/enums/s3/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageLensS3BucketDestinationOutputSchemaVersion = exports.StorageLensS3BucketDestinationFormat = exports.MultiRegionAccessPointPolicyPolicyStatusPropertiesIsPublic = exports.BucketVersioningConfigurationStatus = exports.BucketTransitionStorageClass = exports.BucketTieringAccessTier = exports.BucketSseKmsEncryptedObjectsStatus = exports.BucketServerSideEncryptionByDefaultSSEAlgorithm = exports.BucketRuleStatus = exports.BucketReplicationTimeStatus = exports.BucketReplicationRuleStatus = exports.BucketReplicationDestinationStorageClass = exports.BucketReplicaModificationsStatus = exports.BucketRedirectRuleProtocol = exports.BucketRedirectAllRequestsToProtocol = exports.BucketOwnershipControlsRuleObjectOwnership = exports.BucketNoncurrentVersionTransitionStorageClass = exports.BucketMetricsStatus = exports.BucketInventoryConfigurationScheduleFrequency = exports.BucketInventoryConfigurationOptionalFieldsItem = exports.BucketInventoryConfigurationIncludedObjectVersions = exports.BucketIntelligentTieringConfigurationStatus = exports.BucketDestinationFormat = exports.BucketDeleteMarkerReplicationStatus = exports.BucketDefaultRetentionMode = exports.BucketCorsRuleAllowedMethodsItem = exports.BucketAccessControl = exports.BucketAccelerateConfigurationAccelerationStatus = exports.AccessPointPolicyStatusPropertiesIsPublic = exports.AccessPointNetworkOrigin = void 0;
exports.AccessPointNetworkOrigin = {
    Internet: "Internet",
    Vpc: "VPC",
};
exports.AccessPointPolicyStatusPropertiesIsPublic = {
    True: "true",
    False: "false",
};
exports.BucketAccelerateConfigurationAccelerationStatus = {
    Enabled: "Enabled",
    Suspended: "Suspended",
};
exports.BucketAccessControl = {
    AuthenticatedRead: "AuthenticatedRead",
    AwsExecRead: "AwsExecRead",
    BucketOwnerFullControl: "BucketOwnerFullControl",
    BucketOwnerRead: "BucketOwnerRead",
    LogDeliveryWrite: "LogDeliveryWrite",
    Private: "Private",
    PublicRead: "PublicRead",
    PublicReadWrite: "PublicReadWrite",
};
exports.BucketCorsRuleAllowedMethodsItem = {
    Get: "GET",
    Put: "PUT",
    Head: "HEAD",
    Post: "POST",
    Delete: "DELETE",
};
exports.BucketDefaultRetentionMode = {
    Compliance: "COMPLIANCE",
    Governance: "GOVERNANCE",
};
exports.BucketDeleteMarkerReplicationStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.BucketDestinationFormat = {
    Csv: "CSV",
    Orc: "ORC",
    Parquet: "Parquet",
};
exports.BucketIntelligentTieringConfigurationStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.BucketInventoryConfigurationIncludedObjectVersions = {
    All: "All",
    Current: "Current",
};
exports.BucketInventoryConfigurationOptionalFieldsItem = {
    Size: "Size",
    LastModifiedDate: "LastModifiedDate",
    StorageClass: "StorageClass",
    ETag: "ETag",
    IsMultipartUploaded: "IsMultipartUploaded",
    ReplicationStatus: "ReplicationStatus",
    EncryptionStatus: "EncryptionStatus",
    ObjectLockRetainUntilDate: "ObjectLockRetainUntilDate",
    ObjectLockMode: "ObjectLockMode",
    ObjectLockLegalHoldStatus: "ObjectLockLegalHoldStatus",
    IntelligentTieringAccessTier: "IntelligentTieringAccessTier",
    BucketKeyStatus: "BucketKeyStatus",
};
exports.BucketInventoryConfigurationScheduleFrequency = {
    Daily: "Daily",
    Weekly: "Weekly",
};
exports.BucketMetricsStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.BucketNoncurrentVersionTransitionStorageClass = {
    DeepArchive: "DEEP_ARCHIVE",
    Glacier: "GLACIER",
    GlacierIr: "GLACIER_IR",
    IntelligentTiering: "INTELLIGENT_TIERING",
    OnezoneIa: "ONEZONE_IA",
    StandardIa: "STANDARD_IA",
};
exports.BucketOwnershipControlsRuleObjectOwnership = {
    ObjectWriter: "ObjectWriter",
    BucketOwnerPreferred: "BucketOwnerPreferred",
    BucketOwnerEnforced: "BucketOwnerEnforced",
};
exports.BucketRedirectAllRequestsToProtocol = {
    Http: "http",
    Https: "https",
};
exports.BucketRedirectRuleProtocol = {
    Http: "http",
    Https: "https",
};
exports.BucketReplicaModificationsStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.BucketReplicationDestinationStorageClass = {
    DeepArchive: "DEEP_ARCHIVE",
    Glacier: "GLACIER",
    GlacierIr: "GLACIER_IR",
    IntelligentTiering: "INTELLIGENT_TIERING",
    OnezoneIa: "ONEZONE_IA",
    ReducedRedundancy: "REDUCED_REDUNDANCY",
    Standard: "STANDARD",
    StandardIa: "STANDARD_IA",
};
exports.BucketReplicationRuleStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.BucketReplicationTimeStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.BucketRuleStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.BucketServerSideEncryptionByDefaultSSEAlgorithm = {
    Awskms: "aws:kms",
    Aes256: "AES256",
};
exports.BucketSseKmsEncryptedObjectsStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.BucketTieringAccessTier = {
    ArchiveAccess: "ARCHIVE_ACCESS",
    DeepArchiveAccess: "DEEP_ARCHIVE_ACCESS",
};
exports.BucketTransitionStorageClass = {
    DeepArchive: "DEEP_ARCHIVE",
    Glacier: "GLACIER",
    GlacierIr: "GLACIER_IR",
    IntelligentTiering: "INTELLIGENT_TIERING",
    OnezoneIa: "ONEZONE_IA",
    StandardIa: "STANDARD_IA",
};
exports.BucketVersioningConfigurationStatus = {
    Enabled: "Enabled",
    Suspended: "Suspended",
};
exports.MultiRegionAccessPointPolicyPolicyStatusPropertiesIsPublic = {
    True: "true",
    False: "false",
};
exports.StorageLensS3BucketDestinationFormat = {
    Csv: "CSV",
    Parquet: "Parquet",
};
exports.StorageLensS3BucketDestinationOutputSchemaVersion = {
    V1: "V_1",
};
