// @pulumi/aws-native/iotwireless/fuotaTask.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FuotaTask = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Create and manage FUOTA tasks.
 */
class FuotaTask extends pulumi.CustomResource {
    /**
     * Create a FuotaTask resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.firmwareUpdateImage === undefined) && !opts.urn) {
                throw new Error("Missing required property 'firmwareUpdateImage'");
            }
            if ((!args || args.firmwareUpdateRole === undefined) && !opts.urn) {
                throw new Error("Missing required property 'firmwareUpdateRole'");
            }
            if ((!args || args.loRaWAN === undefined) && !opts.urn) {
                throw new Error("Missing required property 'loRaWAN'");
            }
            resourceInputs["associateMulticastGroup"] = args ? args.associateMulticastGroup : undefined;
            resourceInputs["associateWirelessDevice"] = args ? args.associateWirelessDevice : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["disassociateMulticastGroup"] = args ? args.disassociateMulticastGroup : undefined;
            resourceInputs["disassociateWirelessDevice"] = args ? args.disassociateWirelessDevice : undefined;
            resourceInputs["firmwareUpdateImage"] = args ? args.firmwareUpdateImage : undefined;
            resourceInputs["firmwareUpdateRole"] = args ? args.firmwareUpdateRole : undefined;
            resourceInputs["loRaWAN"] = args ? args.loRaWAN : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["fuotaTaskStatus"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["associateMulticastGroup"] = undefined /*out*/;
            resourceInputs["associateWirelessDevice"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["disassociateMulticastGroup"] = undefined /*out*/;
            resourceInputs["disassociateWirelessDevice"] = undefined /*out*/;
            resourceInputs["firmwareUpdateImage"] = undefined /*out*/;
            resourceInputs["firmwareUpdateRole"] = undefined /*out*/;
            resourceInputs["fuotaTaskStatus"] = undefined /*out*/;
            resourceInputs["loRaWAN"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(FuotaTask.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing FuotaTask resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FuotaTask(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FuotaTask.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FuotaTask.__pulumiType;
    }
}
exports.FuotaTask = FuotaTask;
/** @internal */
FuotaTask.__pulumiType = 'aws-native:iotwireless:FuotaTask';
