// @pulumi/aws-native/backup/framework.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Framework = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Contains detailed information about a framework. Frameworks contain controls, which evaluate and report on your backup events and resources. Frameworks generate daily compliance results.
 */
class Framework extends pulumi.CustomResource {
    /**
     * Create a Framework resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.frameworkControls === undefined) && !opts.urn) {
                throw new Error("Missing required property 'frameworkControls'");
            }
            resourceInputs["frameworkControls"] = args ? args.frameworkControls : undefined;
            resourceInputs["frameworkDescription"] = args ? args.frameworkDescription : undefined;
            resourceInputs["frameworkName"] = args ? args.frameworkName : undefined;
            resourceInputs["frameworkTags"] = args ? args.frameworkTags : undefined;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["deploymentStatus"] = undefined /*out*/;
            resourceInputs["frameworkArn"] = undefined /*out*/;
            resourceInputs["frameworkStatus"] = undefined /*out*/;
        }
        else {
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["deploymentStatus"] = undefined /*out*/;
            resourceInputs["frameworkArn"] = undefined /*out*/;
            resourceInputs["frameworkControls"] = undefined /*out*/;
            resourceInputs["frameworkDescription"] = undefined /*out*/;
            resourceInputs["frameworkName"] = undefined /*out*/;
            resourceInputs["frameworkStatus"] = undefined /*out*/;
            resourceInputs["frameworkTags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Framework.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Framework resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Framework(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Framework.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Framework.__pulumiType;
    }
}
exports.Framework = Framework;
/** @internal */
Framework.__pulumiType = 'aws-native:backup:Framework';
