// @pulumi/aws-native/backup/reportPlan.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportPlan = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Contains detailed information about a report plan in AWS Backup Audit Manager.
 */
class ReportPlan extends pulumi.CustomResource {
    /**
     * Create a ReportPlan resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.reportDeliveryChannel === undefined) && !opts.urn) {
                throw new Error("Missing required property 'reportDeliveryChannel'");
            }
            if ((!args || args.reportSetting === undefined) && !opts.urn) {
                throw new Error("Missing required property 'reportSetting'");
            }
            resourceInputs["reportDeliveryChannel"] = args ? args.reportDeliveryChannel : undefined;
            resourceInputs["reportPlanDescription"] = args ? args.reportPlanDescription : undefined;
            resourceInputs["reportPlanName"] = args ? args.reportPlanName : undefined;
            resourceInputs["reportPlanTags"] = args ? args.reportPlanTags : undefined;
            resourceInputs["reportSetting"] = args ? args.reportSetting : undefined;
            resourceInputs["reportPlanArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["reportDeliveryChannel"] = undefined /*out*/;
            resourceInputs["reportPlanArn"] = undefined /*out*/;
            resourceInputs["reportPlanDescription"] = undefined /*out*/;
            resourceInputs["reportPlanName"] = undefined /*out*/;
            resourceInputs["reportPlanTags"] = undefined /*out*/;
            resourceInputs["reportSetting"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ReportPlan.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ReportPlan resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ReportPlan(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReportPlan.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReportPlan.__pulumiType;
    }
}
exports.ReportPlan = ReportPlan;
/** @internal */
ReportPlan.__pulumiType = 'aws-native:backup:ReportPlan';
