// @pulumi/aws-native/workspaces/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.getWorkspaceOutput = exports.getWorkspace = exports.getConnectionAliasOutput = exports.getConnectionAlias = exports.ConnectionAlias = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ConnectionAlias = null;
utilities.lazyLoad(exports, ["ConnectionAlias"], () => require("./connectionAlias"));
exports.getConnectionAlias = null;
exports.getConnectionAliasOutput = null;
utilities.lazyLoad(exports, ["getConnectionAlias", "getConnectionAliasOutput"], () => require("./getConnectionAlias"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
// Export enums:
__exportStar(require("../types/enums/workspaces"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:workspaces:ConnectionAlias":
                return new exports.ConnectionAlias(name, undefined, { urn });
            case "aws-native:workspaces:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "workspaces", _module);
