// @pulumi/aws-native/types/enums/transfer/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowStepType = exports.WorkflowStepCopyStepDetailsPropertiesOverwriteExisting = exports.ProfileType = exports.ConnectorAs2ConfigPropertiesSigningAlgorithm = exports.ConnectorAs2ConfigPropertiesMdnSigningAlgorithm = exports.ConnectorAs2ConfigPropertiesMdnResponse = exports.ConnectorAs2ConfigPropertiesEncryptionAlgorithm = exports.ConnectorAs2ConfigPropertiesCompression = exports.CertificateUsage = exports.CertificateType = exports.CertificateStatus = exports.AgreementStatus = void 0;
exports.AgreementStatus = {
    Active: "ACTIVE",
    Inactive: "INACTIVE",
};
exports.CertificateStatus = {
    Active: "ACTIVE",
    Pending: "PENDING",
    Inactive: "INACTIVE",
};
exports.CertificateType = {
    Certificate: "CERTIFICATE",
    CertificateWithPrivateKey: "CERTIFICATE_WITH_PRIVATE_KEY",
};
exports.CertificateUsage = {
    Signing: "SIGNING",
    Encryption: "ENCRYPTION",
};
exports.ConnectorAs2ConfigPropertiesCompression = {
    Zlib: "ZLIB",
    Disabled: "DISABLED",
};
exports.ConnectorAs2ConfigPropertiesEncryptionAlgorithm = {
    Aes128Cbc: "AES128_CBC",
    Aes192Cbc: "AES192_CBC",
    Aes256Cbc: "AES256_CBC",
};
exports.ConnectorAs2ConfigPropertiesMdnResponse = {
    Sync: "SYNC",
    None: "NONE",
};
exports.ConnectorAs2ConfigPropertiesMdnSigningAlgorithm = {
    Sha256: "SHA256",
    Sha384: "SHA384",
    Sha512: "SHA512",
    Sha1: "SHA1",
    None: "NONE",
    Default: "DEFAULT",
};
exports.ConnectorAs2ConfigPropertiesSigningAlgorithm = {
    Sha256: "SHA256",
    Sha384: "SHA384",
    Sha512: "SHA512",
    Sha1: "SHA1",
    None: "NONE",
};
exports.ProfileType = {
    Local: "LOCAL",
    Partner: "PARTNER",
};
exports.WorkflowStepCopyStepDetailsPropertiesOverwriteExisting = {
    True: "TRUE",
    False: "FALSE",
};
exports.WorkflowStepType = {
    Copy: "COPY",
    Custom: "CUSTOM",
    Delete: "DELETE",
    Tag: "TAG",
};
