// @pulumi/aws-native/memorydb/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.SubnetGroup = exports.ParameterGroup = exports.getUserOutput = exports.getUser = exports.getSubnetGroupOutput = exports.getSubnetGroup = exports.getParameterGroupOutput = exports.getParameterGroup = exports.getClusterOutput = exports.getCluster = exports.getACLOutput = exports.getACL = exports.Cluster = exports.ACL = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ACL = null;
utilities.lazyLoad(exports, ["ACL"], () => require("./acl"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.getACL = null;
exports.getACLOutput = null;
utilities.lazyLoad(exports, ["getACL", "getACLOutput"], () => require("./getACL"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getParameterGroup = null;
exports.getParameterGroupOutput = null;
utilities.lazyLoad(exports, ["getParameterGroup", "getParameterGroupOutput"], () => require("./getParameterGroup"));
exports.getSubnetGroup = null;
exports.getSubnetGroupOutput = null;
utilities.lazyLoad(exports, ["getSubnetGroup", "getSubnetGroupOutput"], () => require("./getSubnetGroup"));
exports.getUser = null;
exports.getUserOutput = null;
utilities.lazyLoad(exports, ["getUser", "getUserOutput"], () => require("./getUser"));
exports.ParameterGroup = null;
utilities.lazyLoad(exports, ["ParameterGroup"], () => require("./parameterGroup"));
exports.SubnetGroup = null;
utilities.lazyLoad(exports, ["SubnetGroup"], () => require("./subnetGroup"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
// Export enums:
__exportStar(require("../types/enums/memorydb"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:memorydb:ACL":
                return new exports.ACL(name, undefined, { urn });
            case "aws-native:memorydb:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "aws-native:memorydb:ParameterGroup":
                return new exports.ParameterGroup(name, undefined, { urn });
            case "aws-native:memorydb:SubnetGroup":
                return new exports.SubnetGroup(name, undefined, { urn });
            case "aws-native:memorydb:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "memorydb", _module);
