// @pulumi/aws-native/lakeformation/dataCellsFilter.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataCellsFilter = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A resource schema representing a Lake Formation Data Cells Filter.
 */
class DataCellsFilter extends pulumi.CustomResource {
    /**
     * Create a DataCellsFilter resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.tableCatalogId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tableCatalogId'");
            }
            if ((!args || args.tableName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tableName'");
            }
            resourceInputs["columnNames"] = args ? args.columnNames : undefined;
            resourceInputs["columnWildcard"] = args ? args.columnWildcard : undefined;
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["rowFilter"] = args ? args.rowFilter : undefined;
            resourceInputs["tableCatalogId"] = args ? args.tableCatalogId : undefined;
            resourceInputs["tableName"] = args ? args.tableName : undefined;
        }
        else {
            resourceInputs["columnNames"] = undefined /*out*/;
            resourceInputs["columnWildcard"] = undefined /*out*/;
            resourceInputs["databaseName"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["rowFilter"] = undefined /*out*/;
            resourceInputs["tableCatalogId"] = undefined /*out*/;
            resourceInputs["tableName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DataCellsFilter.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DataCellsFilter resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DataCellsFilter(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DataCellsFilter.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataCellsFilter.__pulumiType;
    }
}
exports.DataCellsFilter = DataCellsFilter;
/** @internal */
DataCellsFilter.__pulumiType = 'aws-native:lakeformation:DataCellsFilter';
