// @pulumi/aws-native/kinesisfirehose/deliveryStream.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeliveryStream = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::KinesisFirehose::DeliveryStream
 */
class DeliveryStream extends pulumi.CustomResource {
    /**
     * Create a DeliveryStream resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["amazonopensearchserviceDestinationConfiguration"] = args ? args.amazonopensearchserviceDestinationConfiguration : undefined;
            resourceInputs["deliveryStreamEncryptionConfigurationInput"] = args ? args.deliveryStreamEncryptionConfigurationInput : undefined;
            resourceInputs["deliveryStreamName"] = args ? args.deliveryStreamName : undefined;
            resourceInputs["deliveryStreamType"] = args ? args.deliveryStreamType : undefined;
            resourceInputs["elasticsearchDestinationConfiguration"] = args ? args.elasticsearchDestinationConfiguration : undefined;
            resourceInputs["extendedS3DestinationConfiguration"] = args ? args.extendedS3DestinationConfiguration : undefined;
            resourceInputs["httpEndpointDestinationConfiguration"] = args ? args.httpEndpointDestinationConfiguration : undefined;
            resourceInputs["kinesisStreamSourceConfiguration"] = args ? args.kinesisStreamSourceConfiguration : undefined;
            resourceInputs["redshiftDestinationConfiguration"] = args ? args.redshiftDestinationConfiguration : undefined;
            resourceInputs["s3DestinationConfiguration"] = args ? args.s3DestinationConfiguration : undefined;
            resourceInputs["splunkDestinationConfiguration"] = args ? args.splunkDestinationConfiguration : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        else {
            resourceInputs["amazonopensearchserviceDestinationConfiguration"] = undefined /*out*/;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["deliveryStreamEncryptionConfigurationInput"] = undefined /*out*/;
            resourceInputs["deliveryStreamName"] = undefined /*out*/;
            resourceInputs["deliveryStreamType"] = undefined /*out*/;
            resourceInputs["elasticsearchDestinationConfiguration"] = undefined /*out*/;
            resourceInputs["extendedS3DestinationConfiguration"] = undefined /*out*/;
            resourceInputs["httpEndpointDestinationConfiguration"] = undefined /*out*/;
            resourceInputs["kinesisStreamSourceConfiguration"] = undefined /*out*/;
            resourceInputs["redshiftDestinationConfiguration"] = undefined /*out*/;
            resourceInputs["s3DestinationConfiguration"] = undefined /*out*/;
            resourceInputs["splunkDestinationConfiguration"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DeliveryStream.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DeliveryStream resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DeliveryStream(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DeliveryStream.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DeliveryStream.__pulumiType;
    }
}
exports.DeliveryStream = DeliveryStream;
/** @internal */
DeliveryStream.__pulumiType = 'aws-native:kinesisfirehose:DeliveryStream';
