// @pulumi/aws-native/iottwinmaker/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.Scene = exports.getWorkspaceOutput = exports.getWorkspace = exports.getSceneOutput = exports.getScene = exports.getEntityOutput = exports.getEntity = exports.getComponentTypeOutput = exports.getComponentType = exports.Entity = exports.ComponentType = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ComponentType = null;
utilities.lazyLoad(exports, ["ComponentType"], () => require("./componentType"));
exports.Entity = null;
utilities.lazyLoad(exports, ["Entity"], () => require("./entity"));
exports.getComponentType = null;
exports.getComponentTypeOutput = null;
utilities.lazyLoad(exports, ["getComponentType", "getComponentTypeOutput"], () => require("./getComponentType"));
exports.getEntity = null;
exports.getEntityOutput = null;
utilities.lazyLoad(exports, ["getEntity", "getEntityOutput"], () => require("./getEntity"));
exports.getScene = null;
exports.getSceneOutput = null;
utilities.lazyLoad(exports, ["getScene", "getSceneOutput"], () => require("./getScene"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.Scene = null;
utilities.lazyLoad(exports, ["Scene"], () => require("./scene"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
// Export enums:
__exportStar(require("../types/enums/iottwinmaker"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:iottwinmaker:ComponentType":
                return new exports.ComponentType(name, undefined, { urn });
            case "aws-native:iottwinmaker:Entity":
                return new exports.Entity(name, undefined, { urn });
            case "aws-native:iottwinmaker:Scene":
                return new exports.Scene(name, undefined, { urn });
            case "aws-native:iottwinmaker:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "iottwinmaker", _module);
