// @pulumi/aws-native/iot/accountAuditConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountAuditConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Configures the Device Defender audit settings for this account. Settings include how audit notifications are sent and which audit checks are enabled or disabled.
 */
class AccountAuditConfiguration extends pulumi.CustomResource {
    /**
     * Create a AccountAuditConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.auditCheckConfigurations === undefined) && !opts.urn) {
                throw new Error("Missing required property 'auditCheckConfigurations'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["auditCheckConfigurations"] = args ? args.auditCheckConfigurations : undefined;
            resourceInputs["auditNotificationTargetConfigurations"] = args ? args.auditNotificationTargetConfigurations : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
        }
        else {
            resourceInputs["accountId"] = undefined /*out*/;
            resourceInputs["auditCheckConfigurations"] = undefined /*out*/;
            resourceInputs["auditNotificationTargetConfigurations"] = undefined /*out*/;
            resourceInputs["roleArn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AccountAuditConfiguration.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing AccountAuditConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AccountAuditConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AccountAuditConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AccountAuditConfiguration.__pulumiType;
    }
}
exports.AccountAuditConfiguration = AccountAuditConfiguration;
/** @internal */
AccountAuditConfiguration.__pulumiType = 'aws-native:iot:AccountAuditConfiguration';
