// @pulumi/aws-native/iot/cacertificate.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CACertificate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Registers a CA Certificate in IoT.
 */
class CACertificate extends pulumi.CustomResource {
    /**
     * Create a CACertificate resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.cACertificatePem === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cACertificatePem'");
            }
            if ((!args || args.status === undefined) && !opts.urn) {
                throw new Error("Missing required property 'status'");
            }
            resourceInputs["autoRegistrationStatus"] = args ? args.autoRegistrationStatus : undefined;
            resourceInputs["cACertificatePem"] = args ? args.cACertificatePem : undefined;
            resourceInputs["certificateMode"] = args ? args.certificateMode : undefined;
            resourceInputs["registrationConfig"] = args ? args.registrationConfig : undefined;
            resourceInputs["removeAutoRegistration"] = args ? args.removeAutoRegistration : undefined;
            resourceInputs["status"] = args ? args.status : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["verificationCertificatePem"] = args ? args.verificationCertificatePem : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["autoRegistrationStatus"] = undefined /*out*/;
            resourceInputs["cACertificatePem"] = undefined /*out*/;
            resourceInputs["certificateMode"] = undefined /*out*/;
            resourceInputs["registrationConfig"] = undefined /*out*/;
            resourceInputs["removeAutoRegistration"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["verificationCertificatePem"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CACertificate.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing CACertificate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CACertificate(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CACertificate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CACertificate.__pulumiType;
    }
}
exports.CACertificate = CACertificate;
/** @internal */
CACertificate.__pulumiType = 'aws-native:iot:CACertificate';
