// @pulumi/aws-native/iot/customMetric.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomMetric = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A custom metric published by your devices to Device Defender.
 */
class CustomMetric extends pulumi.CustomResource {
    /**
     * Create a CustomMetric resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.metricType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'metricType'");
            }
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["metricName"] = args ? args.metricName : undefined;
            resourceInputs["metricType"] = args ? args.metricType : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["metricArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["metricArn"] = undefined /*out*/;
            resourceInputs["metricName"] = undefined /*out*/;
            resourceInputs["metricType"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CustomMetric.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing CustomMetric resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CustomMetric(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CustomMetric.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomMetric.__pulumiType;
    }
}
exports.CustomMetric = CustomMetric;
/** @internal */
CustomMetric.__pulumiType = 'aws-native:iot:CustomMetric';
