// @pulumi/aws-native/iot/mitigationAction.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MitigationAction = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Mitigation actions can be used to take actions to mitigate issues that were found in an Audit finding or Detect violation.
 */
class MitigationAction extends pulumi.CustomResource {
    /**
     * Create a MitigationAction resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.actionParams === undefined) && !opts.urn) {
                throw new Error("Missing required property 'actionParams'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["actionName"] = args ? args.actionName : undefined;
            resourceInputs["actionParams"] = args ? args.actionParams : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["mitigationActionArn"] = undefined /*out*/;
            resourceInputs["mitigationActionId"] = undefined /*out*/;
        }
        else {
            resourceInputs["actionName"] = undefined /*out*/;
            resourceInputs["actionParams"] = undefined /*out*/;
            resourceInputs["mitigationActionArn"] = undefined /*out*/;
            resourceInputs["mitigationActionId"] = undefined /*out*/;
            resourceInputs["roleArn"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MitigationAction.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing MitigationAction resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MitigationAction(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MitigationAction.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MitigationAction.__pulumiType;
    }
}
exports.MitigationAction = MitigationAction;
/** @internal */
MitigationAction.__pulumiType = 'aws-native:iot:MitigationAction';
