// @pulumi/aws-native/iot/securityProfile.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A security profile defines a set of expected behaviors for devices in your account.
 */
class SecurityProfile extends pulumi.CustomResource {
    /**
     * Create a SecurityProfile resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["additionalMetricsToRetainV2"] = args ? args.additionalMetricsToRetainV2 : undefined;
            resourceInputs["alertTargets"] = args ? args.alertTargets : undefined;
            resourceInputs["behaviors"] = args ? args.behaviors : undefined;
            resourceInputs["securityProfileDescription"] = args ? args.securityProfileDescription : undefined;
            resourceInputs["securityProfileName"] = args ? args.securityProfileName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["targetArns"] = args ? args.targetArns : undefined;
            resourceInputs["securityProfileArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["additionalMetricsToRetainV2"] = undefined /*out*/;
            resourceInputs["alertTargets"] = undefined /*out*/;
            resourceInputs["behaviors"] = undefined /*out*/;
            resourceInputs["securityProfileArn"] = undefined /*out*/;
            resourceInputs["securityProfileDescription"] = undefined /*out*/;
            resourceInputs["securityProfileName"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["targetArns"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SecurityProfile.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SecurityProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SecurityProfile(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SecurityProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecurityProfile.__pulumiType;
    }
}
exports.SecurityProfile = SecurityProfile;
/** @internal */
SecurityProfile.__pulumiType = 'aws-native:iot:SecurityProfile';
