// @pulumi/aws-native/iot/topicRuleDestination.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopicRuleDestination = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::IoT::TopicRuleDestination
 */
class TopicRuleDestination extends pulumi.CustomResource {
    /**
     * Create a TopicRuleDestination resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["httpUrlProperties"] = args ? args.httpUrlProperties : undefined;
            resourceInputs["status"] = args ? args.status : undefined;
            resourceInputs["vpcProperties"] = args ? args.vpcProperties : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["statusReason"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["httpUrlProperties"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["statusReason"] = undefined /*out*/;
            resourceInputs["vpcProperties"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TopicRuleDestination.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing TopicRuleDestination resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TopicRuleDestination(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TopicRuleDestination.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TopicRuleDestination.__pulumiType;
    }
}
exports.TopicRuleDestination = TopicRuleDestination;
/** @internal */
TopicRuleDestination.__pulumiType = 'aws-native:iot:TopicRuleDestination';
