// @pulumi/aws-native/pinpointemail/identity.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Identity = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::PinpointEmail::Identity
 *
 * @deprecated Identity is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Identity extends pulumi.CustomResource {
    /**
     * Create a Identity resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Identity is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Identity is deprecated: Identity is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["dkimSigningEnabled"] = args ? args.dkimSigningEnabled : undefined;
            resourceInputs["feedbackForwardingEnabled"] = args ? args.feedbackForwardingEnabled : undefined;
            resourceInputs["mailFromAttributes"] = args ? args.mailFromAttributes : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["identityDNSRecordName1"] = undefined /*out*/;
            resourceInputs["identityDNSRecordName2"] = undefined /*out*/;
            resourceInputs["identityDNSRecordName3"] = undefined /*out*/;
            resourceInputs["identityDNSRecordValue1"] = undefined /*out*/;
            resourceInputs["identityDNSRecordValue2"] = undefined /*out*/;
            resourceInputs["identityDNSRecordValue3"] = undefined /*out*/;
        }
        else {
            resourceInputs["dkimSigningEnabled"] = undefined /*out*/;
            resourceInputs["feedbackForwardingEnabled"] = undefined /*out*/;
            resourceInputs["identityDNSRecordName1"] = undefined /*out*/;
            resourceInputs["identityDNSRecordName2"] = undefined /*out*/;
            resourceInputs["identityDNSRecordName3"] = undefined /*out*/;
            resourceInputs["identityDNSRecordValue1"] = undefined /*out*/;
            resourceInputs["identityDNSRecordValue2"] = undefined /*out*/;
            resourceInputs["identityDNSRecordValue3"] = undefined /*out*/;
            resourceInputs["mailFromAttributes"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Identity.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Identity resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Identity is deprecated: Identity is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Identity(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Identity.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Identity.__pulumiType;
    }
}
exports.Identity = Identity;
/** @internal */
Identity.__pulumiType = 'aws-native:pinpointemail:Identity';
