// @pulumi/aws-native/pinpointemail/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Identity = exports.getIdentityOutput = exports.getIdentity = exports.getDedicatedIpPoolOutput = exports.getDedicatedIpPool = exports.getConfigurationSetEventDestinationOutput = exports.getConfigurationSetEventDestination = exports.getConfigurationSetOutput = exports.getConfigurationSet = exports.DedicatedIpPool = exports.ConfigurationSetEventDestination = exports.ConfigurationSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ConfigurationSet = null;
utilities.lazyLoad(exports, ["ConfigurationSet"], () => require("./configurationSet"));
exports.ConfigurationSetEventDestination = null;
utilities.lazyLoad(exports, ["ConfigurationSetEventDestination"], () => require("./configurationSetEventDestination"));
exports.DedicatedIpPool = null;
utilities.lazyLoad(exports, ["DedicatedIpPool"], () => require("./dedicatedIpPool"));
exports.getConfigurationSet = null;
exports.getConfigurationSetOutput = null;
utilities.lazyLoad(exports, ["getConfigurationSet", "getConfigurationSetOutput"], () => require("./getConfigurationSet"));
exports.getConfigurationSetEventDestination = null;
exports.getConfigurationSetEventDestinationOutput = null;
utilities.lazyLoad(exports, ["getConfigurationSetEventDestination", "getConfigurationSetEventDestinationOutput"], () => require("./getConfigurationSetEventDestination"));
exports.getDedicatedIpPool = null;
exports.getDedicatedIpPoolOutput = null;
utilities.lazyLoad(exports, ["getDedicatedIpPool", "getDedicatedIpPoolOutput"], () => require("./getDedicatedIpPool"));
exports.getIdentity = null;
exports.getIdentityOutput = null;
utilities.lazyLoad(exports, ["getIdentity", "getIdentityOutput"], () => require("./getIdentity"));
exports.Identity = null;
utilities.lazyLoad(exports, ["Identity"], () => require("./identity"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:pinpointemail:ConfigurationSet":
                return new exports.ConfigurationSet(name, undefined, { urn });
            case "aws-native:pinpointemail:ConfigurationSetEventDestination":
                return new exports.ConfigurationSetEventDestination(name, undefined, { urn });
            case "aws-native:pinpointemail:DedicatedIpPool":
                return new exports.DedicatedIpPool(name, undefined, { urn });
            case "aws-native:pinpointemail:Identity":
                return new exports.Identity(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "pinpointemail", _module);
