// @pulumi/aws-native/applicationautoscaling/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScalingPolicy = exports.ScalableTarget = exports.getScalingPolicyOutput = exports.getScalingPolicy = exports.getScalableTargetOutput = exports.getScalableTarget = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getScalableTarget = null;
exports.getScalableTargetOutput = null;
utilities.lazyLoad(exports, ["getScalableTarget", "getScalableTargetOutput"], () => require("./getScalableTarget"));
exports.getScalingPolicy = null;
exports.getScalingPolicyOutput = null;
utilities.lazyLoad(exports, ["getScalingPolicy", "getScalingPolicyOutput"], () => require("./getScalingPolicy"));
exports.ScalableTarget = null;
utilities.lazyLoad(exports, ["ScalableTarget"], () => require("./scalableTarget"));
exports.ScalingPolicy = null;
utilities.lazyLoad(exports, ["ScalingPolicy"], () => require("./scalingPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:applicationautoscaling:ScalableTarget":
                return new exports.ScalableTarget(name, undefined, { urn });
            case "aws-native:applicationautoscaling:ScalingPolicy":
                return new exports.ScalingPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "applicationautoscaling", _module);
