// @pulumi/aws-native/appstream/fleet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Fleet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::AppStream::Fleet
 *
 * @deprecated Fleet is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Fleet extends pulumi.CustomResource {
    /**
     * Create a Fleet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Fleet is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Fleet is deprecated: Fleet is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.instanceType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceType'");
            }
            resourceInputs["computeCapacity"] = args ? args.computeCapacity : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["disconnectTimeoutInSeconds"] = args ? args.disconnectTimeoutInSeconds : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["domainJoinInfo"] = args ? args.domainJoinInfo : undefined;
            resourceInputs["enableDefaultInternetAccess"] = args ? args.enableDefaultInternetAccess : undefined;
            resourceInputs["fleetType"] = args ? args.fleetType : undefined;
            resourceInputs["iamRoleArn"] = args ? args.iamRoleArn : undefined;
            resourceInputs["idleDisconnectTimeoutInSeconds"] = args ? args.idleDisconnectTimeoutInSeconds : undefined;
            resourceInputs["imageArn"] = args ? args.imageArn : undefined;
            resourceInputs["imageName"] = args ? args.imageName : undefined;
            resourceInputs["instanceType"] = args ? args.instanceType : undefined;
            resourceInputs["maxConcurrentSessions"] = args ? args.maxConcurrentSessions : undefined;
            resourceInputs["maxUserDurationInSeconds"] = args ? args.maxUserDurationInSeconds : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["platform"] = args ? args.platform : undefined;
            resourceInputs["sessionScriptS3Location"] = args ? args.sessionScriptS3Location : undefined;
            resourceInputs["streamView"] = args ? args.streamView : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["usbDeviceFilterStrings"] = args ? args.usbDeviceFilterStrings : undefined;
            resourceInputs["vpcConfig"] = args ? args.vpcConfig : undefined;
        }
        else {
            resourceInputs["computeCapacity"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["disconnectTimeoutInSeconds"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["domainJoinInfo"] = undefined /*out*/;
            resourceInputs["enableDefaultInternetAccess"] = undefined /*out*/;
            resourceInputs["fleetType"] = undefined /*out*/;
            resourceInputs["iamRoleArn"] = undefined /*out*/;
            resourceInputs["idleDisconnectTimeoutInSeconds"] = undefined /*out*/;
            resourceInputs["imageArn"] = undefined /*out*/;
            resourceInputs["imageName"] = undefined /*out*/;
            resourceInputs["instanceType"] = undefined /*out*/;
            resourceInputs["maxConcurrentSessions"] = undefined /*out*/;
            resourceInputs["maxUserDurationInSeconds"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["platform"] = undefined /*out*/;
            resourceInputs["sessionScriptS3Location"] = undefined /*out*/;
            resourceInputs["streamView"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["usbDeviceFilterStrings"] = undefined /*out*/;
            resourceInputs["vpcConfig"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Fleet.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Fleet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Fleet is deprecated: Fleet is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Fleet(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Fleet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Fleet.__pulumiType;
    }
}
exports.Fleet = Fleet;
/** @internal */
Fleet.__pulumiType = 'aws-native:appstream:Fleet';
