// @pulumi/aws-native/appstream/stack.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Stack = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::AppStream::Stack
 *
 * @deprecated Stack is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Stack extends pulumi.CustomResource {
    /**
     * Create a Stack resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Stack is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Stack is deprecated: Stack is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["accessEndpoints"] = args ? args.accessEndpoints : undefined;
            resourceInputs["applicationSettings"] = args ? args.applicationSettings : undefined;
            resourceInputs["attributesToDelete"] = args ? args.attributesToDelete : undefined;
            resourceInputs["deleteStorageConnectors"] = args ? args.deleteStorageConnectors : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["embedHostDomains"] = args ? args.embedHostDomains : undefined;
            resourceInputs["feedbackURL"] = args ? args.feedbackURL : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["redirectURL"] = args ? args.redirectURL : undefined;
            resourceInputs["storageConnectors"] = args ? args.storageConnectors : undefined;
            resourceInputs["streamingExperienceSettings"] = args ? args.streamingExperienceSettings : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["userSettings"] = args ? args.userSettings : undefined;
        }
        else {
            resourceInputs["accessEndpoints"] = undefined /*out*/;
            resourceInputs["applicationSettings"] = undefined /*out*/;
            resourceInputs["attributesToDelete"] = undefined /*out*/;
            resourceInputs["deleteStorageConnectors"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["embedHostDomains"] = undefined /*out*/;
            resourceInputs["feedbackURL"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["redirectURL"] = undefined /*out*/;
            resourceInputs["storageConnectors"] = undefined /*out*/;
            resourceInputs["streamingExperienceSettings"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["userSettings"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Stack.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Stack resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Stack is deprecated: Stack is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Stack(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Stack.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Stack.__pulumiType;
    }
}
exports.Stack = Stack;
/** @internal */
Stack.__pulumiType = 'aws-native:appstream:Stack';
