// @pulumi/aws-native/types/enums/refactorspaces/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceEndpointType = exports.RouteType = exports.RouteMethod = exports.RouteActivationState = exports.EnvironmentNetworkFabricType = exports.ApplicationProxyType = exports.ApplicationApiGatewayEndpointType = void 0;
exports.ApplicationApiGatewayEndpointType = {
    Regional: "REGIONAL",
    Private: "PRIVATE",
};
exports.ApplicationProxyType = {
    ApiGateway: "API_GATEWAY",
};
exports.EnvironmentNetworkFabricType = {
    TransitGateway: "TRANSIT_GATEWAY",
};
exports.RouteActivationState = {
    Inactive: "INACTIVE",
    Active: "ACTIVE",
};
exports.RouteMethod = {
    Delete: "DELETE",
    Get: "GET",
    Head: "HEAD",
    Options: "OPTIONS",
    Patch: "PATCH",
    Post: "POST",
    Put: "PUT",
};
exports.RouteType = {
    Default: "DEFAULT",
    UriPath: "URI_PATH",
};
exports.ServiceEndpointType = {
    Lambda: "LAMBDA",
    Url: "URL",
};
