// @pulumi/aws-native/applicationinsights/application.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::ApplicationInsights::Application
 */
class Application extends pulumi.CustomResource {
    /**
     * Create a Application resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["autoConfigurationEnabled"] = args ? args.autoConfigurationEnabled : undefined;
            resourceInputs["cWEMonitorEnabled"] = args ? args.cWEMonitorEnabled : undefined;
            resourceInputs["componentMonitoringSettings"] = args ? args.componentMonitoringSettings : undefined;
            resourceInputs["customComponents"] = args ? args.customComponents : undefined;
            resourceInputs["groupingType"] = args ? args.groupingType : undefined;
            resourceInputs["logPatternSets"] = args ? args.logPatternSets : undefined;
            resourceInputs["opsCenterEnabled"] = args ? args.opsCenterEnabled : undefined;
            resourceInputs["opsItemSNSTopicArn"] = args ? args.opsItemSNSTopicArn : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["applicationARN"] = undefined /*out*/;
        }
        else {
            resourceInputs["applicationARN"] = undefined /*out*/;
            resourceInputs["autoConfigurationEnabled"] = undefined /*out*/;
            resourceInputs["cWEMonitorEnabled"] = undefined /*out*/;
            resourceInputs["componentMonitoringSettings"] = undefined /*out*/;
            resourceInputs["customComponents"] = undefined /*out*/;
            resourceInputs["groupingType"] = undefined /*out*/;
            resourceInputs["logPatternSets"] = undefined /*out*/;
            resourceInputs["opsCenterEnabled"] = undefined /*out*/;
            resourceInputs["opsItemSNSTopicArn"] = undefined /*out*/;
            resourceInputs["resourceGroupName"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Application.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Application resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Application(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Application.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Application.__pulumiType;
    }
}
exports.Application = Application;
/** @internal */
Application.__pulumiType = 'aws-native:applicationinsights:Application';
