// @pulumi/aws-native/types/enums/kinesisfirehose/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeliveryStreamType = exports.DeliveryStreamSplunkDestinationConfigurationHECEndpointType = exports.DeliveryStreamS3DestinationConfigurationCompressionFormat = exports.DeliveryStreamRedshiftDestinationConfigurationS3BackupMode = exports.DeliveryStreamProcessorType = exports.DeliveryStreamHttpEndpointRequestConfigurationContentEncoding = exports.DeliveryStreamExtendedS3DestinationConfigurationS3BackupMode = exports.DeliveryStreamExtendedS3DestinationConfigurationCompressionFormat = exports.DeliveryStreamEncryptionConfigurationNoEncryptionConfig = exports.DeliveryStreamEncryptionConfigurationInputKeyType = exports.DeliveryStreamElasticsearchDestinationConfigurationS3BackupMode = exports.DeliveryStreamElasticsearchDestinationConfigurationIndexRotationPeriod = exports.DeliveryStreamAmazonopensearchserviceDestinationConfigurationS3BackupMode = exports.DeliveryStreamAmazonopensearchserviceDestinationConfigurationIndexRotationPeriod = void 0;
exports.DeliveryStreamAmazonopensearchserviceDestinationConfigurationIndexRotationPeriod = {
    NoRotation: "NoRotation",
    OneHour: "OneHour",
    OneDay: "OneDay",
    OneWeek: "OneWeek",
    OneMonth: "OneMonth",
};
exports.DeliveryStreamAmazonopensearchserviceDestinationConfigurationS3BackupMode = {
    FailedDocumentsOnly: "FailedDocumentsOnly",
    AllDocuments: "AllDocuments",
};
exports.DeliveryStreamElasticsearchDestinationConfigurationIndexRotationPeriod = {
    NoRotation: "NoRotation",
    OneHour: "OneHour",
    OneDay: "OneDay",
    OneWeek: "OneWeek",
    OneMonth: "OneMonth",
};
exports.DeliveryStreamElasticsearchDestinationConfigurationS3BackupMode = {
    FailedDocumentsOnly: "FailedDocumentsOnly",
    AllDocuments: "AllDocuments",
};
exports.DeliveryStreamEncryptionConfigurationInputKeyType = {
    AwsOwnedCmk: "AWS_OWNED_CMK",
    CustomerManagedCmk: "CUSTOMER_MANAGED_CMK",
};
exports.DeliveryStreamEncryptionConfigurationNoEncryptionConfig = {
    NoEncryption: "NoEncryption",
};
exports.DeliveryStreamExtendedS3DestinationConfigurationCompressionFormat = {
    Uncompressed: "UNCOMPRESSED",
    Gzip: "GZIP",
    Zip: "ZIP",
    Snappy: "Snappy",
    HadoopSnappy: "HADOOP_SNAPPY",
};
exports.DeliveryStreamExtendedS3DestinationConfigurationS3BackupMode = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.DeliveryStreamHttpEndpointRequestConfigurationContentEncoding = {
    None: "NONE",
    Gzip: "GZIP",
};
exports.DeliveryStreamProcessorType = {
    RecordDeAggregation: "RecordDeAggregation",
    Lambda: "Lambda",
    MetadataExtraction: "MetadataExtraction",
    AppendDelimiterToRecord: "AppendDelimiterToRecord",
};
exports.DeliveryStreamRedshiftDestinationConfigurationS3BackupMode = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.DeliveryStreamS3DestinationConfigurationCompressionFormat = {
    Uncompressed: "UNCOMPRESSED",
    Gzip: "GZIP",
    Zip: "ZIP",
    Snappy: "Snappy",
    HadoopSnappy: "HADOOP_SNAPPY",
};
exports.DeliveryStreamSplunkDestinationConfigurationHECEndpointType = {
    Raw: "Raw",
    Event: "Event",
};
exports.DeliveryStreamType = {
    DirectPut: "DirectPut",
    KinesisStreamAsSource: "KinesisStreamAsSource",
};
