// @pulumi/aws-native/types/enums/cur/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportDefinitionTimeUnit = exports.ReportDefinitionReportVersioning = exports.ReportDefinitionFormat = exports.ReportDefinitionCompression = exports.ReportDefinitionAdditionalSchemaElementsItem = exports.ReportDefinitionAdditionalArtifactsItem = void 0;
exports.ReportDefinitionAdditionalArtifactsItem = {
    Redshift: "REDSHIFT",
    Quicksight: "QUICKSIGHT",
    Athena: "ATHENA",
};
exports.ReportDefinitionAdditionalSchemaElementsItem = {
    Resources: "RESOURCES",
};
exports.ReportDefinitionCompression = {
    Zip: "ZIP",
    Gzip: "GZIP",
    Parquet: "Parquet",
};
exports.ReportDefinitionFormat = {
    TextORcsv: "textORcsv",
    Parquet: "Parquet",
};
exports.ReportDefinitionReportVersioning = {
    CreateNewReport: "CREATE_NEW_REPORT",
    OverwriteReport: "OVERWRITE_REPORT",
};
exports.ReportDefinitionTimeUnit = {
    Hourly: "HOURLY",
    Daily: "DAILY",
    Monthly: "MONTHLY",
};
