// @pulumi/aws-native/acmpca/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Permission = exports.getCertificateAuthorityActivationOutput = exports.getCertificateAuthorityActivation = exports.getCertificateAuthorityOutput = exports.getCertificateAuthority = exports.getCertificateOutput = exports.getCertificate = exports.CertificateAuthorityActivation = exports.CertificateAuthority = exports.Certificate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Certificate = null;
utilities.lazyLoad(exports, ["Certificate"], () => require("./certificate"));
exports.CertificateAuthority = null;
utilities.lazyLoad(exports, ["CertificateAuthority"], () => require("./certificateAuthority"));
exports.CertificateAuthorityActivation = null;
utilities.lazyLoad(exports, ["CertificateAuthorityActivation"], () => require("./certificateAuthorityActivation"));
exports.getCertificate = null;
exports.getCertificateOutput = null;
utilities.lazyLoad(exports, ["getCertificate", "getCertificateOutput"], () => require("./getCertificate"));
exports.getCertificateAuthority = null;
exports.getCertificateAuthorityOutput = null;
utilities.lazyLoad(exports, ["getCertificateAuthority", "getCertificateAuthorityOutput"], () => require("./getCertificateAuthority"));
exports.getCertificateAuthorityActivation = null;
exports.getCertificateAuthorityActivationOutput = null;
utilities.lazyLoad(exports, ["getCertificateAuthorityActivation", "getCertificateAuthorityActivationOutput"], () => require("./getCertificateAuthorityActivation"));
exports.Permission = null;
utilities.lazyLoad(exports, ["Permission"], () => require("./permission"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:acmpca:Certificate":
                return new exports.Certificate(name, undefined, { urn });
            case "aws-native:acmpca:CertificateAuthority":
                return new exports.CertificateAuthority(name, undefined, { urn });
            case "aws-native:acmpca:CertificateAuthorityActivation":
                return new exports.CertificateAuthorityActivation(name, undefined, { urn });
            case "aws-native:acmpca:Permission":
                return new exports.Permission(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "acmpca", _module);
