// @pulumi/aws-native/types/enums/appflow/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowZendeskConnectorOperator = exports.FlowWriteOperationType = exports.FlowVeevaConnectorOperator = exports.FlowTriggerType = exports.FlowTrendmicroConnectorOperator = exports.FlowTaskType = exports.FlowSlackConnectorOperator = exports.FlowSingularConnectorOperator = exports.FlowServiceNowConnectorOperator = exports.FlowScheduledTriggerPropertiesDataPullMode = exports.FlowSalesforceConnectorOperator = exports.FlowSAPODataConnectorOperator = exports.FlowS3InputFormatConfigS3InputFileType = exports.FlowS3ConnectorOperator = exports.FlowPrefixType = exports.FlowPrefixFormat = exports.FlowOperatorPropertiesKeys = exports.FlowOperator = exports.FlowMarketoConnectorOperator = exports.FlowInforNexusConnectorOperator = exports.FlowGoogleAnalyticsConnectorOperator = exports.FlowFileType = exports.FlowDynatraceConnectorOperator = exports.FlowDatadogConnectorOperator = exports.FlowDataTransferApi = exports.FlowConnectorType = exports.FlowAmplitudeConnectorOperator = exports.FlowAggregationType = exports.ConnectorProfileOAuth2GrantType = exports.ConnectorProfileConnectorType = exports.ConnectorProfileConnectionMode = exports.ConnectorProfileAuthenticationType = void 0;
exports.ConnectorProfileAuthenticationType = {
    Oauth2: "OAUTH2",
    Apikey: "APIKEY",
    Basic: "BASIC",
    Custom: "CUSTOM",
};
exports.ConnectorProfileConnectionMode = {
    Public: "Public",
    Private: "Private",
};
exports.ConnectorProfileConnectorType = {
    Salesforce: "Salesforce",
    Singular: "Singular",
    Slack: "Slack",
    Redshift: "Redshift",
    Marketo: "Marketo",
    Googleanalytics: "Googleanalytics",
    Zendesk: "Zendesk",
    Servicenow: "Servicenow",
    SAPOData: "SAPOData",
    Datadog: "Datadog",
    Trendmicro: "Trendmicro",
    Snowflake: "Snowflake",
    Dynatrace: "Dynatrace",
    Infornexus: "Infornexus",
    Amplitude: "Amplitude",
    Veeva: "Veeva",
    CustomConnector: "CustomConnector",
};
exports.ConnectorProfileOAuth2GrantType = {
    ClientCredentials: "CLIENT_CREDENTIALS",
    AuthorizationCode: "AUTHORIZATION_CODE",
};
exports.FlowAggregationType = {
    None: "None",
    SingleFile: "SingleFile",
};
exports.FlowAmplitudeConnectorOperator = {
    Between: "BETWEEN",
};
exports.FlowConnectorType = {
    SAPOData: "SAPOData",
    Salesforce: "Salesforce",
    Singular: "Singular",
    Slack: "Slack",
    Redshift: "Redshift",
    S3: "S3",
    Marketo: "Marketo",
    Googleanalytics: "Googleanalytics",
    Zendesk: "Zendesk",
    Servicenow: "Servicenow",
    Datadog: "Datadog",
    Trendmicro: "Trendmicro",
    Snowflake: "Snowflake",
    Dynatrace: "Dynatrace",
    Infornexus: "Infornexus",
    Amplitude: "Amplitude",
    Veeva: "Veeva",
    CustomConnector: "CustomConnector",
    EventBridge: "EventBridge",
    Upsolver: "Upsolver",
    LookoutMetrics: "LookoutMetrics",
};
exports.FlowDataTransferApi = {
    Automatic: "AUTOMATIC",
    Bulkv2: "BULKV2",
    RestSync: "REST_SYNC",
};
exports.FlowDatadogConnectorOperator = {
    Projection: "PROJECTION",
    Between: "BETWEEN",
    EqualTo: "EQUAL_TO",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowDynatraceConnectorOperator = {
    Projection: "PROJECTION",
    Between: "BETWEEN",
    EqualTo: "EQUAL_TO",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowFileType = {
    Csv: "CSV",
    Json: "JSON",
    Parquet: "PARQUET",
};
exports.FlowGoogleAnalyticsConnectorOperator = {
    Projection: "PROJECTION",
    Between: "BETWEEN",
};
exports.FlowInforNexusConnectorOperator = {
    Projection: "PROJECTION",
    Between: "BETWEEN",
    EqualTo: "EQUAL_TO",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowMarketoConnectorOperator = {
    Projection: "PROJECTION",
    LessThan: "LESS_THAN",
    GreaterThan: "GREATER_THAN",
    Between: "BETWEEN",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowOperator = {
    Projection: "PROJECTION",
    LessThan: "LESS_THAN",
    GreaterThan: "GREATER_THAN",
    Contains: "CONTAINS",
    Between: "BETWEEN",
    LessThanOrEqualTo: "LESS_THAN_OR_EQUAL_TO",
    GreaterThanOrEqualTo: "GREATER_THAN_OR_EQUAL_TO",
    EqualTo: "EQUAL_TO",
    NotEqualTo: "NOT_EQUAL_TO",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowOperatorPropertiesKeys = {
    Value: "VALUE",
    Values: "VALUES",
    DataType: "DATA_TYPE",
    UpperBound: "UPPER_BOUND",
    LowerBound: "LOWER_BOUND",
    SourceDataType: "SOURCE_DATA_TYPE",
    DestinationDataType: "DESTINATION_DATA_TYPE",
    ValidationAction: "VALIDATION_ACTION",
    MaskValue: "MASK_VALUE",
    MaskLength: "MASK_LENGTH",
    TruncateLength: "TRUNCATE_LENGTH",
    MathOperationFieldsOrder: "MATH_OPERATION_FIELDS_ORDER",
    ConcatFormat: "CONCAT_FORMAT",
    SubfieldCategoryMap: "SUBFIELD_CATEGORY_MAP",
    ExcludeSourceFieldsList: "EXCLUDE_SOURCE_FIELDS_LIST",
    IncludeNewFields: "INCLUDE_NEW_FIELDS",
};
exports.FlowPrefixFormat = {
    Year: "YEAR",
    Month: "MONTH",
    Day: "DAY",
    Hour: "HOUR",
    Minute: "MINUTE",
};
exports.FlowPrefixType = {
    Filename: "FILENAME",
    Path: "PATH",
    PathAndFilename: "PATH_AND_FILENAME",
};
exports.FlowS3ConnectorOperator = {
    Projection: "PROJECTION",
    LessThan: "LESS_THAN",
    GreaterThan: "GREATER_THAN",
    Between: "BETWEEN",
    LessThanOrEqualTo: "LESS_THAN_OR_EQUAL_TO",
    GreaterThanOrEqualTo: "GREATER_THAN_OR_EQUAL_TO",
    EqualTo: "EQUAL_TO",
    NotEqualTo: "NOT_EQUAL_TO",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowS3InputFormatConfigS3InputFileType = {
    Csv: "CSV",
    Json: "JSON",
};
exports.FlowSAPODataConnectorOperator = {
    Projection: "PROJECTION",
    LessThan: "LESS_THAN",
    Contains: "CONTAINS",
    GreaterThan: "GREATER_THAN",
    Between: "BETWEEN",
    LessThanOrEqualTo: "LESS_THAN_OR_EQUAL_TO",
    GreaterThanOrEqualTo: "GREATER_THAN_OR_EQUAL_TO",
    EqualTo: "EQUAL_TO",
    NotEqualTo: "NOT_EQUAL_TO",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowSalesforceConnectorOperator = {
    Projection: "PROJECTION",
    LessThan: "LESS_THAN",
    Contains: "CONTAINS",
    GreaterThan: "GREATER_THAN",
    Between: "BETWEEN",
    LessThanOrEqualTo: "LESS_THAN_OR_EQUAL_TO",
    GreaterThanOrEqualTo: "GREATER_THAN_OR_EQUAL_TO",
    EqualTo: "EQUAL_TO",
    NotEqualTo: "NOT_EQUAL_TO",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowScheduledTriggerPropertiesDataPullMode = {
    Incremental: "Incremental",
    Complete: "Complete",
};
exports.FlowServiceNowConnectorOperator = {
    Projection: "PROJECTION",
    LessThan: "LESS_THAN",
    Contains: "CONTAINS",
    GreaterThan: "GREATER_THAN",
    Between: "BETWEEN",
    LessThanOrEqualTo: "LESS_THAN_OR_EQUAL_TO",
    GreaterThanOrEqualTo: "GREATER_THAN_OR_EQUAL_TO",
    EqualTo: "EQUAL_TO",
    NotEqualTo: "NOT_EQUAL_TO",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowSingularConnectorOperator = {
    Projection: "PROJECTION",
    EqualTo: "EQUAL_TO",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowSlackConnectorOperator = {
    Projection: "PROJECTION",
    Between: "BETWEEN",
    EqualTo: "EQUAL_TO",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowTaskType = {
    Arithmetic: "Arithmetic",
    Filter: "Filter",
    Map: "Map",
    MapAll: "Map_all",
    Mask: "Mask",
    Merge: "Merge",
    Passthrough: "Passthrough",
    Truncate: "Truncate",
    Validate: "Validate",
};
exports.FlowTrendmicroConnectorOperator = {
    Projection: "PROJECTION",
    EqualTo: "EQUAL_TO",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowTriggerType = {
    Scheduled: "Scheduled",
    Event: "Event",
    OnDemand: "OnDemand",
};
exports.FlowVeevaConnectorOperator = {
    Projection: "PROJECTION",
    LessThan: "LESS_THAN",
    GreaterThan: "GREATER_THAN",
    Between: "BETWEEN",
    LessThanOrEqualTo: "LESS_THAN_OR_EQUAL_TO",
    GreaterThanOrEqualTo: "GREATER_THAN_OR_EQUAL_TO",
    EqualTo: "EQUAL_TO",
    NotEqualTo: "NOT_EQUAL_TO",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
exports.FlowWriteOperationType = {
    Insert: "INSERT",
    Upsert: "UPSERT",
    Update: "UPDATE",
    Delete: "DELETE",
};
exports.FlowZendeskConnectorOperator = {
    Projection: "PROJECTION",
    GreaterThan: "GREATER_THAN",
    Addition: "ADDITION",
    Multiplication: "MULTIPLICATION",
    Division: "DIVISION",
    Subtraction: "SUBTRACTION",
    MaskAll: "MASK_ALL",
    MaskFirstN: "MASK_FIRST_N",
    MaskLastN: "MASK_LAST_N",
    ValidateNonNull: "VALIDATE_NON_NULL",
    ValidateNonZero: "VALIDATE_NON_ZERO",
    ValidateNonNegative: "VALIDATE_NON_NEGATIVE",
    ValidateNumeric: "VALIDATE_NUMERIC",
    NoOp: "NO_OP",
};
